<h1>Baseline Atlas of Gene Expression: Help</h1>



<h2>About The Baseline Atlas</h2>
	
<p>The Baseline Atlas aims primarily at representing information about which gene
products are present (and at what abundance) in which "normal" condition (e.g.,
tissue, cell type). In the first instance, the Baseline Atlas will consist of
highly curated RNA&ndash;seq experiments from <a
	href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a>, such as: </p>

<ul>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-513" rel="nofollow" title="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-513">Illumina Body Map</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-599" rel="nofollow" title="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-599">Transcriptome of DBAxC57BL/6J mice</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284" rel="nofollow" title="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284">RNA&ndash;seq of long coding and long non coding RNAs from ENCODE cell lines</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-30352" rel="nofollow" title="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-30352">RNA&ndash;seq of 6 tissues from 10 species to investigate the evolution of gene expression levels in mammalian organs</a></li>
</ul>

<p>The above experiments have been re&ndash;analyzed using an in&ndash;house RNA&ndash;seq
processing pipeline. The resulting data matrix stores gene
expression profiles across the analyzed experimental factor values as a set of
normalized expression levels (<a
	class="externallink"
	title="FPKMs"
	href="http://www.nature.com/nbt/journal/v28/n5/full/nbt.1621.html">FPKMs</a>).
You can search the data by a combination of genes attributes, experimental
factor values, and a minimum FPKM cutoff. See below for more details on searching.</p>



<h2>Baseline Atlas At a Glance</h2>

<p><img alt="Baseline Atlas Experiment View 'At a Glance'" src="resources/images/help/exptPageHelp.png" align="bottom" /></p>



<h2>Filtering Baseline Atlas Data</h2>

<p><img alt="filtering" src="resources/images/help/filtering.png" align="bottom" /></p>
<p>Some experiments have <b>more than one</b> experimental variable (factor). For example, in experiment
<a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284"
	rel="nofollow"
	title="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284">E-GEOD-26284</a>,
three different <b>types of RNA</b> were extracted from <b>six cellular components</b> from <b>23
human cell lines</b>. Results for these experiments are limited to a certain
combination of variables, e.g. showing expression of <b>total RNA</b> in the
<b>whole cell</b>. To limit the search results differently, use the "Change
filters" menu. In a 3&ndash;factor experiment, you can choose one value from any two
of the factors, and then search among the values for the remaining factor.
E.g., if you choose to limit to <b>poly(A) RNA</b> extracted from the
<b>cytosol</b>, you can then search among the <b>cell lines</b> for which this
data is available.</p>



<h2>Searching Baseline Atlas</h2>


<h3>Searching by genes</h3>

<p>You can search by <a class="externallink" href="http://ensembl.org">Ensembl</a> gene symbols (e.g. SAA4), identifiers (e.g. ENSG00000148965) or <a title="Ensembl Glossary" class="externallink" href="http://www.ensembl.org/Help/Glossary">biotypes</a> (e.g. protein_coding), <a class="externallink" href="http://www.uniprot.org">UniProt</a> accessions (e.g. P35542), <a class="externallink" href="http://www.geneontology.org">GO</a> ("phosphoglycerate kinase activity") or <a href="http://www.ebi.ac.uk/interpro/">InterPro</a> terms (e.g. phosphatase). A <b>space&ndash;separated</b> list of gene attributes will bring back genes that match <b>at least one</b> of the attributes in the query.</p>


<h3>Searching by experimental factor values</h3>

<p>Use the factor value input box to search for experimental factor (e.g. <i>organism part</i> or <i>cell line</i>) values, or leave it blank to search in all. You can search:</p>
	
<ul>
	<li>within a <b>single&ndash;factor</b> experiment, e.g. <a href="experiments/E-MTAB-513" title="E-MTAB-513">Illumina Body Map</a><a title="E-MTAB-513" href="experiments/E-MTAB-513">,</a> or</li>
	<li>within a pre&ndash;selected subset of a <b>multi&ndash;factorial</b> experiment, e.g. <a href="experiments/E-GEOD-26284" rel="nofollow" title="E-GEOD-26284">RNA&ndash;seq of long coding and long non coding RNAs from ENCODE cell lines</a></li>
</ul>

<p>When you click on the experiment factor value input box, a dropdown menu appears, containing the list of all factor values available for selection:</p>
<p><img alt="Factor values dropdown" src="resources/images/help/factor_values_dropdown.png" align="bottom" /></p>
<p>Select one or more factor values to search with, or start typing to see suggestions. To see all factor values in an experiment, click the <img alt="Experiment Design" src="resources/images/help/expDes_btn.png" align="bottom" /> button.</p>


<h4><i>Specific</i> vs. <i>non&ndash;specific</i> search</h4>
	
<p>The default Baseline Atlas search reports first genes with more "specific" expression (with the <i>Specific</i> option selected by default). What we mean by this is outlined below:</p>
	
	<ul>
		<li>If the query contains <b>no experimental factor values</b>, the <i>Specific</i> search will report genes expressed in just <b>one</b> experimental factor value, then genes expressed in just <b>two</b> factor values, then three and so on, reporting any genes expressed in all the available factor values at the end of the list of results. Within each group of genes expressed in N factor values only, genes are sorted by the highest average FPKM across the N factor values in which they are expressed.</li>
		<li>If the query contains <b>at least one</b> experimental factor value, the <i>Specific</i> search will promote genes with <b>higher FPKM</b> in the <b>queried</b> factor value(s), and at the same time penalize genes with a higher FPKM across the <b>non&ndash;queried</b> factor values. In other words, genes with the <b>greatest difference</b> in expression between queried and non&ndash;queried tissues will be pushed towards the top of the list.</li>
  </ul>
  
  <p>If the <i>Specific</i> option is not selected, the Baseline Atlas search will <b>ignore</b> specificity of expression when ordering the results. In particular:</p>
 
  <ul>
	  <li>If the query contains <b>no experimental factor values</b>, genes with the highest expression across <b>all</b> factor values will be reported first.</li>
	  <li>If the query contains <b>at least one</b> experimental factor value, genes with higher FPKM across the <b>selected</b> factor values are reported first (expression levels across the non&ndash;selected factor values is <b>ignored</b> in the ordering of results).</li>
  </ul>




<h3>Specifying the minimum expression level</h3>

<p>You can specify a minimum expression level to act as a <b>cutoff</b>, and only genes expressed <b>above</b> this level are displayed. This is to aid distinction between "background noise" and "real" gene expression. Our default FPKM cutoff is 0.5. To change it, enter a new value into the "Expression level cutoff" input box, or drag the slider until the "Expression level cutoff" input box shows your desired cutoff value. A histogram showing the number of genes expressed above a range of cutoffs is also available. This lets you see how many genes you are excluding by selecting a given cutoff. If you see that you're excluding a lot of genes, you may wish to <b>decrease</b> the cutoff. Clicking on the <img alt="Histogram enable/disable button" src="resources/images/help/histogram_btn.png" align="bottom" /> button toggles display of the histogram.</p>

<ul>
	<li>If you have not selected any experimental factors, the histogram shows the number of genes expressed in <b>any</b> factor value for each of the shown cutoffs.</li>
	<li>If you have selected <b>at least one</b> factor value, the histogram shows the number of genes expressed in <b>at least one of the selected factor values</b>, for each of the shown cutoffs.</li>
</ul>



<h2>Heatmap Results Display</h2>

<p>Results for your query are shown in a heatmap table. The heatmap rows and columns are labelled with gene symbols and experimental factor values, respectively. The heatmap shows expressions by <b>colour intensity</b>, according to the gradient bar,<img alt="heatmap colour gradient" src="resources/images/help/gradient.png" align="bottom" />, displayed above the heatmap. Greater colour intensity means higher relative gene expression. To see the exact FPKM for a given gene in a given factor value, click on the corresponding box. To reveal this for all genes, click on the <img alt="Display levels" src="resources/images/help/display_levels.png" align="bottom" /> button. This also reveals minimum and maximum FPKMs on each side of the gradient bar. The gradient shows intensities corresponding to expression levels for the <b>top 50</b> genes currently displayed (rather than all FPKM values for all genes returned by the query).</p>

  <p>When displayed factor values are organism parts, mousing over each heatmap column highlights the corresponding tissue in the anatomogram on the left (if available). If the selected organism part is gender&ndash;specifc, and anatomogram is available for each gender, you can switch between them by clicking the <img alt="Gender selection button" src="resources/images/help/gender_sel.png" align="bottom" />.</p>



<h2>Anatomogram</h2>

<p>The anatomogram provides a visualisation of the tissues studied in a given experiment. Mouseover a tissue in the anatomogram to highlight the corresponding column in the heatmap table.</p>



<h2>Downloading Query Results</h2>

<p>The <b>top 50</b> genes resulting from your search are displayed on the page. Click on the <img alt="Download button" src="resources/images/help/download_button.png" align="bottom" /> button to download the <b>full results</b> of your query in tab&ndash;delimited format with no ordering.</p>



<h2>Experiment Design Page</h2>

<p>The Experiment Design page (e.g. <a title="Illumina Body Map &mdash;
	Experiment Design page"
	href="experiments/E-MTAB-513/experiment-design">Illumina Body Map</a>)
shows RNA&ndash;seq processing run acessions (from <a
	href="http://www.ebi.ac.uk/ena/">ENA</a>), along with their coresponding
biological sample characteristcs and factor values. You can access it from the
experiment page by clicking the <img alt="Experiment Design button"
src="resources/images/help/expDes_btn.png" align="bottom" /> button. Only the
runs used in the Baseline Atlas are shown, although the dataset these were
taken from may contain more. To see all runs in the dataset, uncheck <i>Show
Analysed only</i>.</p>

<p>You can sort the experiment design table by clicking on a column. Click the
column again to reverse the sort order. To then sort by another column while
retaining the ordering of the first column, hold down the <b>shift</b> key and
click another column. The first selected column forms the primary sort order
and the shift+click&ndash;ed columns form secondary, tertiary, etc. sort orders
in turn. If you shift+click a column to sort by it and then wish to undo this,
just shift+click it again until the column is unsorted (<img alt="unsorted
arrows" src="resources/images/help/unsortedArrows.png" align="center" />).</p>
  
<p>The experiment design table is also searchable &mdash; entering a keyword
automatically selects the subsection of the table that matches the keyword.
Click on the <img alt="Download button"
src="resources/images/help/download_button.png" align="bottom" /> button on the
right&ndash;hand side to download the full results of the current query (with
no ordering applied) in tab&ndash;delimited format.</p>



<h2>Analysis Methods Page</h2>

<p>The Analysis Methods page (e.g. <a title="Illumina Body Map &mdash; Analysis
	Methods page" href="experiments/E-MTAB-513/analysis-methods">Illumina Body
	Map</a>) lists the data analysis methods we applied to the raw experimental
data in <a title="FASTQ format" class="externallink"
	href="http://nar.oxfordjournals.org/content/38/6/1767.full">FASTQ</a>
format, to obtain the gene expression levels shown in the Baseline Atlas. You can
access it by clicking the <img alt="Analysis Methods
button" src="resources/images/help/am_btn.png" align="bottom" /> button.</p>



<h2>Providing Feedback</h2>

<p>The Baseline Atlas is still a prototype and thus we value your feedback on what we are doing right, what does not work or what could be achieved more intuitively. We would also be grateful for any feedback on the analysis methods we have adopted &mdash; we are passionate about not only about intuitive data presentation, the highest level of experimental metadata curation, but also about the quality and biological validity of the expression data presented in Baseline Atlas.</p>

<p>To send us your feedback, please fill in form accessed by clicking on the <i>Feedback</i> link on the top right of the page:</p>
<p><img alt="Feedback Pop-up" src="resources/images/help/feedback.png" align="bottom" /></p>
