<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Tue 26 Aug 2014<br />
SRR307924_2.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307924_2.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>31203036</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>50</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td>81273</td>
<td>0.2604650393634773</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAG</td>
<td>40464</td>
<td>0.12967968886104544</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td>39458</td>
<td>0.12645564361108963</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCCAG</td>
<td>4643380</td>
<td>2.1595998</td>
<td>7.179963</td>
<td>21</td>
</tr>
<tr>
<td>CCAGG</td>
<td>4584720</td>
<td>2.1226816</td>
<td>6.477019</td>
<td>62</td>
</tr>
<tr>
<td>CCAGC</td>
<td>4404020</td>
<td>2.0482755</td>
<td>7.0409913</td>
<td>22</td>
</tr>
<tr>
<td>CTGGG</td>
<td>4685820</td>
<td>2.0040905</td>
<td>6.068676</td>
<td>71</td>
</tr>
<tr>
<td>GGCTG</td>
<td>4675440</td>
<td>1.9996512</td>
<td>6.5645714</td>
<td>36</td>
</tr>
<tr>
<td>CAGGA</td>
<td>3929800</td>
<td>1.9187517</td>
<td>6.561303</td>
<td>63</td>
</tr>
<tr>
<td>CAGCT</td>
<td>4140665</td>
<td>1.8845693</td>
<td>6.661664</td>
<td>23</td>
</tr>
<tr>
<td>GCCTG</td>
<td>4377925</td>
<td>1.880906</td>
<td>6.5614667</td>
<td>12</td>
</tr>
<tr>
<td>GGAGG</td>
<td>4023130</td>
<td>1.8458742</td>
<td>6.5504932</td>
<td>33</td>
</tr>
<tr>
<td>TGCCT</td>
<td>4357095</td>
<td>1.8402027</td>
<td>6.3836784</td>
<td>11</td>
</tr>
<tr>
<td>AGGAG</td>
<td>3654895</td>
<td>1.7764632</td>
<td>6.5220723</td>
<td>64</td>
</tr>
<tr>
<td>GAGGA</td>
<td>3647960</td>
<td>1.7730923</td>
<td>6.582924</td>
<td>47</td>
</tr>
<tr>
<td>GGTGG</td>
<td>4019455</td>
<td>1.7113225</td>
<td>6.320519</td>
<td>2</td>
</tr>
<tr>
<td>TTCTG</td>
<td>4102385</td>
<td>1.6955396</td>
<td>5.270538</td>
<td>69</td>
</tr>
<tr>
<td>CTGAG</td>
<td>3705300</td>
<td>1.6787976</td>
<td>6.3653946</td>
<td>38</td>
</tr>
<tr>
<td>TCCCA</td>
<td>3663570</td>
<td>1.674995</td>
<td>6.3630376</td>
<td>20</td>
</tr>
<tr>
<td>GCTGA</td>
<td>3655490</td>
<td>1.6562297</td>
<td>6.41605</td>
<td>37</td>
</tr>
<tr>
<td>AGGCT</td>
<td>3429675</td>
<td>1.5539174</td>
<td>6.3658967</td>
<td>35</td>
</tr>
<tr>
<td>CCTGT</td>
<td>3667485</td>
<td>1.5489488</td>
<td>6.0296235</td>
<td>13</td>
</tr>
<tr>
<td>TCTGG</td>
<td>3617290</td>
<td>1.5208452</td>
<td>5.377984</td>
<td>70</td>
</tr>
<tr>
<td>GGGAG</td>
<td>3268045</td>
<td>1.4994295</td>
<td>6.316799</td>
<td>32</td>
</tr>
<tr>
<td>GAGGC</td>
<td>3169510</td>
<td>1.4608216</td>
<td>6.396273</td>
<td>34</td>
</tr>
<tr>
<td>GTGGC</td>
<td>3402805</td>
<td>1.4553543</td>
<td>6.075704</td>
<td>3</td>
</tr>
<tr>
<td>TGAGG</td>
<td>3210515</td>
<td>1.4480469</td>
<td>6.1158333</td>
<td>39</td>
</tr>
<tr>
<td>TGGGC</td>
<td>3281765</td>
<td>1.4035865</td>
<td>5.4452333</td>
<td>72</td>
</tr>
<tr>
<td>TTGAG</td>
<td>2839490</td>
<td>1.2589782</td>
<td>5.566941</td>
<td>56</td>
</tr>
<tr>
<td>CTTGA</td>
<td>2807110</td>
<td>1.2502716</td>
<td>5.711458</td>
<td>55</td>
</tr>
<tr>
<td>GTGCC</td>
<td>2890425</td>
<td>1.2418251</td>
<td>5.854425</td>
<td>10</td>
</tr>
<tr>
<td>AGGAT</td>
<td>2556890</td>
<td>1.2216963</td>
<td>5.885671</td>
<td>48</td>
</tr>
<tr>
<td>GCTTG</td>
<td>2856975</td>
<td>1.2011801</td>
<td>5.270836</td>
<td>54</td>
</tr>
<tr>
<td>TGTAG</td>
<td>2580410</td>
<td>1.144107</td>
<td>5.403021</td>
<td>15</td>
</tr>
<tr>
<td>GGAGT</td>
<td>2477885</td>
<td>1.117607</td>
<td>5.0311456</td>
<td>65</td>
</tr>
<tr>
<td>CTGTA</td>
<td>2385505</td>
<td>1.062491</td>
<td>5.733845</td>
<td>14</td>
</tr>
<tr>
<td>AGTCC</td>
<td>2326210</td>
<td>1.0587438</td>
<td>5.707173</td>
<td>18</td>
</tr>
<tr>
<td>CTACT</td>
<td>2233690</td>
<td>0.9993897</td>
<td>5.6288934</td>
<td>26</td>
</tr>
<tr>
<td>GATCG</td>
<td>2195625</td>
<td>0.99479395</td>
<td>5.56234</td>
<td>50</td>
</tr>
<tr>
<td>TGGCG</td>
<td>2259500</td>
<td>0.96637136</td>
<td>5.7899513</td>
<td>4</td>
</tr>
<tr>
<td>GTCCC</td>
<td>2234560</td>
<td>0.9644013</td>
<td>5.2668643</td>
<td>19</td>
</tr>
<tr>
<td>GGCGC</td>
<td>2063275</td>
<td>0.9017495</td>
<td>5.578601</td>
<td>5</td>
</tr>
<tr>
<td>GGATC</td>
<td>1961460</td>
<td>0.8886986</td>
<td>5.3459287</td>
<td>49</td>
</tr>
<tr>
<td>AGCTA</td>
<td>1838235</td>
<td>0.8823061</td>
<td>5.8416195</td>
<td>24</td>
</tr>
<tr>
<td>GCTAC</td>
<td>1861090</td>
<td>0.84705067</td>
<td>5.539873</td>
<td>25</td>
</tr>
<tr>
<td>CGGTG</td>
<td>1755460</td>
<td>0.7507973</td>
<td>5.598269</td>
<td>1</td>
</tr>
<tr>
<td>CTCGG</td>
<td>1716075</td>
<td>0.73728436</td>
<td>5.210924</td>
<td>29</td>
</tr>
<tr>
<td>GCGTG</td>
<td>1672380</td>
<td>0.71526456</td>
<td>5.2161207</td>
<td>8</td>
</tr>
<tr>
<td>TACTC</td>
<td>1589780</td>
<td>0.71129376</td>
<td>5.348895</td>
<td>27</td>
</tr>
<tr>
<td>CGGGA</td>
<td>1541420</td>
<td>0.7104378</td>
<td>5.466768</td>
<td>31</td>
</tr>
<tr>
<td>CGTGC</td>
<td>1594690</td>
<td>0.6851332</td>
<td>5.2460117</td>
<td>9</td>
</tr>
<tr>
<td>GTAGT</td>
<td>1461230</td>
<td>0.6478828</td>
<td>5.095218</td>
<td>16</td>
</tr>
<tr>
<td>ATCGC</td>
<td>1379585</td>
<td>0.6279</td>
<td>5.001167</td>
<td>51</td>
</tr>
<tr>
<td>TCGGG</td>
<td>1394280</td>
<td>0.59632325</td>
<td>5.0022283</td>
<td>30</td>
</tr>
<tr>
<td>GCGCG</td>
<td>1342860</td>
<td>0.5868938</td>
<td>5.207449</td>
<td>6</td>
</tr>
<tr>
<td>TAGTC</td>
<td>1252120</td>
<td>0.55768746</td>
<td>5.0275507</td>
<td>17</td>
</tr>
<tr>
<td>ACTCG</td>
<td>936035</td>
<td>0.42602405</td>
<td>5.10607</td>
<td>28</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
