<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR317067_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317067_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>76054995</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>50</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>260907</td>
<td>0.3430504465880249</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>219666</td>
<td>0.288825211282967</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>216540</td>
<td>0.28471502759286227</td>
<td>No Hit</td>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGTCTTCTGCTTGAAAA</td>
<td>161047</td>
<td>0.21175072064629022</td>
<td>Illumina Paired End PCR Primer 2 (98% over 60bp)</td>
</tr>
<tr>
<td>CGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAA</td>
<td>95661</td>
<td>0.12577872104258242</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGAC</td>
<td>79220</td>
<td>0.10416146894756878</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CATCA</td>
<td>9555865</td>
<td>2.275612</td>
<td>5.302661</td>
<td>14</td>
</tr>
<tr>
<td>CTGCT</td>
<td>8537820</td>
<td>2.26567</td>
<td>5.0541587</td>
<td>54</td>
</tr>
<tr>
<td>CACAG</td>
<td>10808835</td>
<td>2.207511</td>
<td>6.034875</td>
<td>30</td>
</tr>
<tr>
<td>GGAAG</td>
<td>12436970</td>
<td>2.0982418</td>
<td>5.2530236</td>
<td>5</td>
</tr>
<tr>
<td>CCACA</td>
<td>8676285</td>
<td>1.9496171</td>
<td>5.653704</td>
<td>29</td>
</tr>
<tr>
<td>CAACA</td>
<td>9133720</td>
<td>1.9081525</td>
<td>5.0892453</td>
<td>6</td>
</tr>
<tr>
<td>CTGGG</td>
<td>9150420</td>
<td>1.892757</td>
<td>6.7885427</td>
<td>1</td>
</tr>
<tr>
<td>CTTCT</td>
<td>6048770</td>
<td>1.8716344</td>
<td>5.5652685</td>
<td>51</td>
</tr>
<tr>
<td>GGCTG</td>
<td>9036445</td>
<td>1.8691818</td>
<td>6.2864585</td>
<td>1</td>
</tr>
<tr>
<td>TCTTC</td>
<td>5941715</td>
<td>1.8385087</td>
<td>5.49704</td>
<td>50</td>
</tr>
<tr>
<td>GGAGG</td>
<td>11038770</td>
<td>1.8206239</td>
<td>5.4335847</td>
<td>2</td>
</tr>
<tr>
<td>GCTGG</td>
<td>8487490</td>
<td>1.7556307</td>
<td>6.1824546</td>
<td>1</td>
</tr>
<tr>
<td>GGCAG</td>
<td>9353525</td>
<td>1.6973301</td>
<td>5.070092</td>
<td>2</td>
</tr>
<tr>
<td>ATCCA</td>
<td>7086295</td>
<td>1.6875144</td>
<td>6.8447223</td>
<td>37</td>
</tr>
<tr>
<td>GGCAT</td>
<td>7917265</td>
<td>1.6752146</td>
<td>6.189771</td>
<td>4</td>
</tr>
<tr>
<td>GGGAG</td>
<td>9643695</td>
<td>1.5905342</td>
<td>5.951234</td>
<td>2</td>
</tr>
<tr>
<td>CATGT</td>
<td>6164370</td>
<td>1.5208564</td>
<td>5.5801525</td>
<td>8</td>
</tr>
<tr>
<td>TCCAA</td>
<td>6307710</td>
<td>1.5021038</td>
<td>5.3048754</td>
<td>38</td>
</tr>
<tr>
<td>GGGCA</td>
<td>8225235</td>
<td>1.492586</td>
<td>5.3519793</td>
<td>1</td>
</tr>
<tr>
<td>AGCTC</td>
<td>6386555</td>
<td>1.4868033</td>
<td>5.5067234</td>
<td>16</td>
</tr>
<tr>
<td>CAGTT</td>
<td>6015045</td>
<td>1.4840155</td>
<td>5.5632033</td>
<td>32</td>
</tr>
<tr>
<td>ACCAC</td>
<td>6463250</td>
<td>1.4523339</td>
<td>5.8957133</td>
<td>28</td>
</tr>
<tr>
<td>ACAGT</td>
<td>6652085</td>
<td>1.4397745</td>
<td>5.1522145</td>
<td>31</td>
</tr>
<tr>
<td>GTGGG</td>
<td>7550665</td>
<td>1.4195397</td>
<td>8.890311</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>7368130</td>
<td>1.3852228</td>
<td>5.6886764</td>
<td>2</td>
</tr>
<tr>
<td>GAGCG</td>
<td>7569805</td>
<td>1.373649</td>
<td>5.3909035</td>
<td>54</td>
</tr>
<tr>
<td>GTGGT</td>
<td>6210780</td>
<td>1.3614845</td>
<td>5.152868</td>
<td>1</td>
</tr>
<tr>
<td>GCTCT</td>
<td>5124385</td>
<td>1.3598514</td>
<td>5.884657</td>
<td>17</td>
</tr>
<tr>
<td>TGGGG</td>
<td>7206780</td>
<td>1.3548887</td>
<td>6.1768794</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>8063475</td>
<td>1.3299086</td>
<td>6.908058</td>
<td>1</td>
</tr>
<tr>
<td>CTCTA</td>
<td>4724465</td>
<td>1.2824596</td>
<td>5.798107</td>
<td>18</td>
</tr>
<tr>
<td>GAATT</td>
<td>5517435</td>
<td>1.2655714</td>
<td>5.58982</td>
<td>23</td>
</tr>
<tr>
<td>GGGGT</td>
<td>6638920</td>
<td>1.24813</td>
<td>6.7300005</td>
<td>1</td>
</tr>
<tr>
<td>TACCA</td>
<td>5115405</td>
<td>1.2181711</td>
<td>5.62479</td>
<td>27</td>
</tr>
<tr>
<td>TTACC</td>
<td>4471695</td>
<td>1.213845</td>
<td>6.2385902</td>
<td>26</td>
</tr>
<tr>
<td>GGGGG</td>
<td>7503895</td>
<td>1.2098873</td>
<td>7.874745</td>
<td>1</td>
</tr>
<tr>
<td>AATTA</td>
<td>5149465</td>
<td>1.2082391</td>
<td>5.8074007</td>
<td>24</td>
</tr>
<tr>
<td>GTTAT</td>
<td>4539980</td>
<td>1.1870422</td>
<td>5.432468</td>
<td>34</td>
</tr>
<tr>
<td>TTATC</td>
<td>4006080</td>
<td>1.1524534</td>
<td>6.0615654</td>
<td>35</td>
</tr>
<tr>
<td>ATTAC</td>
<td>4410545</td>
<td>1.1130974</td>
<td>5.303973</td>
<td>25</td>
</tr>
<tr>
<td>TGTAT</td>
<td>4237100</td>
<td>1.10785</td>
<td>5.588327</td>
<td>10</td>
</tr>
<tr>
<td>TATCC</td>
<td>4035460</td>
<td>1.0954288</td>
<td>5.799492</td>
<td>36</td>
</tr>
<tr>
<td>TAGCT</td>
<td>4232030</td>
<td>1.044115</td>
<td>5.142871</td>
<td>15</td>
</tr>
<tr>
<td>ATTAG</td>
<td>4504090</td>
<td>1.0331335</td>
<td>5.009313</td>
<td>13</td>
</tr>
<tr>
<td>GTATT</td>
<td>3896015</td>
<td>1.0186684</td>
<td>5.435213</td>
<td>11</td>
</tr>
<tr>
<td>TCTAG</td>
<td>4004250</td>
<td>0.98791754</td>
<td>5.1807666</td>
<td>19</td>
</tr>
<tr>
<td>TATTA</td>
<td>3677725</td>
<td>0.98363256</td>
<td>5.5454226</td>
<td>12</td>
</tr>
<tr>
<td>TTAGC</td>
<td>3784620</td>
<td>0.9337311</td>
<td>5.3267245</td>
<td>14</td>
</tr>
<tr>
<td>GTCGG</td>
<td>4104290</td>
<td>0.8489691</td>
<td>5.213034</td>
<td>1</td>
</tr>
<tr>
<td>CCGTC</td>
<td>3382535</td>
<td>0.8469919</td>
<td>5.200364</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>4069525</td>
<td>0.72192717</td>
<td>5.7437162</td>
<td>1</td>
</tr>
<tr>
<td>CCCGT</td>
<td>2737925</td>
<td>0.6855806</td>
<td>5.224929</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
