<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317049_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317049_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>89965132</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>52</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>232007</td>
<td>0.257885466115917</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>199099</td>
<td>0.2213068503028484</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>191777</td>
<td>0.21316814163069311</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CACAG</td>
<td>12315295</td>
<td>2.1876874</td>
<td>5.5885677</td>
<td>30</td>
</tr>
<tr>
<td>CCACA</td>
<td>9840970</td>
<td>2.0031676</td>
<td>5.5465174</td>
<td>29</td>
</tr>
<tr>
<td>GGCTG</td>
<td>12453040</td>
<td>1.929996</td>
<td>5.446328</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>12116625</td>
<td>1.8778578</td>
<td>5.674245</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>11753125</td>
<td>1.8215216</td>
<td>5.2866187</td>
<td>1</td>
</tr>
<tr>
<td>ATCCA</td>
<td>7627615</td>
<td>1.6683397</td>
<td>6.067153</td>
<td>37</td>
</tr>
<tr>
<td>GGCAT</td>
<td>9363055</td>
<td>1.6104102</td>
<td>5.569892</td>
<td>4</td>
</tr>
<tr>
<td>AGCTC</td>
<td>8112415</td>
<td>1.5988512</td>
<td>5.0405583</td>
<td>16</td>
</tr>
<tr>
<td>CATGT</td>
<td>7369060</td>
<td>1.5605808</td>
<td>5.2067366</td>
<td>8</td>
</tr>
<tr>
<td>GGGAG</td>
<td>12385430</td>
<td>1.5098569</td>
<td>5.045112</td>
<td>2</td>
</tr>
<tr>
<td>ACCAC</td>
<td>7223270</td>
<td>1.4703246</td>
<td>5.294375</td>
<td>28</td>
</tr>
<tr>
<td>GCTCT</td>
<td>6702175</td>
<td>1.4655174</td>
<td>5.304785</td>
<td>17</td>
</tr>
<tr>
<td>GTGGG</td>
<td>9891330</td>
<td>1.3378173</td>
<td>6.968063</td>
<td>1</td>
</tr>
<tr>
<td>TACCA</td>
<td>6079205</td>
<td>1.3296659</td>
<td>5.3655753</td>
<td>27</td>
</tr>
<tr>
<td>CTCTA</td>
<td>5352630</td>
<td>1.2989141</td>
<td>5.408059</td>
<td>18</td>
</tr>
<tr>
<td>GGGGA</td>
<td>9899320</td>
<td>1.2067854</td>
<td>5.3688817</td>
<td>1</td>
</tr>
<tr>
<td>TTACC</td>
<td>4932665</td>
<td>1.1970017</td>
<td>5.52534</td>
<td>26</td>
</tr>
<tr>
<td>AATTA</td>
<td>5252395</td>
<td>1.1955521</td>
<td>5.4741564</td>
<td>24</td>
</tr>
<tr>
<td>ATTAC</td>
<td>4910580</td>
<td>1.1541039</td>
<td>5.1973667</td>
<td>25</td>
</tr>
<tr>
<td>TGTAT</td>
<td>4934815</td>
<td>1.1229528</td>
<td>5.133281</td>
<td>10</td>
</tr>
<tr>
<td>GGGGT</td>
<td>8261760</td>
<td>1.1174154</td>
<td>5.167365</td>
<td>2</td>
</tr>
<tr>
<td>TTATC</td>
<td>4230060</td>
<td>1.1030014</td>
<td>5.5480967</td>
<td>35</td>
</tr>
<tr>
<td>TATCC</td>
<td>4225425</td>
<td>1.0253769</td>
<td>5.1092854</td>
<td>36</td>
</tr>
<tr>
<td>TATTA</td>
<td>3715045</td>
<td>0.9381939</td>
<td>5.298395</td>
<td>12</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
