<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317057_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317057_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>69956083</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>61</td>
</tr>
<tr>
<td>%GC</td>
<td>54</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GCCCCTCTCCGGCCCCGGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCT</td>
<td>358392</td>
<td>0.5123099873959496</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCC</td>
<td>175890</td>
<td>0.25142917164187134</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTCACCCGGCCCGGACACGGACAGGATTGACAGATTGATAGCTCTTTCTCGATTCCGTGG</td>
<td>167008</td>
<td>0.23873263458733104</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCAGTTTTATCCGGTAAAGCGAATGATTAGAGGTCTTGGGGCCGAAACGATCTCAACCTA</td>
<td>105811</td>
<td>0.1512534656921829</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCCGTTCTTAGTTGGTGGAGCGATTTGTCTGGTTAATTCCGATAACGAACGAGACTCTGG</td>
<td>105619</td>
<td>0.15097900778692827</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTTTGACTGGGGCGGTACACCTGTCAAACGGTAACGCAGGTGTCCTAAGGCGAGCTCAGGG</td>
<td>92532</td>
<td>0.13227155671365992</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTCGGGCCGATCGCACG</td>
<td>92137</td>
<td>0.13170691675232876</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGT</td>
<td>83441</td>
<td>0.11927626079350383</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGTGTTAGGACCCGAAAGATGGTGAACTATGCCTGGGCAGGGCGAAGCCAGAGGAAACTCT</td>
<td>82953</td>
<td>0.11857868028431495</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGAAGGCCCGCGGCGGGTGTTGACGCGATGTGATTTCTGCCCAGTGCTCTGAATGTCAAAG</td>
<td>80341</td>
<td>0.11484490919824657</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCG</td>
<td>79023</td>
<td>0.11296087003613395</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCCTCTCCGGCCCCGGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTC</td>
<td>75399</td>
<td>0.10778047707445257</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTCGGGCCGATCGCACGC</td>
<td>72614</td>
<td>0.10379940798000367</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TGGTG</td>
<td>10663285</td>
<td>2.3756063</td>
<td>6.5940394</td>
<td>43</td>
</tr>
<tr>
<td>AACCT</td>
<td>6082780</td>
<td>1.8082955</td>
<td>8.467089</td>
<td>57</td>
</tr>
<tr>
<td>ATAAC</td>
<td>4438105</td>
<td>1.6138834</td>
<td>9.0808735</td>
<td>55</td>
</tr>
<tr>
<td>CTTTG</td>
<td>6137555</td>
<td>1.5941243</td>
<td>7.3906155</td>
<td>40</td>
</tr>
<tr>
<td>GGTGA</td>
<td>6686935</td>
<td>1.5875942</td>
<td>6.383145</td>
<td>44</td>
</tr>
<tr>
<td>TCAAA</td>
<td>4317620</td>
<td>1.5700697</td>
<td>7.181808</td>
<td>24</td>
</tr>
<tr>
<td>TTGGT</td>
<td>6004805</td>
<td>1.5475446</td>
<td>7.0632033</td>
<td>42</td>
</tr>
<tr>
<td>AAAGA</td>
<td>3983035</td>
<td>1.5315638</td>
<td>5.5380654</td>
<td>26</td>
</tr>
<tr>
<td>TGACT</td>
<td>5389075</td>
<td>1.4916605</td>
<td>7.017394</td>
<td>46</td>
</tr>
<tr>
<td>AAGAT</td>
<td>3980920</td>
<td>1.4364</td>
<td>5.303151</td>
<td>27</td>
</tr>
<tr>
<td>GATAA</td>
<td>3811285</td>
<td>1.3751922</td>
<td>8.650603</td>
<td>54</td>
</tr>
<tr>
<td>ACTCT</td>
<td>4919495</td>
<td>1.3723308</td>
<td>6.947682</td>
<td>48</td>
</tr>
<tr>
<td>TTTGG</td>
<td>5252275</td>
<td>1.3536043</td>
<td>6.9572883</td>
<td>41</td>
</tr>
<tr>
<td>GCTTT</td>
<td>5133845</td>
<td>1.3334279</td>
<td>6.6473417</td>
<td>39</td>
</tr>
<tr>
<td>GGCTT</td>
<td>5826560</td>
<td>1.3082123</td>
<td>5.9172025</td>
<td>38</td>
</tr>
<tr>
<td>TCTCC</td>
<td>5530625</td>
<td>1.2612618</td>
<td>5.813099</td>
<td>6</td>
</tr>
<tr>
<td>CCTCT</td>
<td>5501845</td>
<td>1.2546985</td>
<td>5.700464</td>
<td>4</td>
</tr>
<tr>
<td>AGATA</td>
<td>3422995</td>
<td>1.235089</td>
<td>8.4597435</td>
<td>53</td>
</tr>
<tr>
<td>GTGAC</td>
<td>5132700</td>
<td>1.2281201</td>
<td>5.8970585</td>
<td>45</td>
</tr>
<tr>
<td>GCCGG</td>
<td>7199345</td>
<td>1.2173624</td>
<td>5.035883</td>
<td>18</td>
</tr>
<tr>
<td>TAGAT</td>
<td>3505505</td>
<td>1.1868985</td>
<td>8.938684</td>
<td>52</td>
</tr>
<tr>
<td>ATTAA</td>
<td>2800630</td>
<td>1.1689838</td>
<td>5.4810405</td>
<td>30</td>
</tr>
<tr>
<td>CTCTC</td>
<td>5123795</td>
<td>1.168484</td>
<td>5.7366</td>
<td>5</td>
</tr>
<tr>
<td>GACTC</td>
<td>4453295</td>
<td>1.073888</td>
<td>5.9229617</td>
<td>47</td>
</tr>
<tr>
<td>CTAGA</td>
<td>3384070</td>
<td>0.99821496</td>
<td>7.3238173</td>
<td>51</td>
</tr>
<tr>
<td>CTCTA</td>
<td>3418555</td>
<td>0.953632</td>
<td>6.6113043</td>
<td>49</td>
</tr>
<tr>
<td>TCTAG</td>
<td>3411835</td>
<td>0.9443734</td>
<td>6.8726377</td>
<td>50</td>
</tr>
<tr>
<td>TAACC</td>
<td>2976485</td>
<td>0.8848527</td>
<td>7.24925</td>
<td>56</td>
</tr>
<tr>
<td>GGGGG</td>
<td>5152000</td>
<td>0.8577045</td>
<td>5.118938</td>
<td>21</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
