<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR307931_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307931_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>31785771</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGC</td>
<td>41131</td>
<td>0.12940066799071823</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTTGGTTTTGATCTGATAAATGCACGCATCCCCCCCGCGAAGGGGGTCAG</td>
<td>39596</td>
<td>0.12457146312417591</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGATGTGTCTGGAGTCTTGGAAGCTTGACTACCCTACGTTCTCCTACAAA</td>
<td>35033</td>
<td>0.11021598312024586</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGAGTCTTGGAAGCTTGACTACCCTACGTTCTCCTACAAATGGACCTT</td>
<td>31999</td>
<td>0.10067083161204428</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>5282030</td>
<td>2.479242</td>
<td>5.130826</td>
<td>17</td>
</tr>
<tr>
<td>GGAAG</td>
<td>4974700</td>
<td>2.2531037</td>
<td>6.6345425</td>
<td>5</td>
</tr>
<tr>
<td>GCAGG</td>
<td>4342080</td>
<td>2.2021334</td>
<td>5.490266</td>
<td>19</td>
</tr>
<tr>
<td>GGAGG</td>
<td>4232990</td>
<td>2.108828</td>
<td>5.165322</td>
<td>2</td>
</tr>
<tr>
<td>GGCTG</td>
<td>3917905</td>
<td>2.019522</td>
<td>5.4501204</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>3611155</td>
<td>1.9788811</td>
<td>9.338541</td>
<td>1</td>
</tr>
<tr>
<td>GAAGA</td>
<td>4734350</td>
<td>1.9493728</td>
<td>5.150629</td>
<td>6</td>
</tr>
<tr>
<td>CTGGG</td>
<td>3750380</td>
<td>1.9331698</td>
<td>6.1318727</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>3603295</td>
<td>1.795121</td>
<td>5.577344</td>
<td>2</td>
</tr>
<tr>
<td>GCTGG</td>
<td>3441770</td>
<td>1.7740936</td>
<td>6.5212326</td>
<td>1</td>
</tr>
<tr>
<td>GGCAG</td>
<td>3456700</td>
<td>1.7531033</td>
<td>5.266007</td>
<td>2</td>
</tr>
<tr>
<td>AGAGC</td>
<td>3721220</td>
<td>1.71574</td>
<td>5.0709004</td>
<td>8</td>
</tr>
<tr>
<td>CTGGA</td>
<td>3589095</td>
<td>1.6818995</td>
<td>5.0477457</td>
<td>3</td>
</tr>
<tr>
<td>GTGGG</td>
<td>3023640</td>
<td>1.5309918</td>
<td>7.8275685</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>3069155</td>
<td>1.5290185</td>
<td>6.8560944</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>2943180</td>
<td>1.4902517</td>
<td>5.813535</td>
<td>2</td>
</tr>
<tr>
<td>GGGCA</td>
<td>2925965</td>
<td>1.4839352</td>
<td>5.373138</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>2759995</td>
<td>1.3974977</td>
<td>6.124474</td>
<td>1</td>
</tr>
<tr>
<td>GGGGT</td>
<td>2722765</td>
<td>1.3786465</td>
<td>7.099504</td>
<td>3</td>
</tr>
<tr>
<td>CCCCC</td>
<td>2239800</td>
<td>1.3419688</td>
<td>5.0938973</td>
<td>31</td>
</tr>
<tr>
<td>GGTTT</td>
<td>2800510</td>
<td>1.2106494</td>
<td>5.7081404</td>
<td>4</td>
</tr>
<tr>
<td>GTTGG</td>
<td>2423465</td>
<td>1.1338321</td>
<td>5.7426167</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>1810700</td>
<td>1.0101178</td>
<td>6.798602</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
