<h1>Baseline Expression Atlas help</h1>


<!-- Beginning of main text section -->
<section class="grid_18 alpha">


<h2><a name="about">About the Baseline Atlas</a></h2>
	
<p>The Baseline Atlas represents information about which gene
products are present (and at what abundance) in which "normal" condition (e.g.,
tissue, cell type). The Baseline Atlas consists of
highly curated RNA&ndash;seq experiments from <a
	href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a>, such as: </p>

<ul>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-1733">RNA-seq of coding RNA from tissue samples of 95 human individuals representing 27 different tissues in order to determine tissue-specificity of all protein-coding genes</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-513">Illumina Body Map</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-599">Transcriptome of DBAxC57BL/6J mice</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284">RNA&ndash;seq of long coding and long non coding RNAs from ENCODE cell lines</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-30352">RNA&ndash;seq of 6 tissues from 10 species to investigate the evolution of gene expression levels in mammalian organs</a></li>
  <li><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2037/">Transcription profiling by RNA-seq of seven tissues from Oryza sativa japonica Nipponbare</a></li>
</ul>

<p>The above experiments have been re&ndash;analyzed using an in&ndash;house RNA&ndash;seq
processing pipeline. The resulting data matrix stores gene
expression profiles across the analyzed experimental factor values as a set of
normalized expression levels (<a
	class="externallink"
	title="FPKMs"
	href="http://www.nature.com/nbt/journal/v28/n5/full/nbt.1621.html">FPKMs</a>).
You can search the data by a combination of genes attributes, experimental
factor values, and a minimum FPKM cutoff. See below for more details on searching.</p>




<h2><a name="at-a-glance">Baseline Atlas at&ndash;a&ndash;glance</a></h2>

<p><a href="resources/images/help/baseline-atlas-overview.pdf"><img alt="Baseline Atlas Experiment View 'At a Glance'" src="resources/images/help/exptPageHelp.png" align="bottom" /></a></p>




<h2><a name="filtering">Filtering Baseline Atlas data</a></h2>

<p><img alt="filtering" src="resources/images/help/filtering.png" align="bottom" /></p>
<p>Some experiments have <b>more than one</b> experimental variable (factor). For example, in experiment
<a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-26284">E-GEOD-26284</a>,
three different <b>types of RNA</b> were extracted from <b>six cellular components</b> from <b>23
human cell lines</b>. Results for these experiments are limited to a certain
combination of variables, e.g. showing expression of <b>total RNA</b> in the
<b>whole cell</b>. To limit the search results differently, use the "Change
filters" menu. In a 3&ndash;factor experiment, you can choose one value from any two
of the factors, and then search among the values for the remaining factor.
E.g., if you choose to limit to <b>poly(A) RNA</b> extracted from the
<b>cytosol</b>, you can then search among the <b>cell lines</b> for which this
data is available.</p>




<h2><a name="searching">Searching Baseline Atlas</a></h2>


<h3><a name="minExprLevel">Specifying the minimum expression level</a></h3>

<p>You can specify a minimum expression level to act as a <b>cutoff</b>, and only genes expressed <b>above</b> this level are displayed. This is to aid distinction between "background noise" and "real" gene expression. Our default FPKM cutoff is 0.5. To change it, enter a new value into the "Expression level cutoff" input box, or drag the slider until the "Expression level cutoff" input box shows your desired cutoff value. A histogram showing the number of genes expressed above a range of cutoffs is also available. This lets you see how many genes you are excluding by selecting a given cutoff. If you see that you're excluding a lot of genes, you may wish to <b>decrease</b> the cutoff. Clicking on the <img alt="Histogram enable/disable button" src="resources/images/help/histogram_btn.png" align="bottom" /> button toggles display of the histogram.</p>

<ul>
	<li>If you have not selected any experimental factors, the histogram shows the number of genes expressed in <b>any</b> factor value for each of the shown cutoffs.</li>
	<li>If you have selected <b>at least one</b> factor value, the histogram shows the number of genes expressed in <b>at least one of the selected factor values</b>, for each of the shown cutoffs.</li>
</ul>


<h3><a name="geneSearch">Searching with genes</a></h3>

<p>You can search with <a class="externallink" href="http://ensembl.org">Ensembl</a> gene symbols (e.g. SAA4), identifiers (e.g. ENSG00000148965) or <a href="http://www.ensembl.org/Help/Glossary">biotypes</a> (e.g. protein_coding), <a href="http://www.uniprot.org">UniProt</a> accessions (e.g. P35542), <a href="http://www.geneontology.org">GO</a> ("phosphoglycerate kinase activity") or <a href="http://www.ebi.ac.uk/interpro/">InterPro</a> terms (e.g. phosphatase). A <b>space&ndash;separated</b> list of gene attributes will bring back genes that match <b>at least one</b> of the attributes in the query. Put multi-word query terms in quotes e.g. "transcription factor binding".</p>

<p>If the "Exact match" box is checked, only genes with annotations that fully match your query will be returned. For example, to find the expression of the <a href="http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000198668">calmodulin 1</a> gene, check the "Exact match" box, type "CALM1" and click "Search". If the box is not checked, genes with annotations that contain words that fully match your query will be returned. For example, to find expressions for genes with "calmodulin" in their annotations (e.g. "calmodulin binding", "calmodulin regulated", &hellip;), uncheck the "Exact match" box, type "calmodulin" and click search.</p>


<h4>Gene sets</h4>

<p>If you search for a term that matches multiple genes, e.g. a <a href="http://www.reactome.org/">Reactome</a> pathway ID such as "REACT_1258", the heatmap will show one row per gene matching your term:</p>

<p><img alt="Gene set query before" src="resources/images/help/geneSet_before.png" align="bottom" /></p>

<p>You can summarize the results for each "gene set" term by clicking "show by gene set":</p>

<p><img alt="Gene set query after" src="resources/images/help/geneSet_after.png" align="bottom" /></p>

<p>Click "show individual genes" to return to the original view.</p>


<h3><a name="efvSearch">Searching with experimental factor values</a></h3>

<p>Use the factor value input box to search for experimental factor (e.g. <i>organism part</i> or <i>cell line</i>) values, or leave it blank to search in all. You can search:</p>
	
<ul>
	<li>within a <b>single&ndash;factor</b> experiment, e.g. <a href="experiments/E-MTAB-513" title="E-MTAB-513">Illumina Body Map</a><a title="E-MTAB-513" href="experiments/E-MTAB-513">,</a> or</li>
	<li>within a pre&ndash;selected subset of a <b>multi&ndash;factorial</b> experiment, e.g. <a href="experiments/E-GEOD-26284" rel="nofollow" title="E-GEOD-26284">RNA&ndash;seq of long coding and long non coding RNAs from ENCODE cell lines</a></li>
</ul>

<p>When you click on the experiment factor value input box, a dropdown menu appears, containing the list of all factor values available for selection:</p>
<p><img alt="Factor values dropdown" src="resources/images/help/factor_values_dropdown.png" align="bottom" /></p>
<p>Select one or more factor values to search with, or start typing to see suggestions. To see all factor values in an experiment, click the <img alt="Experiment Design" src="resources/images/help/expDes_btn.png" align="bottom" /> button.</p>




<h4><i>Specific</i> vs. <i>non&ndash;specific</i> search</h4>
	
<p>The default Baseline Atlas search reports genes with more "specific" expression (with the <i>Specific</i> option selected by default). What we mean by this is outlined below.</p>
	
<h5>If the query contains no factor values&hellip;</h5>	

<p>If the query contains <b>no experimental factor values</b>, the <i>Specific</i> search will report genes expressed in just <b>one</b> experimental factor value, then genes expressed in just <b>two</b> factor values, then three and so on, reporting any genes expressed in all the available factor values at the end of the list of results. Within each group of genes expressed in N factor values only, genes are sorted by the highest average FPKM across the N factor values in which they are expressed.</p>

<h5>If the query contains at least one factor value&hellip;</h5>
<p>If the query contains <b>at least one</b> experimental factor value, the <i>Specific</i> search will promote genes with <b>higher average FPKM</b> in the <b>queried</b> factor value(s), and at the same time penalize genes with higher FPKMs in the <b>non&ndash;queried</b> factor values. This is done using the "fold difference" between the query and non-query FPKMs:</p>

<ul>
	<li>Fold difference is calculated by dividing the <b>average FPKM</b> of the queried factor values by the <b>highest FPKM</b> of the non-query factor values.</li>
	<li>If <b>none</b> of the non-query FPKMs are above the selected expression level cutoff, the average of the query FPKMs is divided by the cutoff instead.</li>
	<li>If <b>none</b> of the non-query FPKMs are above the cutoff and the cutoff is <b>zero</b>, 0.09 is used as the denominator instead of the cutoff.</li>
</ul>

	<p>Genes with the <b>greatest fold difference</b> in expression between query and non&ndash;query factor values will be pushed towards the top of the list. Genes that have higher expression in non-query factor values than in the query factor values are not shown.</p>

	<h5><i>Non&ndash;specific</i> search</h5>
  <p>If the <i>Specific</i> option is not selected, the Baseline Atlas search will <b>ignore</b> specificity of expression when ordering the results. In particular:</p>
 
  <ul>
	  <li>If the query contains <b>no experimental factor values</b>, genes with the highest expression across <b>all</b> factor values will be reported first.</li>
	  <li>If the query contains <b>at least one</b> experimental factor value, genes with higher FPKM across the <b>selected</b> factor values are reported first. Expression levels across the non&ndash;selected factor values are <b>ignored</b> in the ordering of results.</li>
  </ul>




<h2><a name="heatmap">Heatmap results display</a></h2>

<p>Results for your query are shown in a heatmap table. The heatmap rows and columns are labelled with gene symbols and experimental factor values, respectively. The heatmap shows expressions by <b>colour intensity</b>, according to the gradient bar,<img alt="heatmap colour gradient" src="resources/images/help/gradient.png" align="bottom" />, displayed above the heatmap. Greater colour intensity means higher relative gene expression. To see the exact FPKM for all genes, click on the <img alt="Display levels" src="resources/images/help/display_levels.png" align="bottom" /> button. This also reveals minimum and maximum FPKMs on each side of the gradient bar. The gradient shows intensities corresponding to expression levels for the <b>top 50</b> genes currently displayed (rather than all FPKM values for all genes returned by the query).</p>

<p>The Baseline Atlas also includes information about transcript-level expression. To see this for a given gene, click the corresponding filled box in the heatmap. This brings up a pie chart detailing the proportion of expression contributed by each transcript. A maximum of three top contributing transcripts are shown, and the remaining ones are aggregated as "Others". Transcripts that are not expressed above the selected minimum expression level are not shown. White portions of the pie chart indicate that the data for that transcript is of low confidence. Please refer to the <a href="http://cufflinks.cbcb.umd.edu/manual.html#fpkm_tracking_format">Cufflinks documentation</a> for more details.</p>

<p>When displayed factor values are organism parts, mousing over each heatmap column highlights the corresponding tissue in the anatomogram on the left (if available). If the selected organism part is gender&ndash;specifc, and anatomogram is available for each gender, you can switch between them by clicking the <img alt="Gender selection button" src="resources/images/help/gender_sel.png" align="bottom" /> button.</p>

<p>If you mouse-over a gene name in the heatmap, synonyms, <a href="http://www.geneontology.org/">Gene Ontology</a> terms, and <a href="http://www.ebi.ac.uk/interpro/">Interpro</a> terms are displayed:</p>

<img alt="gene mouseover popup" src="resources/images/help/geneMouseover.png" align="bottom" />

<p>Any terms that match your search query term(s) are highlighted in yellow here. Click the gene name to see more information about that gene, including other Expression Atlas experiments in which it was found.</p>

<h3><a name="ensembl">Visualising baseline expression at Ensembl</a></h3>

<p>If you select a gene name and a condition heading from the heatmap, the Ensembl Genome Browser <strong>Open</strong> button, to the left of the heatmap, will become clickable.</p>

<p><img src="resources/images/help/ensembl_gene_baseline_select.png" alt="screenshots showing how to visualise data in Ensembl" align="bottom" /></p>

<p>Clicking the <strong>Open</strong> button will take you to the <a
href="http://www.ensembl.org/index.html">Ensembl</a> or <a
href="http://ensemblgenomes.org/">Ensembl Genomes</a> browser, which will
display the FPKM for the condition (e.g. tissue) that you selected in the
context of the genomic location of the gene you selected. For more details
about how to use the genome browsers, please see the relevant documentation for
<a href="http://www.ensembl.org/info/index.html">Ensembl</a> or <a
	href="http://ensemblgenomes.org/info/faqs">Ensembl Genomes</a>.</p>

<h2><a name="anatomogram">Anatomogram</a></h2>

<p>The anatomogram provides a visualisation of the tissues studied in a given experiment. Mouseover a tissue in the anatomogram to highlight the corresponding column in the heatmap table.</p>




<h2><a name="download">Downloading query results</a></h2>

<p>The <b>top 50</b> genes resulting from your search are displayed on the page. Click on the <img alt="Download button" src="resources/images/help/download_button.png" align="bottom" /> button to download the <b>full results</b> of your query in tab&ndash;delimited format with no ordering.</p>




<h2><a name="exptDesign">Experiment design page</a></h2>

<p>The Experiment Design page (e.g. <a title="Illumina Body Map &mdash;
	Experiment Design page"
	href="experiments/E-MTAB-513/experiment-design">Illumina Body Map</a>)
shows RNA&ndash;seq processing run acessions (from <a
	href="http://www.ebi.ac.uk/ena/">ENA</a>), along with their corresponding
biological sample characteristcs and factor values. You can access it from the
experiment page by clicking the <img alt="Experiment Design button"
src="resources/images/help/expDes_btn.png" align="bottom" /> button. Only the
runs used in the Baseline Atlas are shown, although the dataset these were
taken from may contain more. To see all runs in the dataset, uncheck <i>Show
Analysed only</i>.</p>

<p>You can sort the experiment design table by clicking on a column. Click the
column again to reverse the sort order. To then sort by another column while
retaining the ordering of the first column, hold down the <b>shift</b> key and
click another column. The first selected column forms the primary sort order
and the shift+click&ndash;ed columns form secondary, tertiary, etc. sort orders
in turn. If you shift+click a column to sort by it and then wish to undo this,
just shift+click it again until the column is unsorted (<img alt="unsorted
arrows" src="resources/images/help/unsortedArrows.png" align="center" />).</p>
  
<p>The experiment design table is also searchable &mdash; entering a keyword
automatically selects the subsection of the table that matches the keyword.
Click on the <img alt="Download button"
src="resources/images/help/download_button.png" align="bottom" /> button on the
right&ndash;hand side to download the full results of the current query (with
no ordering applied) in tab&ndash;delimited format.</p>




<h2><a name="anaMethods">Analysis methods page</a></h2>

<p>The Analysis Methods page (e.g. <a title="Illumina Body Map &mdash; Analysis
	Methods page" href="experiments/E-MTAB-513/analysis-methods">Illumina Body
	Map</a>) lists the data analysis methods we applied to the raw experimental
data in <a title="FASTQ format" class="externallink"
	href="http://nar.oxfordjournals.org/content/38/6/1767.full">FASTQ</a>
format, to obtain the gene expression levels shown in the Baseline Atlas. You can
access it by clicking the <img alt="Analysis Methods
button" src="resources/images/help/am_btn.png" align="bottom" /> button.</p>




<h2><a name="feedback">Providing feedback</a></h2>

<p>We value your feedback on what Expression Atlas is doing right, what does not work or what could be achieved more intuitively. We would also be grateful for any feedback on the analysis methods we have adopted &mdash; we are passionate about not only about intuitive data presentation, the highest level of experimental metadata curation, but also about the quality and biological validity of the expression data presented in Baseline Atlas.</p>

<p>To send us your feedback, please fill in form accessed by clicking on the <i>Feedback</i> link on the top right of the page:</p>
<p><img alt="Feedback Pop-up" src="resources/images/help/feedback.png" align="bottom" /></p>

</section>
<!-- End of main text section -->

<!-- This part is for the navigation section at the right-hand side -->
<aside class="grid_6 omega">

<h3>Jump to...</h3>

<ul>
	<li><a href="help/baseline-atlas.html#about">About the Baseline Atlas</a></li>
	<li><a href="help/baseline-atlas.html#at-a-glance">Baseline Atlas at&ndash;a&ndash;glance</a></li>
	<li><a href="help/baseline-atlas.html#filtering">Filtering Baseline Atlas data</a></li>
	<li><a href="help/baseline-atlas.html#searching">Searching Baseline Atlas</a>
		<ul>
			<li><a href="help/baseline-atlas.html#minExprLevel">Specifying the minimum expression level</a></li>
			<li><a href="help/baseline-atlas.html#geneSearch">Searching with genes</a></li>
			<li><a href="help/baseline-atlas.html#efvSearch">Searching with experimental factor values</a></li>
		</ul>
	</li>
	<li><a href="help/baseline-atlas.html#heatmap">Heatmap results display</a>
		<ul>
			<li><a href="help/baseline-atlas.html#ensembl">Visualising baseline expression at Ensembl</a></li>
		</ul>
	</li>
	<li><a href="help/baseline-atlas.html#anatomogram">Anatomogram</a></li>
	<li><a href="help/baseline-atlas.html#download">Downloading query results</a></li>
	<li><a href="help/baseline-atlas.html#exptDesign">Experiment design page</a></li>
	<li><a href="help/baseline-atlas.html#anaMethods">Analysis methods page</a></li>
	<li><a href="help/baseline-atlas.html#feedback">Providing feedback</a></li>
</ul>
</aside>
