<h1>Expression Atlas data in R</h1>

<p>For <a href="help/differential-atlas.html">differential expression</a> experiments in Expression Atlas,
you can download a file containing the expression data and meta-data that you
can load into <a href="http://www.r-project.org/">R</a>. Click the button on the top-right of any
differential experiment page to download one.</p>

<!-- SCREENSHOT OF DOWNLOAD -->

<h2>How to load an Atlas experiment summary in R</h2>

<p>Start an R session on your computer.</p>

<!-- screenshot of new R session -->

<p>Load the object you downloaded into your R session, e.g.:</p>

<p><code>load( "/path/to/E-GEOD-38400-atlasExperimentSummary.Rdata" )</code></p>

<p>This has created an object called <code>experimentSummary</code>. This
object is a <a href="">SimpleList</a> object. Each element is one of three <a
href="">Bioconductor</a> objects: 
	<ul>
		<li><a href="">SummarizedExperiment</a> (RNA-seq experiments)</li>
		<li><a href="">ExpressionSet</a> (1-colour microarray experiments)</li>
		<li><a href="">MAList</a> (2-colour microarray experiments)</li>
	</ul>
</p>

<h2>How to use it</h2>

<h3>What does the object contain?</h3>

<p>You can see the names of the elements in the list, and thus how many
elements there are, by typing <code>names( experimentSummary )</code>.</p>

<h3>RNA-seq data</h3>

<p>Data from an RNA-seq experiment is contained in a single <a
	href="">SummarizedExperiment</a> object in the SimpleList you have loaded.
In order to use it properly you will need to install and load the <a
	href="">GenomicRanges</a> package from <a href="">Bioconductor</a>. Do this
by running the following three commands:</p>

<p><code>source( &quot;http://bioconductor.org/biocLite.R&quot; )</code></p>
<p><code>biocLite( &quot;GenomicRanges&quot; )</code></p>
<p><code>library( GenomicRanges )</code></p>

<p>For more details about using using this package please see the <a href="">documentation</a>.

<p>The SummarizedExperiment object is stored under the name "rnaseq", so you
can assign it to a new variable like this:</p>

<p><code>summarizedExperiment &lt;- experimentSummary$rnaseq</code></p>

<p>The SummarizedExperiment object contains the following:
	<ul>
		<li>Matrix of raw counts (not normalized), in the <strong>assays</strong> slot.</li>
		<li>Sample annotations, in the <strong>colData</strong> slot.</li>
		<li>Brief outline of methods, from QC of FASTQ files to production of
		raw counts, in the <strong>exptData</strong> slot.</li>
	</ul>
</p>

<!-- screenshot(s) of these slots? -->

<p>For more information on how to use a SummarizedExperiment object, please see
the <a href="">documentation</a> from <a href="">Bioconductor</a>.</p>

<h3>One-colour microarray data</h3>

<p>Data from a one-colour, or single-channel, microarray experiment is stored
in potentially multiple <a href="">ExpressionSet</a> objects in the SimpleList
summary you have loaded. There is one ExpressionSet per array design used in
the experiment. The ExpressionSets are indexed by the <a
	href="">ArrayExpress</a> accession of the array design used.</p>

<!-- screenshot of multi-array names -->

<p>You can access each ExpressionSet via its array design accession, by typing
e.g. <code>expressionSet &lt;- experimentSummary[[ "A-AFFY-18" ]]</code>.</p>

<!-- screenshot of assignment -->

<p>Each ExpressionSet object contains the following:
	<ul>
		<li>Matrix of normalized intensity values, in the
		<strong>assayData</strong>, accessed via <code>exprs( expressionSet
			)</code>.</li>
		<li>Sample annotations, in the <strong>phenoData</strong>, accessed via
		<code>pData( expressionSet )</code>.</li>
		<li>Brief outline of normalization method applied, in the
		<strong>experimentData</strong> slot, accessed via <code>preproc(
			experimentData( expressionSet ) )</code>.</li>
	</ul>
</p>

<!-- screenshot(s) of above? -->

<p>For more information on how to use an ExpressionSet object, please see the
<a href="">documentation</a> from <a href="">Bioconductor</a>.

<h3>Two-colour microarray data</h3>

<p> Data from a two-colour microarray experiment is stored in one or more <a
href="">MAList</a> objects. As for ExpressionSet objects, there is one MAList
per array design used in the experiment. This means you can access each MAList
object via its <a herf="">ArrayExpress</a> array design accession.</p>

<!-- screenshot of assignment -->

<p>Each MAList object contains the following:
	<ul>
		<li>Vector of probe names, in the <strong>genes</strong> element.</li>
		<li>Matrix of log<sub>2</sub>(fold-changes), in the <strong>M</strong> element.</li>
		<li>Matrix of LOESS normalized average intensities, in the <strong>A</strong> element.</li>
	</ul>
</p>

<!-- screenshot(s) of above? -->

<p>For more information on how to use an MAList object, please see the <a
href="">documentation</a> from <a href="">Bioconductor</a>.</p>
