<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR534303_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR534303_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>149659336</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>101</td>
</tr>
<tr>
<td>%GC</td>
<td>55</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTTTT</td>
<td>37613485</td>
<td>4.736946</td>
<td>11.1081</td>
<td>1</td>
</tr>
<tr>
<td>AAAAA</td>
<td>26063465</td>
<td>3.5432775</td>
<td>4.518759</td>
<td>75-79</td>
</tr>
<tr>
<td>GCAGG</td>
<td>47733820</td>
<td>2.4149034</td>
<td>5.0978184</td>
<td>2</td>
</tr>
<tr>
<td>CTGGG</td>
<td>47577480</td>
<td>2.3704534</td>
<td>6.2376285</td>
<td>1</td>
</tr>
<tr>
<td>GGCAG</td>
<td>46103940</td>
<td>2.332446</td>
<td>6.699342</td>
<td>1</td>
</tr>
<tr>
<td>GGCTG</td>
<td>43991145</td>
<td>2.191771</td>
<td>5.5386972</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>43918140</td>
<td>2.188134</td>
<td>5.269927</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>37907470</td>
<td>1.8542883</td>
<td>5.5986943</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>38141035</td>
<td>1.8373898</td>
<td>6.3320427</td>
<td>2</td>
</tr>
<tr>
<td>GGGGG</td>
<td>42048075</td>
<td>1.5930096</td>
<td>5.414129</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>31510145</td>
<td>1.5413555</td>
<td>5.774756</td>
<td>1</td>
</tr>
<tr>
<td>GTGGG</td>
<td>29127490</td>
<td>1.4031752</td>
<td>7.402326</td>
<td>1</td>
</tr>
<tr>
<td>GGGGT</td>
<td>26705430</td>
<td>1.2864958</td>
<td>5.472075</td>
<td>1</td>
</tr>
<tr>
<td>GCGGG</td>
<td>19519830</td>
<td>0.76483804</td>
<td>5.0659695</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
