<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR315328_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315328_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>38939373</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTA</td>
<td>283782</td>
<td>0.7287790689387834</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGT</td>
<td>282877</td>
<td>0.7264549431753818</td>
<td>No Hit</td>
</tr>
<tr>
<td>TGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTAC</td>
<td>60676</td>
<td>0.15582171803331296</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGCC</td>
<td>5092880</td>
<td>2.2641487</td>
<td>10.123424</td>
<td>2</td>
</tr>
<tr>
<td>CCTGC</td>
<td>5018470</td>
<td>2.2310684</td>
<td>10.09355</td>
<td>1</td>
</tr>
<tr>
<td>TGTCT</td>
<td>4459000</td>
<td>2.0857422</td>
<td>10.068817</td>
<td>19</td>
</tr>
<tr>
<td>CCATG</td>
<td>4573680</td>
<td>1.996492</td>
<td>9.544677</td>
<td>36</td>
</tr>
<tr>
<td>TGCCA</td>
<td>4562345</td>
<td>1.9915441</td>
<td>9.311445</td>
<td>3</td>
</tr>
<tr>
<td>AGCCA</td>
<td>4680995</td>
<td>1.9559578</td>
<td>9.435866</td>
<td>34</td>
</tr>
<tr>
<td>AAGAT</td>
<td>4474385</td>
<td>1.8830237</td>
<td>9.445914</td>
<td>26</td>
</tr>
<tr>
<td>GCCAG</td>
<td>4206405</td>
<td>1.8110046</td>
<td>9.22516</td>
<td>3</td>
</tr>
<tr>
<td>GCCAT</td>
<td>4104455</td>
<td>1.791667</td>
<td>9.342744</td>
<td>35</td>
</tr>
<tr>
<td>AAAGA</td>
<td>4322220</td>
<td>1.7412004</td>
<td>8.846557</td>
<td>25</td>
</tr>
<tr>
<td>CAAAG</td>
<td>4199205</td>
<td>1.7228547</td>
<td>8.997343</td>
<td>25</td>
</tr>
<tr>
<td>CCAGT</td>
<td>3898155</td>
<td>1.7016134</td>
<td>9.141727</td>
<td>4</td>
</tr>
<tr>
<td>TGCTT</td>
<td>3584545</td>
<td>1.6767073</td>
<td>9.910668</td>
<td>16</td>
</tr>
<tr>
<td>ATGCT</td>
<td>3694670</td>
<td>1.6543155</td>
<td>9.399814</td>
<td>14</td>
</tr>
<tr>
<td>TCTCA</td>
<td>3680515</td>
<td>1.6289347</td>
<td>9.419458</td>
<td>21</td>
</tr>
<tr>
<td>AGATT</td>
<td>3531245</td>
<td>1.5524973</td>
<td>9.50598</td>
<td>27</td>
</tr>
<tr>
<td>ATGCA</td>
<td>3576215</td>
<td>1.5328008</td>
<td>8.957731</td>
<td>38</td>
</tr>
<tr>
<td>CATGC</td>
<td>3507380</td>
<td>1.5310333</td>
<td>9.080309</td>
<td>36</td>
</tr>
<tr>
<td>CTCAA</td>
<td>3586340</td>
<td>1.5193787</td>
<td>8.8788805</td>
<td>23</td>
</tr>
<tr>
<td>AAGCC</td>
<td>3598580</td>
<td>1.5036696</td>
<td>8.730116</td>
<td>33</td>
</tr>
<tr>
<td>AGCAT</td>
<td>3493595</td>
<td>1.4973893</td>
<td>9.305292</td>
<td>10</td>
</tr>
<tr>
<td>GCATG</td>
<td>3377670</td>
<td>1.4916489</td>
<td>9.111341</td>
<td>39</td>
</tr>
<tr>
<td>CATGT</td>
<td>3202800</td>
<td>1.434077</td>
<td>9.227203</td>
<td>41</td>
</tr>
<tr>
<td>TCAAA</td>
<td>3400655</td>
<td>1.4146125</td>
<td>8.644404</td>
<td>24</td>
</tr>
<tr>
<td>ATGTC</td>
<td>3158755</td>
<td>1.4143555</td>
<td>9.290203</td>
<td>41</td>
</tr>
<tr>
<td>CTTGT</td>
<td>2930225</td>
<td>1.3706425</td>
<td>9.573882</td>
<td>17</td>
</tr>
<tr>
<td>TGCAT</td>
<td>3059595</td>
<td>1.369956</td>
<td>9.132906</td>
<td>38</td>
</tr>
<tr>
<td>GTCTC</td>
<td>2983285</td>
<td>1.3604327</td>
<td>9.275961</td>
<td>21</td>
</tr>
<tr>
<td>GCTTG</td>
<td>2776320</td>
<td>1.2808535</td>
<td>9.414434</td>
<td>16</td>
</tr>
<tr>
<td>TTGTC</td>
<td>2671080</td>
<td>1.2494248</td>
<td>9.369844</td>
<td>18</td>
</tr>
<tr>
<td>AAGTA</td>
<td>2897575</td>
<td>1.2194307</td>
<td>8.586001</td>
<td>48</td>
</tr>
<tr>
<td>AGTAC</td>
<td>2782580</td>
<td>1.1926411</td>
<td>8.65839</td>
<td>48</td>
</tr>
<tr>
<td>CAGTA</td>
<td>2782420</td>
<td>1.1925727</td>
<td>8.687965</td>
<td>6</td>
</tr>
<tr>
<td>GGCCG</td>
<td>2666545</td>
<td>1.1828909</td>
<td>8.826219</td>
<td>58</td>
</tr>
<tr>
<td>ATTAA</td>
<td>2726370</td>
<td>1.1633235</td>
<td>8.704798</td>
<td>29</td>
</tr>
<tr>
<td>ATATG</td>
<td>2586840</td>
<td>1.1372935</td>
<td>8.88898</td>
<td>13</td>
</tr>
<tr>
<td>CATAT</td>
<td>2388785</td>
<td>1.038084</td>
<td>9.053656</td>
<td>12</td>
</tr>
<tr>
<td>GCCGG</td>
<td>2325905</td>
<td>1.0317816</td>
<td>8.726541</td>
<td>58</td>
</tr>
<tr>
<td>GATTA</td>
<td>2308255</td>
<td>1.0148147</td>
<td>8.974976</td>
<td>28</td>
</tr>
<tr>
<td>TTAAG</td>
<td>2304810</td>
<td>1.0133002</td>
<td>8.637195</td>
<td>30</td>
</tr>
<tr>
<td>CGGCC</td>
<td>2308985</td>
<td>1.0124401</td>
<td>8.83157</td>
<td>57</td>
</tr>
<tr>
<td>TAGCA</td>
<td>2292015</td>
<td>0.9823801</td>
<td>8.647975</td>
<td>9</td>
</tr>
<tr>
<td>TATGC</td>
<td>2161685</td>
<td>0.96791023</td>
<td>8.782043</td>
<td>14</td>
</tr>
<tr>
<td>CTAAG</td>
<td>2243510</td>
<td>0.9615904</td>
<td>8.477602</td>
<td>45</td>
</tr>
<tr>
<td>TCTAA</td>
<td>2189440</td>
<td>0.95145553</td>
<td>8.510535</td>
<td>45</td>
</tr>
<tr>
<td>GTCTA</td>
<td>1981315</td>
<td>0.88714826</td>
<td>8.71394</td>
<td>44</td>
</tr>
<tr>
<td>CACGG</td>
<td>2022235</td>
<td>0.870643</td>
<td>8.323233</td>
<td>55</td>
</tr>
<tr>
<td>AGTAG</td>
<td>1984825</td>
<td>0.8606606</td>
<td>8.510624</td>
<td>7</td>
</tr>
<tr>
<td>TAAGC</td>
<td>1967440</td>
<td>0.8432641</td>
<td>8.318159</td>
<td>31</td>
</tr>
<tr>
<td>GCATA</td>
<td>1966815</td>
<td>0.8429963</td>
<td>8.365607</td>
<td>11</td>
</tr>
<tr>
<td>TAAGT</td>
<td>1903790</td>
<td>0.8369934</td>
<td>8.561884</td>
<td>47</td>
</tr>
<tr>
<td>CGCAC</td>
<td>1964890</td>
<td>0.8361789</td>
<td>8.318085</td>
<td>53</td>
</tr>
<tr>
<td>GTAGC</td>
<td>1870950</td>
<td>0.8262501</td>
<td>8.708449</td>
<td>8</td>
</tr>
<tr>
<td>ACGGC</td>
<td>1887990</td>
<td>0.81284595</td>
<td>8.296626</td>
<td>55</td>
</tr>
<tr>
<td>GCACG</td>
<td>1748380</td>
<td>0.75273895</td>
<td>8.17767</td>
<td>53</td>
</tr>
<tr>
<td>ACGCA</td>
<td>1655490</td>
<td>0.6917479</td>
<td>7.996663</td>
<td>52</td>
</tr>
<tr>
<td>CCGGT</td>
<td>1492375</td>
<td>0.67122334</td>
<td>8.446519</td>
<td>59</td>
</tr>
<tr>
<td>GTACG</td>
<td>1407580</td>
<td>0.62161636</td>
<td>8.267404</td>
<td>49</td>
</tr>
<tr>
<td>TACGC</td>
<td>1382720</td>
<td>0.60358167</td>
<td>8.183659</td>
<td>51</td>
</tr>
<tr>
<td>CGGTA</td>
<td>932755</td>
<td>0.4119239</td>
<td>8.190827</td>
<td>60</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
