<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR387661_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR387661_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>100574020</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>16229170</td>
<td>3.0276752</td>
<td>3.9145195</td>
<td>3</td>
</tr>
<tr>
<td>GGCAG</td>
<td>14183425</td>
<td>2.0914128</td>
<td>7.90947</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>13839865</td>
<td>2.0523913</td>
<td>5.907252</td>
<td>1</td>
</tr>
<tr>
<td>GGGCA</td>
<td>13631330</td>
<td>2.010004</td>
<td>6.330231</td>
<td>1</td>
</tr>
<tr>
<td>GCAGG</td>
<td>12988170</td>
<td>1.9151669</td>
<td>5.0130467</td>
<td>2</td>
</tr>
<tr>
<td>GGAGG</td>
<td>14706875</td>
<td>1.832457</td>
<td>5.3553376</td>
<td>2</td>
</tr>
<tr>
<td>GGGAG</td>
<td>12647325</td>
<td>1.5758401</td>
<td>6.2762594</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>12243795</td>
<td>1.5255609</td>
<td>7.658296</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>11806620</td>
<td>1.4794786</td>
<td>5.242462</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>12644600</td>
<td>1.4737015</td>
<td>8.03156</td>
<td>1</td>
</tr>
<tr>
<td>GTGGG</td>
<td>11553465</td>
<td>1.4477559</td>
<td>7.623588</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>11389710</td>
<td>1.4272358</td>
<td>5.1845694</td>
<td>2</td>
</tr>
<tr>
<td>GGGGC</td>
<td>10259120</td>
<td>1.4150109</td>
<td>5.496961</td>
<td>2</td>
</tr>
<tr>
<td>GGGGT</td>
<td>10794715</td>
<td>1.3526776</td>
<td>6.2219796</td>
<td>3</td>
</tr>
<tr>
<td>CGGGG</td>
<td>8445130</td>
<td>1.1648123</td>
<td>7.5813427</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
