<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Thu 28 Aug 2014<br />
SRR315321_2.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315321_2.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>29002949</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>52</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/tick.png" alt="[OK]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td>68968</td>
<td>0.23779650821025133</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAG</td>
<td>45400</td>
<td>0.15653580606579007</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td>34085</td>
<td>0.11752253193287344</td>
<td>No Hit</td>
</tr>
<tr>
<td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td>
<td>33495</td>
<td>0.11548825603906693</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGA</td>
<td>29457</td>
<td>0.10156553390484532</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCAGC</td>
<td>4389685</td>
<td>1.9280517</td>
<td>5.88825</td>
<td>22</td>
</tr>
<tr>
<td>CCCAG</td>
<td>4332300</td>
<td>1.9028467</td>
<td>5.8790827</td>
<td>21</td>
</tr>
<tr>
<td>CCAGG</td>
<td>4349570</td>
<td>1.8975593</td>
<td>5.3759093</td>
<td>62</td>
</tr>
<tr>
<td>GGCTG</td>
<td>4412035</td>
<td>1.8846712</td>
<td>5.719078</td>
<td>36</td>
</tr>
<tr>
<td>CAGCT</td>
<td>3900025</td>
<td>1.8520657</td>
<td>6.0419173</td>
<td>23</td>
</tr>
<tr>
<td>GCCTG</td>
<td>4258315</td>
<td>1.8313473</td>
<td>5.594009</td>
<td>12</td>
</tr>
<tr>
<td>CAGGA</td>
<td>3758600</td>
<td>1.7984358</td>
<td>5.815444</td>
<td>63</td>
</tr>
<tr>
<td>CTGGG</td>
<td>4201370</td>
<td>1.7946823</td>
<td>5.242368</td>
<td>71</td>
</tr>
<tr>
<td>GAGGA</td>
<td>3748315</td>
<td>1.7814295</td>
<td>5.786798</td>
<td>47</td>
</tr>
<tr>
<td>GCTGA</td>
<td>3722620</td>
<td>1.7559067</td>
<td>5.9123616</td>
<td>37</td>
</tr>
<tr>
<td>GGTGG</td>
<td>4052535</td>
<td>1.7194402</td>
<td>5.473859</td>
<td>2</td>
</tr>
<tr>
<td>GGAGG</td>
<td>3967295</td>
<td>1.7075402</td>
<td>5.3831673</td>
<td>33</td>
</tr>
<tr>
<td>AGGAG</td>
<td>3571910</td>
<td>1.6975911</td>
<td>5.814755</td>
<td>64</td>
</tr>
<tr>
<td>TTCTG</td>
<td>3305435</td>
<td>1.673036</td>
<td>5.4445696</td>
<td>69</td>
</tr>
<tr>
<td>TGCCT</td>
<td>3571455</td>
<td>1.6719301</td>
<td>5.7782135</td>
<td>11</td>
</tr>
<tr>
<td>CTGAG</td>
<td>3471780</td>
<td>1.6375891</td>
<td>5.651465</td>
<td>38</td>
</tr>
<tr>
<td>TCTGG</td>
<td>3178130</td>
<td>1.4777751</td>
<td>5.1431947</td>
<td>70</td>
</tr>
<tr>
<td>CCTGT</td>
<td>3141590</td>
<td>1.4706944</td>
<td>5.474715</td>
<td>13</td>
</tr>
<tr>
<td>GTGGC</td>
<td>3305755</td>
<td>1.412106</td>
<td>5.2276654</td>
<td>3</td>
</tr>
<tr>
<td>TCCCA</td>
<td>2953275</td>
<td>1.411982</td>
<td>5.5163655</td>
<td>20</td>
</tr>
<tr>
<td>AGGCT</td>
<td>2945575</td>
<td>1.3893857</td>
<td>5.590683</td>
<td>35</td>
</tr>
<tr>
<td>TGAGG</td>
<td>2932735</td>
<td>1.3740079</td>
<td>5.432674</td>
<td>39</td>
</tr>
<tr>
<td>GAGGC</td>
<td>3082390</td>
<td>1.3356733</td>
<td>5.1662574</td>
<td>34</td>
</tr>
<tr>
<td>CTTGA</td>
<td>2449845</td>
<td>1.2578573</td>
<td>5.642241</td>
<td>55</td>
</tr>
<tr>
<td>TTGAG</td>
<td>2395540</td>
<td>1.2216868</td>
<td>5.4684334</td>
<td>56</td>
</tr>
<tr>
<td>AGGAT</td>
<td>2304440</td>
<td>1.1921693</td>
<td>5.5834</td>
<td>48</td>
</tr>
<tr>
<td>TGGCG</td>
<td>2616910</td>
<td>1.1178548</td>
<td>5.1359754</td>
<td>4</td>
</tr>
<tr>
<td>GATCG</td>
<td>2239265</td>
<td>1.0562294</td>
<td>5.0612283</td>
<td>50</td>
</tr>
<tr>
<td>CTACT</td>
<td>2041230</td>
<td>1.0551665</td>
<td>5.533244</td>
<td>26</td>
</tr>
<tr>
<td>AGTCC</td>
<td>2215890</td>
<td>1.0522944</td>
<td>5.1558757</td>
<td>18</td>
</tr>
<tr>
<td>TGTAG</td>
<td>2026060</td>
<td>1.033258</td>
<td>5.0255303</td>
<td>15</td>
</tr>
<tr>
<td>CTGTA</td>
<td>1947770</td>
<td>1.0000701</td>
<td>5.346424</td>
<td>14</td>
</tr>
<tr>
<td>AGCTA</td>
<td>1617200</td>
<td>0.84231114</td>
<td>5.311835</td>
<td>24</td>
</tr>
<tr>
<td>TACTC</td>
<td>1365125</td>
<td>0.70566964</td>
<td>5.232747</td>
<td>27</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
