<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 26 Sep 2014<br />
c786-O.4_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>c786-O.4_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>2106792</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>85</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>583090</td>
<td>2.7851589</td>
<td>7.0609365</td>
<td>2</td>
</tr>
<tr>
<td>TTTTT</td>
<td>505680</td>
<td>2.504496</td>
<td>6.137917</td>
<td>2</td>
</tr>
<tr>
<td>CCAGG</td>
<td>355075</td>
<td>2.4269</td>
<td>5.0130873</td>
<td>8</td>
</tr>
<tr>
<td>CCAGC</td>
<td>354330</td>
<td>2.355004</td>
<td>5.0040097</td>
<td>8</td>
</tr>
<tr>
<td>CTGGG</td>
<td>324475</td>
<td>2.297244</td>
<td>6.47448</td>
<td>1</td>
</tr>
<tr>
<td>CTGGA</td>
<td>349500</td>
<td>2.2309752</td>
<td>6.8560834</td>
<td>1</td>
</tr>
<tr>
<td>TCCAG</td>
<td>350420</td>
<td>2.1751456</td>
<td>6.8328824</td>
<td>7</td>
</tr>
<tr>
<td>CTTCA</td>
<td>368230</td>
<td>2.0758038</td>
<td>5.260229</td>
<td>1</td>
</tr>
<tr>
<td>CTCCA</td>
<td>340095</td>
<td>2.0528235</td>
<td>5.6298957</td>
<td>6</td>
</tr>
<tr>
<td>CAGAA</td>
<td>336985</td>
<td>1.9254564</td>
<td>5.000725</td>
<td>1</td>
</tr>
<tr>
<td>AAAAT</td>
<td>399825</td>
<td>1.9236691</td>
<td>5.5086</td>
<td>3</td>
</tr>
<tr>
<td>GAAAA</td>
<td>361700</td>
<td>1.9162022</td>
<td>5.209506</td>
<td>1</td>
</tr>
<tr>
<td>CTTTG</td>
<td>287965</td>
<td>1.6815158</td>
<td>5.235931</td>
<td>1</td>
</tr>
<tr>
<td>GGAAA</td>
<td>285355</td>
<td>1.6767051</td>
<td>5.175895</td>
<td>1</td>
</tr>
<tr>
<td>CTTGG</td>
<td>252035</td>
<td>1.62052</td>
<td>5.158614</td>
<td>1</td>
</tr>
<tr>
<td>ATCCA</td>
<td>219970</td>
<td>1.231075</td>
<td>5.39679</td>
<td>6</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
