# Introduction
[Expression Atlas](https://www.ebi.ac.uk/gxa) is an open science resource that gives users a powerful way to find information about gene and protein expression across species and biological conditions such as different tissues, cell types, developmental stages and diseases among others. Expression Atlas aims to help answering questions such as ‘where is a certain gene expressed?’ or ‘how does its expression change in a disease?’. It provides gene expression results on more than 3,000 experiments (microarray and RNA-sequencing) from over 30 different organisms. Expression profiles of tissues from [Human Protein Atlas](https://www.ebi.ac.uk/gxa/experiments/E-PROT-3), [GTEx](https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5214) and [FANTOM5](https://www.ebi.ac.uk/gxa/experiments?experimentSet=FANTOM5), and of [cancer cell lines from ENCODE](https://www.ebi.ac.uk/gxa/experiments?experimentSet=ENCODE), [CCLE](https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2770) and [Genentech projects](https://www.ebi.ac.uk/gxa/experiments/E-GEOD-75431) can be explored in Expression Atlas. All data are manually curated, annotated to ontology terms allowing for much richer queries and re-analysed using standardised methods.

For more details, please read the [Help section](https://www.ebi.ac.uk/gxa/help/index.html) or the [Frequently Asked Questions](https://www.ebi.ac.uk/gxa/FAQ.html).

# Architecture overview
Expression Atlas is made up of a collection of web services, online resources and software to explore gene expression patterns associated or in response to experimental conditions. At its core, it is a [Java EE 7](http://docs.oracle.com/javaee/7/index.html) application built with several components from the [Spring framework](http://projects.spring.io/spring-framework/). It specifically depends on modules from the [*Data Access/Integration*, *Web* and *Core Container* groups](http://docs.spring.io/spring-framework/docs/current/spring-framework-reference/html/overview.html#overview-modules). If in doubt, refer to the `<dependencies>` section in [the parent POM file of the project](https://github.com/gxa/atlas/blob/master/pom.xml).

All the code, resources and static assets are packaged with Maven using the [WAR plugin](https://maven.apache.org/plugins/maven-war-plugin/); the end result is a file with extension `.war` which is actually a zipped archive that follows [the conventions and directory layout of Java EE applications](https://docs.oracle.com/javaee/7/tutorial/packaging003.htm#BCGGCCHE). This WAR file is later then mounted by [Apache Tomcat](http://tomcat.apache.org/). [Other](https://www.eclipse.org/jetty/) [servlet](https://javaee.github.io/glassfish/) [containers](http://undertow.io/) might work although none other but Tomcat 7 have been tested.

# Code layout
Code is split into three modules: [`base`](https://github.com/gxa/atlas/tree/master/base), [`gxa`](https://github.com/gxa/atlas/tree/master/gxa) and [`sc`](https://github.com/gxa/atlas/tree/master/sc). The **base** module contains mostly business logic that is not strictly dependent on typical web application components (e.g. Spring MVC, Tiles, etc.). Classes that represent the model of the problem domain, an API to load/delete/update experiments, utility caches and services to access Solr and DB transactions are contained here. The **gxa** module depends on **base** and is the web layer of *Expression Atlas*. Controllers, static assets (JSPs, CSS files, HTML documents, JS files) plus all the Spring servlet configuration and Java EE web context XML files are in this directory. Finally, **sc** contains the web layer of *Single Cell Expression Atlas*. Go to its [README file](https://github.com/gxa/atlas/blob/master/sc/README.md) to know more. 

We use the [React JavaScript library](https://reactjs.org/) and its ecosystem of components to build rich and single-page applications. Most of the UI code is organised as NPM packages in the [`javascript` directory](https://github.com/gxa/atlas/tree/master/gxa/src/main/javascript). The dependencies are, as usual, given through `package.json` files. Some of them have been moved to their own repositories; see the section *NPM packages* below.

The rest of the JavaScript code, third party or otherwise, may be found under the [`js` directory](https://github.com/gxa/atlas/tree/master/gxa/src/main/webapp/resources/js). Be aware that we provide [two](https://github.com/gxa/atlas/blob/master/gxa/src/main/webapp/resources/js/lib/babel-polyfill.min.js) [polyfills](https://github.com/gxa/atlas/blob/master/gxa/src/main/webapp/resources/js/lib/fetch-polyfill.min.js) for legacy browsers. 
 
The remaining UI code are JSP templates that can be found under the [`jsp` directory](https://github.com/gxa/atlas/tree/master/gxa/src/main/webapp/WEB-INF/jsp). The views are managed with the [Tiles framework](https://tiles.apache.org/). If you want to have a look at the detailed XML definitions, please go to the [`tiles` directory](https://github.com/gxa/atlas/tree/master/gxa/src/main/webapp/WEB-INF/tiles).
 
# NPM packages

- [anatomogram](https://www.npmjs.com/package/anatomogram)
- [expression-atlas-heatmap-highcharts](https://www.npmjs.com/package/expression-atlas-heatmap-highcharts)
- [expression-atlas-experiment-page](https://www.npmjs.com/package/expression-atlas-experiment-page)

# Back end services
The project relies on a [PostgreSQL 9](https://www.postgresql.org/) database and two [Solr 6](http://lucene.apache.org/solr/) cores for both data storage and search. Migrations and  schemas can be found at [`db_updates`](https://github.com/gxa/atlas/tree/master/atlas-misc/scripts/db_updates) and [`solr`](https://github.com/gxa/atlas/tree/master/atlas-misc/solr), respectively.

