<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR315305_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315305_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>36254600</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGAT</td>
<td>105646</td>
<td>0.2914002636906765</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTTGGTGCGCGGACACGCACTGCCTGCGTAACTAGAGGGAGCTGACGGATGACGCCCCCGCGC</td>
<td>85812</td>
<td>0.23669272313030623</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGA</td>
<td>77188</td>
<td>0.21290539683240195</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCGGGGGTGCGTCGGGTCTGCGAGAGCGCCAGCTATCCTGAGGGAAACTTCGGAGGGAACCAG</td>
<td>65531</td>
<td>0.18075223557838177</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>59049</td>
<td>0.16287312506550894</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGATG</td>
<td>52643</td>
<td>0.14520364312390704</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>49353</td>
<td>0.13612893260441433</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>42952</td>
<td>0.11847324201618553</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGGG</td>
<td>4440745</td>
<td>2.448708</td>
<td>8.028926</td>
<td>1</td>
</tr>
<tr>
<td>GGAGG</td>
<td>4987650</td>
<td>2.393192</td>
<td>5.8465543</td>
<td>2</td>
</tr>
<tr>
<td>CCAGC</td>
<td>4492160</td>
<td>2.3327444</td>
<td>5.5767326</td>
<td>30</td>
</tr>
<tr>
<td>GGGAG</td>
<td>4756930</td>
<td>2.2824872</td>
<td>6.656832</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>4031990</td>
<td>2.223313</td>
<td>8.968708</td>
<td>1</td>
</tr>
<tr>
<td>GGCTG</td>
<td>3963900</td>
<td>2.185767</td>
<td>5.287164</td>
<td>1</td>
</tr>
<tr>
<td>TTCAT</td>
<td>4210510</td>
<td>2.010798</td>
<td>6.1305194</td>
<td>42</td>
</tr>
<tr>
<td>CTGGA</td>
<td>3755075</td>
<td>1.8308239</td>
<td>5.2273245</td>
<td>2</td>
</tr>
<tr>
<td>GGCAG</td>
<td>3715515</td>
<td>1.8303919</td>
<td>5.2455254</td>
<td>1</td>
</tr>
<tr>
<td>GTGGG</td>
<td>3362225</td>
<td>1.8057786</td>
<td>7.932273</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>3244230</td>
<td>1.7605422</td>
<td>10.442358</td>
<td>2</td>
</tr>
<tr>
<td>GGGGA</td>
<td>3609940</td>
<td>1.7321343</td>
<td>7.5713286</td>
<td>1</td>
</tr>
<tr>
<td>GCGCC</td>
<td>2929720</td>
<td>1.7206451</td>
<td>5.9519453</td>
<td>27</td>
</tr>
<tr>
<td>TCACT</td>
<td>3408880</td>
<td>1.6888304</td>
<td>5.0482397</td>
<td>52</td>
</tr>
<tr>
<td>TGGGG</td>
<td>2901725</td>
<td>1.558454</td>
<td>6.7581654</td>
<td>2</td>
</tr>
<tr>
<td>GGGGT</td>
<td>2830760</td>
<td>1.5203401</td>
<td>8.740468</td>
<td>3</td>
</tr>
<tr>
<td>GGGTC</td>
<td>2730955</td>
<td>1.5058986</td>
<td>5.0995717</td>
<td>4</td>
</tr>
<tr>
<td>GGGTG</td>
<td>2791350</td>
<td>1.499174</td>
<td>5.3702474</td>
<td>2</td>
</tr>
<tr>
<td>GTGGT</td>
<td>2795315</td>
<td>1.4858378</td>
<td>5.50389</td>
<td>1</td>
</tr>
<tr>
<td>ATCCA</td>
<td>3206175</td>
<td>1.419075</td>
<td>5.449774</td>
<td>37</td>
</tr>
<tr>
<td>CGGGG</td>
<td>2350350</td>
<td>1.309516</td>
<td>8.484901</td>
<td>1</td>
</tr>
<tr>
<td>GGGCT</td>
<td>2360805</td>
<td>1.3017911</td>
<td>5.052591</td>
<td>3</td>
</tr>
<tr>
<td>GGTGC</td>
<td>2217010</td>
<td>1.2224998</td>
<td>6.4515944</td>
<td>5</td>
</tr>
<tr>
<td>GGCGG</td>
<td>2151000</td>
<td>1.1984466</td>
<td>7.5278153</td>
<td>1</td>
</tr>
<tr>
<td>GCGGG</td>
<td>2050535</td>
<td>1.1424717</td>
<td>6.127789</td>
<td>1</td>
</tr>
<tr>
<td>GTGCG</td>
<td>1346120</td>
<td>0.7422752</td>
<td>5.024965</td>
<td>6</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
