<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR307913_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307913_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>33983928</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>53</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGT</td>
<td>100953</td>
<td>0.29706101072248037</td>
<td>Illumina Paired End PCR Primer 2 (98% over 50bp)</td>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGT</td>
<td>50096</td>
<td>0.14741085845050048</td>
<td>Illumina Paired End PCR Primer 2 (100% over 50bp)</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGA</td>
<td>40603</td>
<td>0.11947706574707902</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAG</td>
<td>37779</td>
<td>0.11116725529785726</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>7970425</td>
<td>2.921796</td>
<td>7.1280875</td>
<td>17</td>
</tr>
<tr>
<td>CTGCT</td>
<td>4864660</td>
<td>2.4182625</td>
<td>7.664131</td>
<td>54</td>
</tr>
<tr>
<td>CAGGA</td>
<td>7513330</td>
<td>2.4012403</td>
<td>6.2023997</td>
<td>20</td>
</tr>
<tr>
<td>AAAAA</td>
<td>5332800</td>
<td>2.3507233</td>
<td>6.684351</td>
<td>62</td>
</tr>
<tr>
<td>AGCAG</td>
<td>7273810</td>
<td>2.3246908</td>
<td>5.9954143</td>
<td>18</td>
</tr>
<tr>
<td>TCAGC</td>
<td>5408000</td>
<td>2.3085878</td>
<td>7.623629</td>
<td>16</td>
</tr>
<tr>
<td>TTCAG</td>
<td>4404985</td>
<td>2.2245028</td>
<td>8.080131</td>
<td>15</td>
</tr>
<tr>
<td>GCAGG</td>
<td>8004385</td>
<td>2.1954746</td>
<td>5.3853235</td>
<td>19</td>
</tr>
<tr>
<td>GTTCA</td>
<td>4292650</td>
<td>2.1677737</td>
<td>8.083568</td>
<td>14</td>
</tr>
<tr>
<td>CTTCT</td>
<td>3189240</td>
<td>2.151207</td>
<td>9.958941</td>
<td>51</td>
</tr>
<tr>
<td>AGGAA</td>
<td>6545605</td>
<td>2.1251545</td>
<td>5.922331</td>
<td>21</td>
</tr>
<tr>
<td>TCTTC</td>
<td>3087345</td>
<td>2.0824769</td>
<td>9.532265</td>
<td>50</td>
</tr>
<tr>
<td>GGAAG</td>
<td>7312875</td>
<td>2.037634</td>
<td>6.6301665</td>
<td>5</td>
</tr>
<tr>
<td>GAAGA</td>
<td>6259985</td>
<td>2.0324225</td>
<td>6.775455</td>
<td>6</td>
</tr>
<tr>
<td>TTCTG</td>
<td>3233390</td>
<td>1.9014628</td>
<td>8.340899</td>
<td>52</td>
</tr>
<tr>
<td>CTGGG</td>
<td>5877695</td>
<td>1.8773655</td>
<td>5.4778495</td>
<td>1</td>
</tr>
<tr>
<td>ATGCC</td>
<td>4216730</td>
<td>1.8000536</td>
<td>6.8232946</td>
<td>25</td>
</tr>
<tr>
<td>AAGAG</td>
<td>5334010</td>
<td>1.7317873</td>
<td>6.101979</td>
<td>7</td>
</tr>
<tr>
<td>TCTGC</td>
<td>3401010</td>
<td>1.69067</td>
<td>7.114662</td>
<td>53</td>
</tr>
<tr>
<td>CTTGA</td>
<td>3184155</td>
<td>1.6079875</td>
<td>6.600869</td>
<td>57</td>
</tr>
<tr>
<td>TGCTT</td>
<td>2730790</td>
<td>1.6058984</td>
<td>8.024401</td>
<td>55</td>
</tr>
<tr>
<td>GGTTC</td>
<td>3675570</td>
<td>1.5929804</td>
<td>6.763806</td>
<td>13</td>
</tr>
<tr>
<td>AGAGC</td>
<td>4954470</td>
<td>1.5834355</td>
<td>5.5504203</td>
<td>8</td>
</tr>
<tr>
<td>GCTTG</td>
<td>3399505</td>
<td>1.4733347</td>
<td>6.1740294</td>
<td>56</td>
</tr>
<tr>
<td>GAATG</td>
<td>3868675</td>
<td>1.4626632</td>
<td>5.922143</td>
<td>23</td>
</tr>
<tr>
<td>AATGC</td>
<td>3306970</td>
<td>1.4340935</td>
<td>6.611981</td>
<td>24</td>
</tr>
<tr>
<td>GGAAT</td>
<td>3697845</td>
<td>1.3980757</td>
<td>5.9666805</td>
<td>22</td>
</tr>
<tr>
<td>GTCTT</td>
<td>2311725</td>
<td>1.3594583</td>
<td>8.170162</td>
<td>49</td>
</tr>
<tr>
<td>TGGGG</td>
<td>4793215</td>
<td>1.3347607</td>
<td>5.0803733</td>
<td>1</td>
</tr>
<tr>
<td>TTGAA</td>
<td>2575225</td>
<td>1.3211167</td>
<td>6.079165</td>
<td>58</td>
</tr>
<tr>
<td>TGAAA</td>
<td>2966180</td>
<td>1.3067191</td>
<td>5.5408273</td>
<td>59</td>
</tr>
<tr>
<td>GTGGG</td>
<td>4602865</td>
<td>1.2817541</td>
<td>6.398949</td>
<td>1</td>
</tr>
<tr>
<td>TGCCG</td>
<td>3350050</td>
<td>1.2273213</td>
<td>5.4938273</td>
<td>45</td>
</tr>
<tr>
<td>GATCG</td>
<td>3276945</td>
<td>1.2195894</td>
<td>5.317656</td>
<td>1</td>
</tr>
<tr>
<td>ATCGG</td>
<td>3246205</td>
<td>1.2081487</td>
<td>5.3341684</td>
<td>2</td>
</tr>
<tr>
<td>AGACC</td>
<td>3180870</td>
<td>1.1660424</td>
<td>5.6522403</td>
<td>31</td>
</tr>
<tr>
<td>CGGTT</td>
<td>2641725</td>
<td>1.1449153</td>
<td>6.192166</td>
<td>12</td>
</tr>
<tr>
<td>TCGGA</td>
<td>2916850</td>
<td>1.0855719</td>
<td>5.448589</td>
<td>3</td>
</tr>
<tr>
<td>ATATC</td>
<td>1400060</td>
<td>0.8238304</td>
<td>5.394199</td>
<td>37</td>
</tr>
<tr>
<td>CCGTC</td>
<td>1860320</td>
<td>0.78173536</td>
<td>5.4483056</td>
<td>47</td>
</tr>
<tr>
<td>CCGAT</td>
<td>1723470</td>
<td>0.7357214</td>
<td>5.804867</td>
<td>34</td>
</tr>
<tr>
<td>ACCGA</td>
<td>1984595</td>
<td>0.72751224</td>
<td>5.1482882</td>
<td>33</td>
</tr>
<tr>
<td>TCGTA</td>
<td>1415010</td>
<td>0.7145753</td>
<td>6.8784018</td>
<td>40</td>
</tr>
<tr>
<td>GTATG</td>
<td>1582240</td>
<td>0.6966195</td>
<td>5.99957</td>
<td>42</td>
</tr>
<tr>
<td>TATGC</td>
<td>1364125</td>
<td>0.68887854</td>
<td>6.785692</td>
<td>43</td>
</tr>
<tr>
<td>CGTCT</td>
<td>1242735</td>
<td>0.6177738</td>
<td>6.320628</td>
<td>48</td>
</tr>
<tr>
<td>CGTAT</td>
<td>968900</td>
<td>0.48929122</td>
<td>6.505292</td>
<td>41</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
