<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317049_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317049_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>89965132</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>51</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGT</td>
<td>339439</td>
<td>0.37730061908873763</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTA</td>
<td>136769</td>
<td>0.15202445320704913</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCTGC</td>
<td>12998160</td>
<td>2.0623963</td>
<td>6.1426067</td>
<td>1</td>
</tr>
<tr>
<td>CTGCC</td>
<td>12636340</td>
<td>2.004987</td>
<td>5.77713</td>
<td>2</td>
</tr>
<tr>
<td>AGCCA</td>
<td>9629375</td>
<td>1.9157621</td>
<td>7.3520503</td>
<td>34</td>
</tr>
<tr>
<td>CCATG</td>
<td>10406130</td>
<td>1.887</td>
<td>6.8853235</td>
<td>36</td>
</tr>
<tr>
<td>TGCCA</td>
<td>10393430</td>
<td>1.8846972</td>
<td>6.068281</td>
<td>3</td>
</tr>
<tr>
<td>GCCAG</td>
<td>9543000</td>
<td>1.712825</td>
<td>5.9537497</td>
<td>4</td>
</tr>
<tr>
<td>AAAGA</td>
<td>6561850</td>
<td>1.7051182</td>
<td>8.386582</td>
<td>26</td>
</tr>
<tr>
<td>CAAAG</td>
<td>7449410</td>
<td>1.6937828</td>
<td>7.825828</td>
<td>25</td>
</tr>
<tr>
<td>GCCAT</td>
<td>9053635</td>
<td>1.6417449</td>
<td>6.6040187</td>
<td>35</td>
</tr>
<tr>
<td>TGTCT</td>
<td>9300000</td>
<td>1.6011661</td>
<td>5.9703355</td>
<td>43</td>
</tr>
<tr>
<td>CATGC</td>
<td>8668195</td>
<td>1.5718509</td>
<td>6.418186</td>
<td>37</td>
</tr>
<tr>
<td>AAGAT</td>
<td>6601475</td>
<td>1.5635377</td>
<td>7.7523484</td>
<td>27</td>
</tr>
<tr>
<td>CATGT</td>
<td>8277315</td>
<td>1.5635209</td>
<td>6.6332364</td>
<td>41</td>
</tr>
<tr>
<td>CCAGT</td>
<td>8302380</td>
<td>1.5055157</td>
<td>5.8932743</td>
<td>5</td>
</tr>
<tr>
<td>CTCAA</td>
<td>7127825</td>
<td>1.4326994</td>
<td>5.817409</td>
<td>23</td>
</tr>
<tr>
<td>GCATG</td>
<td>7662510</td>
<td>1.4326189</td>
<td>6.438391</td>
<td>40</td>
</tr>
<tr>
<td>TCTCA</td>
<td>7793815</td>
<td>1.4278657</td>
<td>5.420962</td>
<td>22</td>
</tr>
<tr>
<td>TGCTT</td>
<td>8221040</td>
<td>1.4154035</td>
<td>5.8978705</td>
<td>16</td>
</tr>
<tr>
<td>TCAAA</td>
<td>6000160</td>
<td>1.37833</td>
<td>6.643305</td>
<td>24</td>
</tr>
<tr>
<td>AGCAT</td>
<td>6519395</td>
<td>1.3510839</td>
<td>7.1425295</td>
<td>10</td>
</tr>
<tr>
<td>TGCAT</td>
<td>7034360</td>
<td>1.3287364</td>
<td>6.2950444</td>
<td>39</td>
</tr>
<tr>
<td>ATGCT</td>
<td>7003805</td>
<td>1.3229647</td>
<td>6.234126</td>
<td>15</td>
</tr>
<tr>
<td>CAGTA</td>
<td>6267960</td>
<td>1.2989764</td>
<td>6.8865666</td>
<td>6</td>
</tr>
<tr>
<td>AAGCC</td>
<td>6438790</td>
<td>1.2809957</td>
<td>6.6684775</td>
<td>33</td>
</tr>
<tr>
<td>ATGCA</td>
<td>6093170</td>
<td>1.2627528</td>
<td>6.725354</td>
<td>38</td>
</tr>
<tr>
<td>ATGTC</td>
<td>6565775</td>
<td>1.2402242</td>
<td>6.2295923</td>
<td>42</td>
</tr>
<tr>
<td>AGATT</td>
<td>5551745</td>
<td>1.1984947</td>
<td>6.822084</td>
<td>28</td>
</tr>
<tr>
<td>AGTAC</td>
<td>5591275</td>
<td>1.1587397</td>
<td>6.3380833</td>
<td>49</td>
</tr>
<tr>
<td>GCTTG</td>
<td>6776120</td>
<td>1.1547287</td>
<td>5.602976</td>
<td>17</td>
</tr>
<tr>
<td>ATTAA</td>
<td>4680950</td>
<td>1.1200979</td>
<td>7.5151215</td>
<td>30</td>
</tr>
<tr>
<td>CTTGT</td>
<td>6415370</td>
<td>1.104524</td>
<td>5.5779467</td>
<td>18</td>
</tr>
<tr>
<td>GATTA</td>
<td>4995050</td>
<td>1.0783169</td>
<td>6.9747953</td>
<td>29</td>
</tr>
<tr>
<td>CATAT</td>
<td>4906355</td>
<td>1.0272794</td>
<td>6.9612093</td>
<td>12</td>
</tr>
<tr>
<td>TTGTC</td>
<td>5855145</td>
<td>1.0080708</td>
<td>5.5475755</td>
<td>19</td>
</tr>
<tr>
<td>CTAAG</td>
<td>4838745</td>
<td>1.0027848</td>
<td>6.2630415</td>
<td>46</td>
</tr>
<tr>
<td>AAGTA</td>
<td>4183080</td>
<td>0.9907488</td>
<td>7.30549</td>
<td>48</td>
</tr>
<tr>
<td>TTAAG</td>
<td>4507610</td>
<td>0.97308975</td>
<td>6.6051273</td>
<td>31</td>
</tr>
<tr>
<td>TAGCA</td>
<td>4401255</td>
<td>0.91211915</td>
<td>6.606854</td>
<td>9</td>
</tr>
<tr>
<td>AGTAG</td>
<td>4255085</td>
<td>0.9092017</td>
<td>6.687275</td>
<td>7</td>
</tr>
<tr>
<td>GGCCG</td>
<td>5603480</td>
<td>0.9073408</td>
<td>5.0617604</td>
<td>58</td>
</tr>
<tr>
<td>GTAGC</td>
<td>4826600</td>
<td>0.90240395</td>
<td>5.984965</td>
<td>8</td>
</tr>
<tr>
<td>ATATG</td>
<td>3923125</td>
<td>0.8469129</td>
<td>6.576826</td>
<td>13</td>
</tr>
<tr>
<td>TATGC</td>
<td>4455175</td>
<td>0.8415482</td>
<td>5.807305</td>
<td>14</td>
</tr>
<tr>
<td>GCACG</td>
<td>4681720</td>
<td>0.8402983</td>
<td>5.3733487</td>
<td>54</td>
</tr>
<tr>
<td>TCTAA</td>
<td>3943175</td>
<td>0.8256115</td>
<td>6.2764206</td>
<td>45</td>
</tr>
<tr>
<td>TAAGC</td>
<td>3955380</td>
<td>0.8197157</td>
<td>6.2289715</td>
<td>32</td>
</tr>
<tr>
<td>TAAGT</td>
<td>3684400</td>
<td>0.7953776</td>
<td>6.545222</td>
<td>47</td>
</tr>
<tr>
<td>GCATA</td>
<td>3710125</td>
<td>0.7688889</td>
<td>6.348404</td>
<td>11</td>
</tr>
<tr>
<td>CACGG</td>
<td>3960780</td>
<td>0.7109004</td>
<td>5.3661337</td>
<td>55</td>
</tr>
<tr>
<td>CGCAC</td>
<td>3933875</td>
<td>0.6848124</td>
<td>5.145761</td>
<td>53</td>
</tr>
<tr>
<td>ACGCA</td>
<td>3297700</td>
<td>0.65607667</td>
<td>5.832949</td>
<td>52</td>
</tr>
<tr>
<td>GTACG</td>
<td>3507085</td>
<td>0.65570116</td>
<td>5.398943</td>
<td>50</td>
</tr>
<tr>
<td>ACGGC</td>
<td>3532295</td>
<td>0.6339938</td>
<td>5.2875195</td>
<td>56</td>
</tr>
<tr>
<td>GTCTA</td>
<td>3337565</td>
<td>0.63044035</td>
<td>5.6306157</td>
<td>44</td>
</tr>
<tr>
<td>TACGC</td>
<td>2563785</td>
<td>0.46490508</td>
<td>5.2827907</td>
<td>51</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
