<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Tue 26 Aug 2014<br />
SRR307922_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307922_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>24203201</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCA</td>
<td>30764</td>
<td>0.12710715413221582</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>GAAGA</td>
<td>3937695</td>
<td>2.7829742</td>
<td>5.188479</td>
<td>58</td>
</tr>
<tr>
<td>GGAAG</td>
<td>3533430</td>
<td>2.6586628</td>
<td>5.3170543</td>
<td>60</td>
</tr>
<tr>
<td>AAGAG</td>
<td>3348520</td>
<td>2.3665736</td>
<td>6.5253005</td>
<td>60</td>
</tr>
<tr>
<td>CCCAG</td>
<td>2375385</td>
<td>2.1353836</td>
<td>6.9393663</td>
<td>21</td>
</tr>
<tr>
<td>GATCG</td>
<td>2728375</td>
<td>2.013258</td>
<td>6.4188128</td>
<td>50</td>
</tr>
<tr>
<td>CCAGC</td>
<td>2124180</td>
<td>1.9095596</td>
<td>6.6583505</td>
<td>22</td>
</tr>
<tr>
<td>GCCTG</td>
<td>2269725</td>
<td>1.8529345</td>
<td>6.26482</td>
<td>12</td>
</tr>
<tr>
<td>TGCCT</td>
<td>2559345</td>
<td>1.8520504</td>
<td>5.9559383</td>
<td>11</td>
</tr>
<tr>
<td>GGCTG</td>
<td>2249290</td>
<td>1.7670158</td>
<td>5.807069</td>
<td>36</td>
</tr>
<tr>
<td>GGAGG</td>
<td>2107605</td>
<td>1.6883235</td>
<td>5.7938447</td>
<td>33</td>
</tr>
<tr>
<td>TCCCA</td>
<td>2080300</td>
<td>1.6576964</td>
<td>5.6419687</td>
<td>20</td>
</tr>
<tr>
<td>CAGCT</td>
<td>2157060</td>
<td>1.654053</td>
<td>5.686885</td>
<td>23</td>
</tr>
<tr>
<td>CCTGT</td>
<td>2150635</td>
<td>1.5562904</td>
<td>5.3161654</td>
<td>13</td>
</tr>
<tr>
<td>CTGAG</td>
<td>2094975</td>
<td>1.5458744</td>
<td>5.2544127</td>
<td>38</td>
</tr>
<tr>
<td>GGTGG</td>
<td>2029535</td>
<td>1.5342628</td>
<td>5.5989985</td>
<td>2</td>
</tr>
<tr>
<td>GCTGA</td>
<td>2031260</td>
<td>1.4988592</td>
<td>5.2404656</td>
<td>37</td>
</tr>
<tr>
<td>GAGGC</td>
<td>1711295</td>
<td>1.4245678</td>
<td>5.8295655</td>
<td>34</td>
</tr>
<tr>
<td>AGGCT</td>
<td>1902395</td>
<td>1.4037703</td>
<td>5.2347107</td>
<td>35</td>
</tr>
<tr>
<td>GGGAG</td>
<td>1731900</td>
<td>1.3873602</td>
<td>5.221886</td>
<td>32</td>
</tr>
<tr>
<td>GTGGC</td>
<td>1650150</td>
<td>1.2963386</td>
<td>5.5145946</td>
<td>3</td>
</tr>
<tr>
<td>GTGCC</td>
<td>1464400</td>
<td>1.1954917</td>
<td>5.614582</td>
<td>10</td>
</tr>
<tr>
<td>AGTCC</td>
<td>1455335</td>
<td>1.1159639</td>
<td>5.1787205</td>
<td>18</td>
</tr>
<tr>
<td>GTCCC</td>
<td>1178825</td>
<td>1.0000646</td>
<td>5.097778</td>
<td>19</td>
</tr>
<tr>
<td>TGGCG</td>
<td>1149685</td>
<td>0.90317905</td>
<td>5.535805</td>
<td>4</td>
</tr>
<tr>
<td>GGCGC</td>
<td>869630</td>
<td>0.8009109</td>
<td>5.635286</td>
<td>5</td>
</tr>
<tr>
<td>CGTGC</td>
<td>881555</td>
<td>0.7196747</td>
<td>5.029</td>
<td>9</td>
</tr>
<tr>
<td>GCGCG</td>
<td>524670</td>
<td>0.48321003</td>
<td>5.248751</td>
<td>6</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
