<h1>Frequently Asked Questions (FAQs)</h1>

<!-- Beginning of main text section -->
<section class="grid_18 alpha">

<h2><a name="general">General questions</a></h2>

<h3><a name="whatIsAtlas">What is the Expression Atlas</a></h3>
<p>The Expression Atlas database provides summary level information
about gene expression patterns within different biological conditions.
It consists of two components: the 'differential' and 'baseline'
atlases. The Differential Atlas, containing both microarray and
sequencing data, allows users to query which genes are
up-/down-regulated in different experimental conditions, e.g. 'in
Arabidopsis shoots, what genes are upregulated in plants treated by
X?'  The Baseline Atlas, containing exclusively RNA-seq data, displays
expression levels of gene products under 'normal' condition
(e.g. normal tissues human tissues). Sample characteristcs and
experimental variables are annotated to the <a href="http://www.ebi.ac.uk/efo">Experimental Factor
Ontology</a> which allows us to use ontology-driven query expansion to
broaden searches. E.g. a query for cancer would return experiments
annotated with particular types of cancer such as leukemia or breast carcinoma as well as
those annotated with cancer.</p>

<h3><a name="exptChosen">How are experiments chosen to be in the Expression Atlas?</a></h3>
<p>Experiments in the Expression Atlas are selected from the <a
href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> database of
functional genomics experiments. Different criteria are used for
selecting microarray and high throughput sequencing experiments. For
microarray these criteria include the type of experiment (i.e. those
concerning gene expression), the number of replicates to allow robust
statistical analysis, and whether the array design used can be
re-annotated against Ensembl.  For RNA-seq data the sequences must be
of high quality, there must a good quality reference genome build
available, there must be sufficent replicates for statistical analysis
for differential experiments and the tissues must be 'normal' if the
experiment is to go in the Baseline Atlas.<p> 

<h3><a name="getOrigData">How can I obtain the original data for an experiment?</a></h3>

<h3><a name="wholeExpt">It doesn't look like the whole of the original experiment is in the Expression Atlas, why not?</a></h3>

<h3><a name="microarrayAnalysis">How is microarray data analysed?</a></h3>

<p>Raw single-channel microarray intensities are normalized using <a
	href="http://biostatistics.oxfordjournals.org/content/4/2/249.long">RMA</a>
via the <a
	href="http://www.bioconductor.org/packages/release/bioc/html/oligo.html">oligo</a>
package from <a href="http://www.bioconductor.org/">Bioconductor</a> (<a
	href="http://www.affymetrix.com/">Affymetrix</a> data) or using quantile
normalization via the <a
	href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>
package (<a href="http://www.genomics.agilent.com/">Agilent</a> data).
Two-channel <a href="http://www.genomics.agilent.com/">Agilent</a> data is
normalized using LOESS via
the <a
	href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>
package. Pairwise contrasts are performed using a moderated <i>t</i>-test for
each gene using <a
	href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>.</p>

<h3><a name="rnaseqAnalysis">How is RNA-seq data analysed?</a></h3>

<p><a href="http://nar.oxfordjournals.org/content/38/6/1767.long">FASTQ</a> files are mapped to the reference genome from <a
	href="www.ensembl.org">Ensembl</a> using <a href="http://tophat.cbcb.umd.edu/">TopHat</a>. FPKMs are
generated by <a href="http://cufflinks.cbcb.umd.edu/">cufflinks</a>. Raw counts are generated using <a
	href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html">htseq-count</a>. Pairwise contrasts are performed using a
conditioned test based on the negative binomial distribution, using <a
	href="http://www.bioconductor.org/packages/release/bioc/html/DESeq.html">DESeq</a>.</p>

<h3><a name="analysisMethods">How do I see how an individual experiment was analysed?</a></h3>

<p>On any experiment page, you can see a breakdown of the steps from raw data
to the results you see in Expression Atlas by clicking the <img alt="analysis
methods button" src="resources/images/help/am_btn.png" align="bottom" /> button
to the top right of the page.</p>


<h3><a name="contact">How can I contact you?</a></h3>

<p>If you have any questions, problems or suggestions we would love to hear
from you. Please use email us at <a
	href="mailto:arrayexpress-atlas@ebi.ac.uk">arrayexpress-atlas@ebi.ac.uk</a>.</p>



<h2><a name="searching">Searching</a></h2>


<h3><a name="tissueSearch">How do I find out which genes are expressed in my favourite tissue?</a></h3>

<p>Use the <strong>Experimental conditions</strong> search box on the <a
	href="/gxa">home page</a> to search all of Expression Atlas for organism parts e.g. <a
	href="/gxa/query?geneQuery=&exactMatch=true&_exactMatch=on&condition=kidney">kidney</a>.
You will see results about baseline and differential expression of genes in the
organism part(s) you searched for. You can narrow down the search to specific
genes by also typing gene identifiers in the <strong>Gene query</strong> search
box.</p>

<p>Use the <strong>Organism part</strong> box on a Baseline experiment page (e.g. <a
	href="experiments/E-MTAB-513">Illumina Body Map</a>) to search within a
single experiment. See the <a href="help/baseline-atlas.html">Baseline Atlas
	help</a> for more information about searching within a Baseline experiment.</p>


<h3><a name="geneSearch">How do I search for my favourite gene?</a></h3>

<p>Use the <strong>Gene query</strong> search box on the <a href="/gxa">home
	page</a> to search all of Expression Atlas for genes. You will see results
about baseline and differential expression for the gene or genes you searched
for. You can narrow down the search to specific experimental conditions by also
typing something in the <strong>Experimental conditions</strong> search
box.</p>

<p>Use the <strong>Gene query</strong> search box on an experiment page (e.g. <a
	href="experiments/E-MTAB-513">Illumina Body Map</a>) to search for genes
within a single experiment.</p>


<h3><a name="geneIDs">What gene identifiers can I use to search?</a></h3>

<p>You may use the following identifiers to search using the <strong>Gene query</strong> box:</p>

<ul>
	<li><a href="http://www.ebi.ac.uk/ena/">EMBL</a> e.g. <strong>AC000015</strong></li>
	<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Gene e.g. <strong>ENSG00000171658</strong></li>
	<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Protein e.g. <strong>ENSP00000000233</strong></li>
	<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Transcript e.g. <strong>ENST00000000233</strong></li>
	<li><a href="http://www.geneontology.org/">Gene Ontology</a> e.g. <strong>GO:0008134</strong></li>
	<li><a href="http://www.ebi.ac.uk/interpro/">Interpro</a> e.g. <strong>IPR017892</strong></li>
	<li><a href="http://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> e.g. <strong>NM_212505</strong></li>
	<li><a href="http://www.uniprot.org/">UniProt</a> e.g. <strong>Q8IZT6</strong></li>
	<li><a href="http://www.uniprot.org/">UniProt</a> Metabolic Enzyme e.g. <strong>O15269</strong></li>
	<li>gene or protein name e.g. <strong>WNT1</strong> or <strong>Wnt-1</strong></li>
	<li><a href="http://www.geneontology.org/">GO</a> term e.g. <strong>&quot;transcription factor binding&quot;</strong> (please put multi-word query terms in quotes)</li>
	<li>keyword e.g. <strong>transcription</strong></li>
	<li>synonym e.g. <strong>Calmbp1</strong></li>
</ul>


<h3><a name="exptLink">How do I find a particular experiment?</a></h3>

<p>All experiments currently in Expression Atlas are listed on <a
	href="experiments">this page</a>. Click on the <strong>Description</strong>
to see the experiment in Expression Atlas. Click on the <a
	href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> accession number
(e.g. <strong>E-MTAB-1066</strong>) to see the experiment in ArrayExpress. If you know the
ArrayExpress accession of the experiment you want to see, you can link to the
experiment in Expression Atlas
using the following format:</p>

<p><code>http://www-test.ebi.ac.uk/gxa/experiments/<strong>&lt;ArrayExpress accession&gt;</strong></code></p>

<p>E.g. <a
	href="experiments/E-MTAB-1066">http://www-test.ebi.ac.uk/gxa/experiments/E-MTAB-1066</a></p>


<h3><a name="api">Can I search Expression Atlas programmatically?</a></h3>

<p>Not right now, but we are working to implement an API. Please check back soon!<p>


<h2><a name="results">Results</a></h2>


<h3><a name="deResults">When I search all of Expression Atlas, what do the &quot;Differential Expression&quot; results show?</a></h3>

<p>The table in the <strong>Differential Expression</strong> section of the
results shows all genes that matched your search and were differentially
expressed with adjusted <i>p</i>-value &lt; 0.05. There is one row for each
gene and each contrast the gene showed differential expression in. The genes
are ordered so that the one with the lowest adjusted <i>p</i>-value is at the
top.</p>


<h3><a name="designElement">What is a &quot;design element&quot;?</a></h3>

<p>In results from microarray experiments, you will see the <strong>design
	element</strong> name alongside the gene name. A design element is also
known as a probe or probe set. This is the oligonucleotide probe (or group
thereof) on the microarray that targets that gene.</p>


<h3><a name="contrast">What is a &quot;contrast&quot;?</a></h3>

<p>A <strong>contrast</strong> is a comparison between two groups of samples in
a differential expression experiment. An example of a contrast is
<strong>&quot;dicer knock-out&quot; <i>vs.</i> &quot;wild
	type&quot;</strong>. For each gene, the mean expression level of the
<strong>test</strong> group (e.g. dicer knock-out) is compared with the
mean expression level of the <strong>reference</strong> group (e.g. wild type),
and a statistical test is performed to decide whether the two means are
significantly different.</p>


<h3><a name="pvalue">Where do the <i>p</i>-values come from?</a></h3>

<p>In a microarray experiment, each gene's mean expression level in the
<strong>test</strong> group is compared with its mean expression level in the
<strong>reference</strong> group using a moderated <i>t</i>-test. This is done
using the <a
	href="http://www.bioconductor.org/packages/2.13/bioc/html/limma.html">limma</a>
package from <a href="http://www.bioconductor.org/">Bioconductor</a>.</p>

<p>In an RNA-seq experiment, each gene's mean expression level in the
<strong>test</strong> group is compared with its mean expression level in the
<strong>reference</strong> group using a conditioned test based on the negative
binomial distribution, analogous to Fisher's exact test. This is done using the
<a
	href="http://www.bioconductor.org/packages/2.13/bioc/html/DESeq.html">DESeq</a>
package in <a href="http://www.bioconductor.org/">Bioconductor</a>.</p>

<p>Because the same test is done on thousands of genes at once, <i>p</i>-values
are adjusted for <strong>multiple testing</strong> using the Benjamini and Hochberg (<a
	href="http://www.jstor.org/stable/2346101">1995</a>) false discovery rate
(FDR) correction. This is what is meant by <strong>adjusted
	<i>p</i>-value</strong>.</p>


<h3><a name="deHeatmapColours">What do the red and blue colours in the table mean?</a></h3>

<p>The intensity of the red a blue colour of the filled boxes in the table
represents how low the adjusted <i>p</i>-value is for each gene. The lower the
<i>p</i>-value, the more intense the red or blue colour. A red box indicates
that the gene was <strong>up-regulated</strong> in the test condition. A blue
box means that the gene was <strong>down-regulated</strong> in the test
condition.</p>


<h3><a name="displayPvalues">How do I see the actual <i>p</i>-values in the results table?</a></h3>

<p>Click the <strong>Display levels</strong> button to the top left of the table.</p>


<h3><a name="deBars">What are the blue and red bars above the results table?</a></h3>

<p>The two bars show the red and blue colour intensities for the <strong>top
	50</strong> genes shown on the page. The colour intensities represent the
adjusted <i>p</i>-values for the genes shown. To see the actual
<i>p</i>-values, click the &quot;Display levels&quot; button to the top left of
the table.</p>

<h3><a name="DEexptPage">What does the differential expression experiment page show?</a></h3>

<p>Each experiment in Expression Atlas has its own experiment page, e.g. <a
	href="experiments/E-MTAB-1066"><i>D. melanogaster</i> CDK8 and CycC
	mutants</a>. Here you can see the results of analysis for each contrast in
a single experiment. Please see the <a
	href="help/differential-atlas.html">Differential Atlas help page</a> for
more details.</p>


<h3><a name="downloadAnalytics">Can I download the differential expression results?</a></h3>

<p>Yes, click on the <img src="resources/images/help/dlAna_btn.png"
alt="analytics download button" align="bottom" /> button to the top right of
any differential experiment page, e.g. <a
	href="experiments/E-MTAB-1066"><i>D. melanogaster</i> CDK8 and CycC
	mutants</a>. For more details about the differential expression experiment pages,
please see the <a href="help/differential-atlas.html">Differential Atlas help
	page</a>.</p>

<h3><a name="downloadDEdata">Can I download the data used for differential expression analysis?</a></h3>

<p>Yes, please use the download buttons to the top right of any experiment
page. For single-channel microarray experiments you can download the normalized
intensities using the <img src="resources/images/help/dlNorm_btn.png"
alt="normalized data download button" align="bottom" /> button. For two-channel
microarray experiments you can download the log<sub>2</sub> fold changes using
the <img src="resources/images/help/download_blue_small_logfold.png"
alt="download log2 fold changes button" align="bottom" /> button. For RNA-seq
experiments you can download the raw counts generated by <a
	href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html">htseq-count</a>
using the <img src="resources/images/help/dlRawCounts_btn.png" alt="download
raw RNA-seq counts" align="bottom" /> button. For more details about the differential expression experiment pages,
please see the <a href="help/differential-atlas.html">Differential Atlas help
	page</a>.</p>


</section>
<!-- End of main text section -->

<!-- This part is for the navigation section at the right-hand side -->
<aside class="grid_6 omega">
	<h3>Jump to...</h3>
	<h4><a href="FAQ.html#general">General</a></h4>
	<ul>
		<li><a href="FAQ.html#whatIsAtlas">What is the Expression Atlas</a></li>
		<li><a href="FAQ.html#exptChosen">How are experiments chosen to be in the Expression Atlas?</a></li>
		<li><a href="FAQ.html#getOrigData">How can I obtain the original data for an experiment?</a></li>
		<li><a href="FAQ.html#wholeExpt">It doesn't look like the whole of the original experiment is in the Expression Atlas, why not?</a></li>
		<li><a href="FAQ.html#microarrayAnalysis">How is microarray data analysed?</a></li>
		<li><a href="FAQ.html#rnaseqAnalysis">How is RNA-seq data analysed?</a></li>
		<li><a href="FAQ.html#analysisMethods">How do I see how an individual experiment was analysed?</a></li>
		<li><a href="FAQ.html#contact">How can I contact you?</a></li>
	</ul>
	<h4><a href="FAQ.html#searching">Searching</a></h4>
	<ul>
		<li><a href="FAQ.html#tissueSearch">How do I find out which genes are expressed in my favourite tissue?</a></li>
		<li><a href="FAQ.html#geneSearch">How do I search for my favourite gene?</a></li>
		<li><a href="FAQ.html#geneIDs">What gene identifiers can I use to search?</a></li>
		<li><a href="FAQ.html#exptLink">How do I find a particular experiment?</a></li>
		<li><a href="FAQ.html#api">Can I search Expression Atlas programmatically?</a></li>
	</ul>
	<h4><a href="FAQ.html#results">Results</a></h4>
	<ul>
		<li><a href="FAQ.html#deResults">When I search all of Expression Atlas, what do the &quot;Differential Expression&quot; results show?</a></li>
		<li><a href="FAQ.html#designElement">What is a &quot;design element&quot;?</a></li>
		<li><a href="FAQ.html#contrast">What is a &quot;contrast&quot;?</a></li>
		<li><a href="FAQ.html#pvalue">Where do the <i>p</i>-values come from?</a></li>
		<li><a href="FAQ.html#deHeatmapColours">What do the red and blue colours in the table mean?</a></li>
		<li><a href="FAQ.html#displayPvalues">How do I see the actual <i>p</i>-values in the results table?</a></li>
		<li><a href="FAQ.html#deBars">What are the blue and red bars above the results table?</a></li>
		<li><a href="FAQ.html#downloadAnalytics">Can I download the differential expression results?</a></li>
		<li><a href="FAQ.html#downloadDEdata">Can I download the data used for differential expression analysis?</a></li>
		<!-- <li><a href="FAQ.html#"></a></li> -->
	</ul>
</aside>
