<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Mon 25 Aug 2014<br />
SRR307911_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307911_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>41419823</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>51</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/tick.png" alt="[OK]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>8987860</td>
<td>3.3160212</td>
<td>5.849566</td>
<td>62</td>
</tr>
<tr>
<td>CAGCA</td>
<td>8239350</td>
<td>2.9617977</td>
<td>6.852046</td>
<td>17</td>
</tr>
<tr>
<td>CTGCT</td>
<td>6527245</td>
<td>2.5264492</td>
<td>9.991103</td>
<td>1</td>
</tr>
<tr>
<td>CTTCT</td>
<td>5449300</td>
<td>2.511315</td>
<td>6.431846</td>
<td>51</td>
</tr>
<tr>
<td>TCTTC</td>
<td>5103050</td>
<td>2.3517456</td>
<td>6.0581393</td>
<td>50</td>
</tr>
<tr>
<td>TTTTT</td>
<td>5245630</td>
<td>2.3283818</td>
<td>5.6868362</td>
<td>1</td>
</tr>
<tr>
<td>AGCAG</td>
<td>7357415</td>
<td>2.1799967</td>
<td>5.4604425</td>
<td>18</td>
</tr>
<tr>
<td>TGCTG</td>
<td>6769700</td>
<td>2.1598234</td>
<td>8.969058</td>
<td>2</td>
</tr>
<tr>
<td>CAGGA</td>
<td>7015325</td>
<td>2.0786357</td>
<td>5.435824</td>
<td>20</td>
</tr>
<tr>
<td>GCAGG</td>
<td>8038190</td>
<td>2.0757167</td>
<td>5.0962276</td>
<td>19</td>
</tr>
<tr>
<td>GGCAG</td>
<td>8021130</td>
<td>2.0713117</td>
<td>6.12324</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>7532465</td>
<td>2.0183945</td>
<td>5.190648</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>7498700</td>
<td>2.009347</td>
<td>7.23363</td>
<td>3</td>
</tr>
<tr>
<td>GGAAG</td>
<td>7445395</td>
<td>1.8183877</td>
<td>5.6432214</td>
<td>5</td>
</tr>
<tr>
<td>TGGCA</td>
<td>5859980</td>
<td>1.801714</td>
<td>7.1782045</td>
<td>5</td>
</tr>
<tr>
<td>CTGGC</td>
<td>5464895</td>
<td>1.7765712</td>
<td>7.5603113</td>
<td>4</td>
</tr>
<tr>
<td>TCAGC</td>
<td>4750900</td>
<td>1.7721374</td>
<td>6.0609784</td>
<td>16</td>
</tr>
<tr>
<td>GAAGA</td>
<td>5992860</td>
<td>1.6793981</td>
<td>5.6140285</td>
<td>6</td>
</tr>
<tr>
<td>TTCAG</td>
<td>4501895</td>
<td>1.6480293</td>
<td>5.771738</td>
<td>15</td>
</tr>
<tr>
<td>TGGGG</td>
<td>7367980</td>
<td>1.6273671</td>
<td>5.243339</td>
<td>2</td>
</tr>
<tr>
<td>GGGGG</td>
<td>8339080</td>
<td>1.5469469</td>
<td>5.0377355</td>
<td>1</td>
</tr>
<tr>
<td>GGCAC</td>
<td>4733905</td>
<td>1.4830686</td>
<td>7.078115</td>
<td>6</td>
</tr>
<tr>
<td>GTGGG</td>
<td>6639360</td>
<td>1.4664367</td>
<td>7.5620804</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>6884110</td>
<td>1.4652975</td>
<td>5.8632145</td>
<td>1</td>
</tr>
<tr>
<td>ATGCC</td>
<td>3770430</td>
<td>1.4064114</td>
<td>5.3964505</td>
<td>25</td>
</tr>
<tr>
<td>TGTGT</td>
<td>4277525</td>
<td>1.3393356</td>
<td>5.975922</td>
<td>2</td>
</tr>
<tr>
<td>AAGAG</td>
<td>4774915</td>
<td>1.3380896</td>
<td>5.3622303</td>
<td>7</td>
</tr>
<tr>
<td>GTTCA</td>
<td>3545930</td>
<td>1.2980747</td>
<td>5.287326</td>
<td>14</td>
</tr>
<tr>
<td>GTGTG</td>
<td>4593940</td>
<td>1.2080983</td>
<td>6.1263876</td>
<td>1</td>
</tr>
<tr>
<td>TACAA</td>
<td>2894050</td>
<td>1.1714865</td>
<td>6.704987</td>
<td>6</td>
</tr>
<tr>
<td>CGGGG</td>
<td>4710120</td>
<td>1.060037</td>
<td>5.511722</td>
<td>1</td>
</tr>
<tr>
<td>GTGTA</td>
<td>3082235</td>
<td>0.93004394</td>
<td>5.5215964</td>
<td>3</td>
</tr>
<tr>
<td>AGACC</td>
<td>2560805</td>
<td>0.9205321</td>
<td>5.099381</td>
<td>31</td>
</tr>
<tr>
<td>GTACA</td>
<td>2503165</td>
<td>0.88307977</td>
<td>6.0366154</td>
<td>5</td>
</tr>
<tr>
<td>TGTAC</td>
<td>2324905</td>
<td>0.85108864</td>
<td>6.1263547</td>
<td>4</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
