<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Mon 25 Aug 2014<br />
SRR307906_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307906_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>35869269</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>52</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGT</td>
<td>75500</td>
<td>0.21048658672135193</td>
<td>Illumina Paired End PCR Primer 2 (98% over 50bp)</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>7056525</td>
<td>3.4981394</td>
<td>6.7572083</td>
<td>62</td>
</tr>
<tr>
<td>CAGCA</td>
<td>7313660</td>
<td>2.926425</td>
<td>6.6201735</td>
<td>17</td>
</tr>
<tr>
<td>CTTCT</td>
<td>5108440</td>
<td>2.5005116</td>
<td>6.6667957</td>
<td>51</td>
</tr>
<tr>
<td>TTTTT</td>
<td>4503330</td>
<td>2.411173</td>
<td>6.182234</td>
<td>1</td>
</tr>
<tr>
<td>TCTTC</td>
<td>4873480</td>
<td>2.3855019</td>
<td>6.421984</td>
<td>50</td>
</tr>
<tr>
<td>CTGCT</td>
<td>5756560</td>
<td>2.375445</td>
<td>6.381982</td>
<td>1</td>
</tr>
<tr>
<td>CAGGA</td>
<td>6485070</td>
<td>2.287893</td>
<td>5.5537276</td>
<td>20</td>
</tr>
<tr>
<td>GCAGG</td>
<td>7383320</td>
<td>2.2299945</td>
<td>5.0971437</td>
<td>19</td>
</tr>
<tr>
<td>GGCAG</td>
<td>7361090</td>
<td>2.2232802</td>
<td>7.0320444</td>
<td>1</td>
</tr>
<tr>
<td>AGCAG</td>
<td>6194645</td>
<td>2.185433</td>
<td>5.55669</td>
<td>18</td>
</tr>
<tr>
<td>TGCTG</td>
<td>5962165</td>
<td>2.1692238</td>
<td>6.1516633</td>
<td>2</td>
</tr>
<tr>
<td>CTGGG</td>
<td>6816545</td>
<td>2.0907688</td>
<td>5.4333034</td>
<td>1</td>
</tr>
<tr>
<td>GGAAG</td>
<td>6414055</td>
<td>1.9951344</td>
<td>5.8486605</td>
<td>5</td>
</tr>
<tr>
<td>TTCTG</td>
<td>4321900</td>
<td>1.8652356</td>
<td>5.152801</td>
<td>52</td>
</tr>
<tr>
<td>AGGAA</td>
<td>5002135</td>
<td>1.8174555</td>
<td>5.309256</td>
<td>21</td>
</tr>
<tr>
<td>TCAGC</td>
<td>4327285</td>
<td>1.7583599</td>
<td>5.607161</td>
<td>16</td>
</tr>
<tr>
<td>TTCAG</td>
<td>4052270</td>
<td>1.7221371</td>
<td>5.702897</td>
<td>15</td>
</tr>
<tr>
<td>GAAGA</td>
<td>4728580</td>
<td>1.7180631</td>
<td>5.8590746</td>
<td>6</td>
</tr>
<tr>
<td>TGGGG</td>
<td>6125920</td>
<td>1.6566528</td>
<td>5.067419</td>
<td>2</td>
</tr>
<tr>
<td>GGGAG</td>
<td>5959345</td>
<td>1.5869713</td>
<td>5.0734625</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>6704020</td>
<td>1.5283995</td>
<td>5.037307</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>5389315</td>
<td>1.4351726</td>
<td>5.756491</td>
<td>1</td>
</tr>
<tr>
<td>AAGAG</td>
<td>3913435</td>
<td>1.4218916</td>
<td>5.7115297</td>
<td>7</td>
</tr>
<tr>
<td>GTGGG</td>
<td>5254100</td>
<td>1.4208837</td>
<td>6.733602</td>
<td>1</td>
</tr>
<tr>
<td>GTTCA</td>
<td>3233205</td>
<td>1.37405</td>
<td>5.28512</td>
<td>14</td>
</tr>
<tr>
<td>GTCTT</td>
<td>2932075</td>
<td>1.2654182</td>
<td>5.1816344</td>
<td>49</td>
</tr>
<tr>
<td>ATGCC</td>
<td>3110095</td>
<td>1.2637639</td>
<td>5.041527</td>
<td>25</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
