<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 26 Sep 2014<br />
c786-O.20_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>c786-O.20_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>2932789</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>85</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>818125</td>
<td>2.6961882</td>
<td>6.584104</td>
<td>2</td>
</tr>
<tr>
<td>TTTTT</td>
<td>729740</td>
<td>2.4714413</td>
<td>5.93916</td>
<td>2</td>
</tr>
<tr>
<td>CTGGG</td>
<td>441215</td>
<td>2.3199272</td>
<td>6.797387</td>
<td>1</td>
</tr>
<tr>
<td>CCCAG</td>
<td>454385</td>
<td>2.2307131</td>
<td>5.2947555</td>
<td>1</td>
</tr>
<tr>
<td>CTGGA</td>
<td>479560</td>
<td>2.2289844</td>
<td>7.079823</td>
<td>1</td>
</tr>
<tr>
<td>TCCAG</td>
<td>480020</td>
<td>2.1617541</td>
<td>6.352658</td>
<td>7</td>
</tr>
<tr>
<td>CAGGA</td>
<td>453065</td>
<td>2.0943742</td>
<td>5.0399218</td>
<td>1</td>
</tr>
<tr>
<td>CTTCA</td>
<td>519340</td>
<td>2.078782</td>
<td>5.036786</td>
<td>1</td>
</tr>
<tr>
<td>CTCCA</td>
<td>467705</td>
<td>2.040807</td>
<td>5.301591</td>
<td>6</td>
</tr>
<tr>
<td>CAGAA</td>
<td>473765</td>
<td>1.9359611</td>
<td>5.034402</td>
<td>1</td>
</tr>
<tr>
<td>GAAAA</td>
<td>512425</td>
<td>1.910385</td>
<td>5.032467</td>
<td>1</td>
</tr>
<tr>
<td>AAAAT</td>
<td>569090</td>
<td>1.8857399</td>
<td>5.308993</td>
<td>3</td>
</tr>
<tr>
<td>CTTTT</td>
<td>462935</td>
<td>1.7091291</td>
<td>5.007547</td>
<td>1</td>
</tr>
<tr>
<td>CTTTG</td>
<td>404780</td>
<td>1.6813715</td>
<td>5.0872307</td>
<td>1</td>
</tr>
<tr>
<td>CTTGG</td>
<td>347825</td>
<td>1.6255307</td>
<td>5.078203</td>
<td>1</td>
</tr>
<tr>
<td>ATCCA</td>
<td>308585</td>
<td>1.2284621</td>
<td>5.286684</td>
<td>6</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
