<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR315305_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315305_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>44873239</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGC</td>
<td>124393</td>
<td>0.27720976415364174</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTTGGTGCGCGGACACGCACTGCCTGCGTAACTAGAGGGAGCTGACGGA</td>
<td>109436</td>
<td>0.24387809402392369</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCG</td>
<td>90325</td>
<td>0.20128923610796182</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCGGGGGTGCGTCGGGTCTGCGAGAGCGCCAGCTATCCTGAGGGAAACT</td>
<td>82059</td>
<td>0.1828684575232022</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGG</td>
<td>66042</td>
<td>0.14717457770320524</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCG</td>
<td>61698</td>
<td>0.13749397497247748</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTACGAATGGTTTAGCGCCAGGTTCCCCACGAACGTGCGGTGCGTGACGG</td>
<td>56408</td>
<td>0.1257052115181612</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGA</td>
<td>54656</td>
<td>0.12180088003007762</td>
<td>No Hit</td>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGT</td>
<td>51010</td>
<td>0.11367577009540139</td>
<td>Illumina Paired End PCR Primer 2 (98% over 50bp)</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAG</td>
<td>47234</td>
<td>0.10526095519870986</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCTTGTTTGGAGGTTCTAGCAGGGGAGCGCAGCTACTCGTATACCCTTGA</td>
<td>45350</td>
<td>0.10106246174919534</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTTTT</td>
<td>9018315</td>
<td>3.7597032</td>
<td>5.717579</td>
<td>12</td>
</tr>
<tr>
<td>AAAAA</td>
<td>14015155</td>
<td>3.4013216</td>
<td>4.6024346</td>
<td>62</td>
</tr>
<tr>
<td>GGAGG</td>
<td>7778150</td>
<td>2.3703635</td>
<td>5.581735</td>
<td>2</td>
</tr>
<tr>
<td>CTGGG</td>
<td>6602705</td>
<td>2.2313697</td>
<td>7.0559874</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>7273405</td>
<td>2.2165442</td>
<td>6.103726</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>6359995</td>
<td>2.0517178</td>
<td>9.718915</td>
<td>2</td>
</tr>
<tr>
<td>TTCAT</td>
<td>5606915</td>
<td>1.9833452</td>
<td>6.3588533</td>
<td>42</td>
</tr>
<tr>
<td>GCTGG</td>
<td>5698620</td>
<td>1.9258361</td>
<td>7.8037095</td>
<td>1</td>
</tr>
<tr>
<td>GCGCC</td>
<td>5692060</td>
<td>1.8099908</td>
<td>5.540466</td>
<td>27</td>
</tr>
<tr>
<td>GGGGA</td>
<td>5808945</td>
<td>1.7702553</td>
<td>6.938428</td>
<td>1</td>
</tr>
<tr>
<td>GGCGG</td>
<td>5449340</td>
<td>1.7495246</td>
<td>7.039431</td>
<td>1</td>
</tr>
<tr>
<td>GTGGG</td>
<td>4705460</td>
<td>1.5978514</td>
<td>7.1568723</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>4913070</td>
<td>1.577354</td>
<td>7.641485</td>
<td>1</td>
</tr>
<tr>
<td>GCGGG</td>
<td>4752595</td>
<td>1.525833</td>
<td>6.157521</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>4240505</td>
<td>1.4399648</td>
<td>6.172003</td>
<td>2</td>
</tr>
<tr>
<td>GTGGT</td>
<td>3901800</td>
<td>1.3946743</td>
<td>5.165305</td>
<td>1</td>
</tr>
<tr>
<td>GGGGT</td>
<td>4014365</td>
<td>1.3631735</td>
<td>8.064932</td>
<td>3</td>
</tr>
<tr>
<td>ATCCA</td>
<td>4299110</td>
<td>1.2903249</td>
<td>5.1228967</td>
<td>37</td>
</tr>
<tr>
<td>GGTGC</td>
<td>3272830</td>
<td>1.1060457</td>
<td>5.8929024</td>
<td>5</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
