<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR315316_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315316_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>29550797</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>257004</td>
<td>0.8697024313760472</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>225619</td>
<td>0.7634954820338687</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>154860</td>
<td>0.5240467795166405</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGAC</td>
<td>76553</td>
<td>0.2590556187029406</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAA</td>
<td>54493</td>
<td>0.18440450184812274</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTGTGTACAAAGGGCAGGGACTTAATCAACGCAAGCTTATGACCCGCACTTACTGGGAATTCCT</td>
<td>34227</td>
<td>0.1158242872434202</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCACA</td>
<td>4145635</td>
<td>2.8600667</td>
<td>14.02115</td>
<td>27</td>
</tr>
<tr>
<td>CACAG</td>
<td>4537685</td>
<td>2.8190875</td>
<td>12.95255</td>
<td>28</td>
</tr>
<tr>
<td>AGGAG</td>
<td>5553750</td>
<td>2.7979424</td>
<td>11.251318</td>
<td>48</td>
</tr>
<tr>
<td>ATCCA</td>
<td>3332200</td>
<td>2.1488926</td>
<td>12.584003</td>
<td>35</td>
</tr>
<tr>
<td>ACCAC</td>
<td>3056735</td>
<td>2.1088364</td>
<td>13.289305</td>
<td>26</td>
</tr>
<tr>
<td>AGCTC</td>
<td>3427155</td>
<td>2.102268</td>
<td>11.930306</td>
<td>14</td>
</tr>
<tr>
<td>TCCAA</td>
<td>3128470</td>
<td>2.01751</td>
<td>12.474696</td>
<td>36</td>
</tr>
<tr>
<td>AGAGG</td>
<td>3945425</td>
<td>1.987679</td>
<td>10.031785</td>
<td>46</td>
</tr>
<tr>
<td>GAGGA</td>
<td>3858295</td>
<td>1.9437836</td>
<td>10.013348</td>
<td>47</td>
</tr>
<tr>
<td>GGAGA</td>
<td>3837745</td>
<td>1.9334307</td>
<td>9.986444</td>
<td>44</td>
</tr>
<tr>
<td>CCAAG</td>
<td>3084790</td>
<td>1.9164604</td>
<td>11.941448</td>
<td>37</td>
</tr>
<tr>
<td>GCTCT</td>
<td>3117800</td>
<td>1.8883506</td>
<td>11.69841</td>
<td>15</td>
</tr>
<tr>
<td>ACAGT</td>
<td>3204805</td>
<td>1.8611201</td>
<td>11.351375</td>
<td>29</td>
</tr>
<tr>
<td>GGCAT</td>
<td>3269600</td>
<td>1.8060844</td>
<td>11.458936</td>
<td>4</td>
</tr>
<tr>
<td>CAGTT</td>
<td>3108745</td>
<td>1.7825346</td>
<td>10.959617</td>
<td>30</td>
</tr>
<tr>
<td>AGAAT</td>
<td>3166485</td>
<td>1.7408794</td>
<td>10.581232</td>
<td>20</td>
</tr>
<tr>
<td>CATGT</td>
<td>3032000</td>
<td>1.7385294</td>
<td>10.994709</td>
<td>6</td>
</tr>
<tr>
<td>GCATG</td>
<td>3123680</td>
<td>1.7254802</td>
<td>10.833525</td>
<td>5</td>
</tr>
<tr>
<td>GAGAG</td>
<td>3411035</td>
<td>1.7184569</td>
<td>9.761249</td>
<td>45</td>
</tr>
<tr>
<td>CCAAA</td>
<td>2560755</td>
<td>1.6725212</td>
<td>12.696179</td>
<td>60</td>
</tr>
<tr>
<td>GTAGG</td>
<td>3357225</td>
<td>1.6699866</td>
<td>9.572755</td>
<td>41</td>
</tr>
<tr>
<td>GGAGC</td>
<td>3122435</td>
<td>1.6616036</td>
<td>10.26134</td>
<td>49</td>
</tr>
<tr>
<td>AGTAG</td>
<td>3038480</td>
<td>1.5889795</td>
<td>10.027972</td>
<td>40</td>
</tr>
<tr>
<td>AGCGA</td>
<td>2800790</td>
<td>1.5669093</td>
<td>13.360904</td>
<td>55</td>
</tr>
<tr>
<td>GAGCG</td>
<td>2902925</td>
<td>1.5447913</td>
<td>12.709816</td>
<td>54</td>
</tr>
<tr>
<td>TACCA</td>
<td>2370780</td>
<td>1.5288854</td>
<td>11.940342</td>
<td>25</td>
</tr>
<tr>
<td>CAAGT</td>
<td>2568140</td>
<td>1.491391</td>
<td>10.8498</td>
<td>38</td>
</tr>
<tr>
<td>TAGGA</td>
<td>2838515</td>
<td>1.4844073</td>
<td>9.939646</td>
<td>42</td>
</tr>
<tr>
<td>GTGGG</td>
<td>3137160</td>
<td>1.4843621</td>
<td>5.117027</td>
<td>1</td>
</tr>
<tr>
<td>TAGAA</td>
<td>2639640</td>
<td>1.451229</td>
<td>10.36995</td>
<td>19</td>
</tr>
<tr>
<td>CTCTA</td>
<td>2275465</td>
<td>1.4488851</td>
<td>11.750571</td>
<td>16</td>
</tr>
<tr>
<td>GACCA</td>
<td>2318980</td>
<td>1.4406922</td>
<td>11.726739</td>
<td>58</td>
</tr>
<tr>
<td>GAATT</td>
<td>2641180</td>
<td>1.4337364</td>
<td>10.027926</td>
<td>21</td>
</tr>
<tr>
<td>AATTA</td>
<td>2503745</td>
<td>1.4288636</td>
<td>10.456991</td>
<td>22</td>
</tr>
<tr>
<td>AAGTA</td>
<td>2576935</td>
<td>1.4167548</td>
<td>10.244765</td>
<td>39</td>
</tr>
<tr>
<td>TATCC</td>
<td>2153250</td>
<td>1.3710655</td>
<td>11.958786</td>
<td>34</td>
</tr>
<tr>
<td>ATGTA</td>
<td>2498760</td>
<td>1.3564252</td>
<td>10.397923</td>
<td>7</td>
</tr>
<tr>
<td>ACCAA</td>
<td>2064980</td>
<td>1.3487127</td>
<td>12.298419</td>
<td>59</td>
</tr>
<tr>
<td>GGGGG</td>
<td>2950205</td>
<td>1.3447636</td>
<td>5.840564</td>
<td>2</td>
</tr>
<tr>
<td>TAGCT</td>
<td>2344395</td>
<td>1.344261</td>
<td>10.690751</td>
<td>13</td>
</tr>
<tr>
<td>TCTAG</td>
<td>2296965</td>
<td>1.3170651</td>
<td>10.596487</td>
<td>17</td>
</tr>
<tr>
<td>TATTA</td>
<td>2322140</td>
<td>1.3084863</td>
<td>10.536906</td>
<td>10</td>
</tr>
<tr>
<td>TTATC</td>
<td>2178030</td>
<td>1.2963616</td>
<td>10.827863</td>
<td>33</td>
</tr>
<tr>
<td>ATTAC</td>
<td>2136085</td>
<td>1.2876586</td>
<td>11.054615</td>
<td>23</td>
</tr>
<tr>
<td>AGTTA</td>
<td>2371850</td>
<td>1.2875334</td>
<td>10.1318245</td>
<td>31</td>
</tr>
<tr>
<td>CTAGA</td>
<td>2177990</td>
<td>1.2648199</td>
<td>10.644744</td>
<td>18</td>
</tr>
<tr>
<td>ATTAG</td>
<td>2296415</td>
<td>1.2465844</td>
<td>10.025483</td>
<td>11</td>
</tr>
<tr>
<td>TGTAT</td>
<td>2309900</td>
<td>1.2380683</td>
<td>10.026537</td>
<td>8</td>
</tr>
<tr>
<td>TTACC</td>
<td>1942290</td>
<td>1.2367383</td>
<td>11.387246</td>
<td>24</td>
</tr>
<tr>
<td>TTAGC</td>
<td>2144365</td>
<td>1.2295653</td>
<td>10.628482</td>
<td>12</td>
</tr>
<tr>
<td>GTTAT</td>
<td>2249455</td>
<td>1.2056708</td>
<td>9.888575</td>
<td>32</td>
</tr>
<tr>
<td>GTATT</td>
<td>2235935</td>
<td>1.1984243</td>
<td>10.004508</td>
<td>9</td>
</tr>
<tr>
<td>CGGCA</td>
<td>1919060</td>
<td>1.1340532</td>
<td>11.001096</td>
<td>3</td>
</tr>
<tr>
<td>TCGGC</td>
<td>1918565</td>
<td>1.1194417</td>
<td>10.602741</td>
<td>2</td>
</tr>
<tr>
<td>GCGAG</td>
<td>1867725</td>
<td>0.9939097</td>
<td>9.622282</td>
<td>52</td>
</tr>
<tr>
<td>CGAGC</td>
<td>1651920</td>
<td>0.976189</td>
<td>10.586786</td>
<td>53</td>
</tr>
<tr>
<td>GTCGG</td>
<td>1827935</td>
<td>0.9604501</td>
<td>10.244762</td>
<td>1</td>
</tr>
<tr>
<td>CGACC</td>
<td>1359415</td>
<td>0.8920882</td>
<td>11.631011</td>
<td>57</td>
</tr>
<tr>
<td>CCGTC</td>
<td>1342910</td>
<td>0.8701272</td>
<td>10.10264</td>
<td>1</td>
</tr>
<tr>
<td>GCGAC</td>
<td>1410155</td>
<td>0.83331984</td>
<td>10.466764</td>
<td>56</td>
</tr>
<tr>
<td>CGTCG</td>
<td>1218045</td>
<td>0.71070325</td>
<td>9.039251</td>
<td>2</td>
</tr>
<tr>
<td>CCCGT</td>
<td>1075665</td>
<td>0.6969681</td>
<td>7.3445477</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
