<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Mon 25 Aug 2014<br />
SRR307905_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307905_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>25427541</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>52</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGCT</td>
<td>3432745</td>
<td>2.4692473</td>
<td>10.794138</td>
<td>1</td>
</tr>
<tr>
<td>TGCTG</td>
<td>3850295</td>
<td>2.292465</td>
<td>9.8316</td>
<td>2</td>
</tr>
<tr>
<td>GGCAG</td>
<td>4565245</td>
<td>2.172192</td>
<td>6.480827</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>4429230</td>
<td>2.1314425</td>
<td>7.7897325</td>
<td>3</td>
</tr>
<tr>
<td>CTGGG</td>
<td>4218745</td>
<td>2.0301526</td>
<td>5.490839</td>
<td>1</td>
</tr>
<tr>
<td>GGGCA</td>
<td>3981210</td>
<td>1.8943022</td>
<td>5.213268</td>
<td>1</td>
</tr>
<tr>
<td>TGGCA</td>
<td>3168755</td>
<td>1.8654608</td>
<td>7.92314</td>
<td>5</td>
</tr>
<tr>
<td>CTGGC</td>
<td>3154175</td>
<td>1.8337729</td>
<td>7.920134</td>
<td>4</td>
</tr>
<tr>
<td>TGGGG</td>
<td>3971390</td>
<td>1.5818802</td>
<td>5.3336916</td>
<td>2</td>
</tr>
<tr>
<td>GGCAC</td>
<td>2738745</td>
<td>1.5743458</td>
<td>7.526695</td>
<td>6</td>
</tr>
<tr>
<td>GTGGG</td>
<td>3638335</td>
<td>1.4492182</td>
<td>7.392364</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>3654190</td>
<td>1.4391663</td>
<td>5.8959036</td>
<td>1</td>
</tr>
<tr>
<td>TGTGT</td>
<td>2358065</td>
<td>1.4378494</td>
<td>5.9513617</td>
<td>2</td>
</tr>
<tr>
<td>GTGTG</td>
<td>2730065</td>
<td>1.3454493</td>
<td>6.0061793</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>3846695</td>
<td>1.2383847</td>
<td>5.0814824</td>
<td>1</td>
</tr>
<tr>
<td>TACAA</td>
<td>1330445</td>
<td>1.172191</td>
<td>7.2115855</td>
<td>6</td>
</tr>
<tr>
<td>GTACA</td>
<td>1421045</td>
<td>1.023428</td>
<td>6.25899</td>
<td>5</td>
</tr>
<tr>
<td>CGGGG</td>
<td>2412675</td>
<td>0.9383848</td>
<td>5.2286315</td>
<td>1</td>
</tr>
<tr>
<td>GTGTA</td>
<td>1496820</td>
<td>0.902435</td>
<td>5.4836307</td>
<td>3</td>
</tr>
<tr>
<td>TGTAC</td>
<td>1218005</td>
<td>0.887176</td>
<td>6.1212707</td>
<td>4</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
