<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Wed 27 Aug 2014<br />
SRR307933_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307933_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>41141780</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/tick.png" alt="[OK]"> Overrepresented sequences</h2>
<p>No overrepresented sequences</p>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>10314160</td>
<td>3.161645</td>
<td>4.033576</td>
<td>10</td>
</tr>
<tr>
<td>CTGGG</td>
<td>5755900</td>
<td>2.3092542</td>
<td>7.226339</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>4955710</td>
<td>2.2929096</td>
<td>7.9484944</td>
<td>1</td>
</tr>
<tr>
<td>GGCAG</td>
<td>5329885</td>
<td>2.2667294</td>
<td>6.768312</td>
<td>1</td>
</tr>
<tr>
<td>GGAGG</td>
<td>5318725</td>
<td>2.266357</td>
<td>5.3597383</td>
<td>2</td>
</tr>
<tr>
<td>GCTGG</td>
<td>5492765</td>
<td>2.203685</td>
<td>6.047255</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>4797835</td>
<td>1.9286023</td>
<td>5.73912</td>
<td>2</td>
</tr>
<tr>
<td>GGGCA</td>
<td>4450830</td>
<td>1.8928789</td>
<td>5.521655</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>4393395</td>
<td>1.8720652</td>
<td>6.248411</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>4167745</td>
<td>1.7759136</td>
<td>7.76082</td>
<td>1</td>
</tr>
<tr>
<td>GGGGC</td>
<td>3755715</td>
<td>1.7343423</td>
<td>5.292976</td>
<td>2</td>
</tr>
<tr>
<td>GGGAA</td>
<td>4192715</td>
<td>1.6453401</td>
<td>5.1596026</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>3852480</td>
<td>1.5485947</td>
<td>5.0341916</td>
<td>2</td>
</tr>
<tr>
<td>GTGGG</td>
<td>3711495</td>
<td>1.4919225</td>
<td>7.2611947</td>
<td>1</td>
</tr>
<tr>
<td>GGGGT</td>
<td>3698185</td>
<td>1.4865723</td>
<td>6.297947</td>
<td>1</td>
</tr>
<tr>
<td>TGTGT</td>
<td>4313805</td>
<td>1.308852</td>
<td>6.6271825</td>
<td>2</td>
</tr>
<tr>
<td>GTGTG</td>
<td>3736270</td>
<td>1.3048241</td>
<td>8.0105505</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>2414500</td>
<td>1.114986</td>
<td>7.879843</td>
<td>1</td>
</tr>
<tr>
<td>TACAA</td>
<td>3438745</td>
<td>1.0776491</td>
<td>6.208864</td>
<td>6</td>
</tr>
<tr>
<td>GTACA</td>
<td>2759325</td>
<td>0.9389461</td>
<td>6.7125163</td>
<td>5</td>
</tr>
<tr>
<td>TGTAC</td>
<td>2605080</td>
<td>0.8362488</td>
<td>6.222262</td>
<td>4</td>
</tr>
<tr>
<td>GTGTA</td>
<td>2391830</td>
<td>0.7692785</td>
<td>6.4742804</td>
<td>3</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
