<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sun 24 Aug 2014<br />
SRR307899_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307899_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>42690836</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>51</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GGCAGACGTTCGAATGGGTCGTCGCCGCCACGGGGGGCGTGCGATCGGCC</td>
<td>52521</td>
<td>0.12302640313719788</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>8118340</td>
<td>2.8628306</td>
<td>5.013737</td>
<td>17</td>
</tr>
<tr>
<td>TTTTT</td>
<td>7371210</td>
<td>2.7792833</td>
<td>6.402261</td>
<td>1</td>
</tr>
<tr>
<td>GGCAG</td>
<td>8045695</td>
<td>2.181904</td>
<td>9.403104</td>
<td>1</td>
</tr>
<tr>
<td>GCAGG</td>
<td>7639855</td>
<td>2.0718448</td>
<td>5.6660275</td>
<td>2</td>
</tr>
<tr>
<td>CTGGG</td>
<td>7565630</td>
<td>2.0173147</td>
<td>5.775196</td>
<td>1</td>
</tr>
<tr>
<td>GGAGG</td>
<td>7979400</td>
<td>1.9053084</td>
<td>5.440793</td>
<td>2</td>
</tr>
<tr>
<td>GGGGG</td>
<td>8891755</td>
<td>1.8543996</td>
<td>6.912794</td>
<td>1</td>
</tr>
<tr>
<td>GGGCA</td>
<td>6650030</td>
<td>1.8034151</td>
<td>5.6893706</td>
<td>1</td>
</tr>
<tr>
<td>GGGAA</td>
<td>6306260</td>
<td>1.7240349</td>
<td>5.037718</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>6795085</td>
<td>1.6225193</td>
<td>7.595266</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>6858750</td>
<td>1.6102614</td>
<td>5.4939075</td>
<td>2</td>
</tr>
<tr>
<td>GGGAG</td>
<td>6620570</td>
<td>1.5808489</td>
<td>6.353081</td>
<td>1</td>
</tr>
<tr>
<td>GTGGG</td>
<td>6114000</td>
<td>1.4354129</td>
<td>8.085951</td>
<td>1</td>
</tr>
<tr>
<td>GGGGT</td>
<td>5514010</td>
<td>1.2945505</td>
<td>5.981281</td>
<td>3</td>
</tr>
<tr>
<td>CGGGG</td>
<td>4818025</td>
<td>1.1412005</td>
<td>6.899137</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
