<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317048_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317048_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>70788979</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>53</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGT</td>
<td>264564</td>
<td>0.3737361433055843</td>
<td>Illumina Paired End PCR Primer 2 (98% over 50bp)</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGA</td>
<td>113507</td>
<td>0.1603455814781564</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAG</td>
<td>103182</td>
<td>0.14575997769370286</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGG</td>
<td>90426</td>
<td>0.1277402235169969</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>15209860</td>
<td>2.7845578</td>
<td>7.4149733</td>
<td>17</td>
</tr>
<tr>
<td>AAAAA</td>
<td>12537465</td>
<td>2.780411</td>
<td>8.671848</td>
<td>62</td>
</tr>
<tr>
<td>CTGCT</td>
<td>10586180</td>
<td>2.3842843</td>
<td>8.272919</td>
<td>54</td>
</tr>
<tr>
<td>CAGGA</td>
<td>14575405</td>
<td>2.3778799</td>
<td>6.635242</td>
<td>20</td>
</tr>
<tr>
<td>AGCAG</td>
<td>13760130</td>
<td>2.2448735</td>
<td>6.455414</td>
<td>18</td>
</tr>
<tr>
<td>TCAGC</td>
<td>10996845</td>
<td>2.233022</td>
<td>7.8524528</td>
<td>16</td>
</tr>
<tr>
<td>GTTCA</td>
<td>9506830</td>
<td>2.19646</td>
<td>8.156708</td>
<td>14</td>
</tr>
<tr>
<td>GCAGG</td>
<td>15409930</td>
<td>2.1839335</td>
<td>5.903203</td>
<td>19</td>
</tr>
<tr>
<td>TTCAG</td>
<td>9396690</td>
<td>2.171013</td>
<td>8.060637</td>
<td>15</td>
</tr>
<tr>
<td>CTTCT</td>
<td>7398245</td>
<td>2.1275113</td>
<td>10.109207</td>
<td>51</td>
</tr>
<tr>
<td>AGGAA</td>
<td>12542040</td>
<td>2.0989707</td>
<td>6.4089704</td>
<td>21</td>
</tr>
<tr>
<td>TCTTC</td>
<td>7148445</td>
<td>2.0556765</td>
<td>9.843377</td>
<td>50</td>
</tr>
<tr>
<td>GGAAG</td>
<td>14082270</td>
<td>2.0472941</td>
<td>6.8285017</td>
<td>5</td>
</tr>
<tr>
<td>GAAGA</td>
<td>12228560</td>
<td>2.0465083</td>
<td>7.0651836</td>
<td>6</td>
</tr>
<tr>
<td>TTCTG</td>
<td>7516885</td>
<td>1.9262797</td>
<td>8.641721</td>
<td>52</td>
</tr>
<tr>
<td>CTGGG</td>
<td>11823285</td>
<td>1.8585352</td>
<td>5.7220664</td>
<td>1</td>
</tr>
<tr>
<td>AAGAG</td>
<td>10507905</td>
<td>1.7585484</td>
<td>6.613105</td>
<td>7</td>
</tr>
<tr>
<td>ATGCC</td>
<td>8331490</td>
<td>1.6917943</td>
<td>6.9923267</td>
<td>44</td>
</tr>
<tr>
<td>TCTGC</td>
<td>7433750</td>
<td>1.6742746</td>
<td>7.7039495</td>
<td>53</td>
</tr>
<tr>
<td>AGAGC</td>
<td>9871380</td>
<td>1.6104498</td>
<td>6.179402</td>
<td>8</td>
</tr>
<tr>
<td>GGTTC</td>
<td>7947055</td>
<td>1.5950093</td>
<td>6.978446</td>
<td>13</td>
</tr>
<tr>
<td>CTTGA</td>
<td>6847495</td>
<td>1.5820467</td>
<td>7.7841</td>
<td>57</td>
</tr>
<tr>
<td>TGCTT</td>
<td>6058305</td>
<td>1.5525035</td>
<td>8.326266</td>
<td>55</td>
</tr>
<tr>
<td>GCTTG</td>
<td>7072080</td>
<td>1.4193978</td>
<td>6.823058</td>
<td>56</td>
</tr>
<tr>
<td>GGAAT</td>
<td>7629395</td>
<td>1.4161918</td>
<td>6.4247375</td>
<td>22</td>
</tr>
<tr>
<td>GAATG</td>
<td>7592885</td>
<td>1.4094146</td>
<td>6.3027387</td>
<td>23</td>
</tr>
<tr>
<td>AATGC</td>
<td>6571815</td>
<td>1.3689231</td>
<td>6.8801713</td>
<td>24</td>
</tr>
<tr>
<td>GTCTT</td>
<td>5328180</td>
<td>1.3654014</td>
<td>8.476043</td>
<td>49</td>
</tr>
<tr>
<td>TTGAA</td>
<td>5705820</td>
<td>1.3523045</td>
<td>7.4458914</td>
<td>58</td>
</tr>
<tr>
<td>GAAAA</td>
<td>6950420</td>
<td>1.3389976</td>
<td>6.352843</td>
<td>60</td>
</tr>
<tr>
<td>TGAAA</td>
<td>6180655</td>
<td>1.3206784</td>
<td>7.059423</td>
<td>59</td>
</tr>
<tr>
<td>GTGGG</td>
<td>8985095</td>
<td>1.2586168</td>
<td>6.343701</td>
<td>1</td>
</tr>
<tr>
<td>ATCGG</td>
<td>6929770</td>
<td>1.253955</td>
<td>5.704037</td>
<td>2</td>
</tr>
<tr>
<td>GATCG</td>
<td>6710915</td>
<td>1.2143528</td>
<td>5.6059885</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>9352115</td>
<td>1.1811005</td>
<td>5.0959334</td>
<td>1</td>
</tr>
<tr>
<td>AGACC</td>
<td>6421340</td>
<td>1.1755923</td>
<td>6.1242332</td>
<td>31</td>
</tr>
<tr>
<td>TGCCG</td>
<td>6617205</td>
<td>1.1672636</td>
<td>5.7507367</td>
<td>45</td>
</tr>
<tr>
<td>CGGTT</td>
<td>5621270</td>
<td>1.1282138</td>
<td>6.2747445</td>
<td>12</td>
</tr>
<tr>
<td>GAGAC</td>
<td>6890130</td>
<td>1.1240788</td>
<td>5.4565916</td>
<td>30</td>
</tr>
<tr>
<td>TCGGA</td>
<td>6163955</td>
<td>1.1153795</td>
<td>5.782852</td>
<td>3</td>
</tr>
<tr>
<td>GCGGT</td>
<td>7083815</td>
<td>1.1135244</td>
<td>5.1254992</td>
<td>11</td>
</tr>
<tr>
<td>CGGAA</td>
<td>6517245</td>
<td>1.0632449</td>
<td>5.3784523</td>
<td>4</td>
</tr>
<tr>
<td>GCCGA</td>
<td>6169850</td>
<td>0.9812395</td>
<td>5.068938</td>
<td>27</td>
</tr>
<tr>
<td>CCGAG</td>
<td>5727090</td>
<td>0.9108241</td>
<td>5.1297274</td>
<td>28</td>
</tr>
<tr>
<td>GATAT</td>
<td>3754530</td>
<td>0.8898401</td>
<td>5.667561</td>
<td>36</td>
</tr>
<tr>
<td>ATATC</td>
<td>3149700</td>
<td>0.8376977</td>
<td>6.217742</td>
<td>37</td>
</tr>
<tr>
<td>CGAGA</td>
<td>4987365</td>
<td>0.81365526</td>
<td>5.144592</td>
<td>29</td>
</tr>
<tr>
<td>CCGTC</td>
<td>4095980</td>
<td>0.81080025</td>
<td>5.9934006</td>
<td>47</td>
</tr>
<tr>
<td>ACCGA</td>
<td>4186145</td>
<td>0.76638204</td>
<td>5.703065</td>
<td>33</td>
</tr>
<tr>
<td>TCGTA</td>
<td>3273070</td>
<td>0.7562108</td>
<td>7.078944</td>
<td>40</td>
</tr>
<tr>
<td>ATCGT</td>
<td>3150385</td>
<td>0.72786564</td>
<td>5.451859</td>
<td>39</td>
</tr>
<tr>
<td>GCCGT</td>
<td>3991580</td>
<td>0.7041077</td>
<td>5.457744</td>
<td>46</td>
</tr>
<tr>
<td>CCGAT</td>
<td>3372495</td>
<td>0.68481964</td>
<td>5.9316254</td>
<td>34</td>
</tr>
<tr>
<td>GTATG</td>
<td>3287660</td>
<td>0.6768816</td>
<td>6.2219906</td>
<td>42</td>
</tr>
<tr>
<td>TATGC</td>
<td>2859580</td>
<td>0.66067797</td>
<td>6.9705825</td>
<td>43</td>
</tr>
<tr>
<td>CGTCT</td>
<td>2727015</td>
<td>0.6141949</td>
<td>6.7476563</td>
<td>48</td>
</tr>
<tr>
<td>TATCG</td>
<td>2328095</td>
<td>0.5378836</td>
<td>5.359952</td>
<td>38</td>
</tr>
<tr>
<td>CGTAT</td>
<td>2176955</td>
<td>0.5029641</td>
<td>6.6225877</td>
<td>41</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
