<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Mon 25 Aug 2014<br />
SRR307900_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR307900_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>31900150</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>52</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/tick.png" alt="[OK]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGGATCGCTTGAGCCCAG</td>
<td>65831</td>
<td>0.2063658007877706</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGGATCGCTTGAGCCCA</td>
<td>60335</td>
<td>0.18913704167535264</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCAG</td>
<td>37361</td>
<td>0.11711857154276704</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCA</td>
<td>34675</td>
<td>0.1086985484394274</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>GGCTG</td>
<td>4940505</td>
<td>2.3053868</td>
<td>5.69265</td>
<td>35</td>
</tr>
<tr>
<td>GGAGG</td>
<td>4949360</td>
<td>2.2761946</td>
<td>5.5051017</td>
<td>32</td>
</tr>
<tr>
<td>GAGGA</td>
<td>4247310</td>
<td>2.1462247</td>
<td>5.7745247</td>
<td>46</td>
</tr>
<tr>
<td>CCCAG</td>
<td>4823160</td>
<td>2.1448023</td>
<td>5.484339</td>
<td>20</td>
</tr>
<tr>
<td>CCAGC</td>
<td>4755495</td>
<td>2.1147125</td>
<td>5.514089</td>
<td>21</td>
</tr>
<tr>
<td>CAGCT</td>
<td>4093595</td>
<td>2.075439</td>
<td>5.816417</td>
<td>22</td>
</tr>
<tr>
<td>GCTGA</td>
<td>3964615</td>
<td>2.0327053</td>
<td>5.8662953</td>
<td>36</td>
</tr>
<tr>
<td>CTGAG</td>
<td>3926805</td>
<td>2.0133197</td>
<td>5.6931014</td>
<td>37</td>
</tr>
<tr>
<td>GCCTG</td>
<td>4270000</td>
<td>1.9702988</td>
<td>5.4874434</td>
<td>11</td>
</tr>
<tr>
<td>GGTGG</td>
<td>4086500</td>
<td>1.928378</td>
<td>5.264168</td>
<td>1</td>
</tr>
<tr>
<td>TGCCT</td>
<td>3525960</td>
<td>1.8342681</td>
<td>5.806414</td>
<td>10</td>
</tr>
<tr>
<td>AGGCT</td>
<td>3557645</td>
<td>1.8240469</td>
<td>5.702176</td>
<td>34</td>
</tr>
<tr>
<td>GAGGC</td>
<td>3873115</td>
<td>1.7613777</td>
<td>5.473479</td>
<td>33</td>
</tr>
<tr>
<td>TGAGG</td>
<td>3239160</td>
<td>1.6794766</td>
<td>5.6063952</td>
<td>38</td>
</tr>
<tr>
<td>CCTGT</td>
<td>3165395</td>
<td>1.6466961</td>
<td>5.6397376</td>
<td>12</td>
</tr>
<tr>
<td>GTGGC</td>
<td>3421965</td>
<td>1.5967904</td>
<td>5.6344132</td>
<td>2</td>
</tr>
<tr>
<td>TCCCA</td>
<td>3027740</td>
<td>1.5179429</td>
<td>5.5137944</td>
<td>19</td>
</tr>
<tr>
<td>AGGAT</td>
<td>2492315</td>
<td>1.4198598</td>
<td>5.572473</td>
<td>47</td>
</tr>
<tr>
<td>TTGAG</td>
<td>2183590</td>
<td>1.2764212</td>
<td>5.6009984</td>
<td>55</td>
</tr>
<tr>
<td>CTACT</td>
<td>2071200</td>
<td>1.1838821</td>
<td>5.4096074</td>
<td>25</td>
</tr>
<tr>
<td>CTTGA</td>
<td>2044030</td>
<td>1.1815223</td>
<td>5.4747486</td>
<td>54</td>
</tr>
<tr>
<td>GCTTG</td>
<td>2169915</td>
<td>1.141554</td>
<td>5.017247</td>
<td>53</td>
</tr>
<tr>
<td>CTGTA</td>
<td>1951735</td>
<td>1.1281724</td>
<td>5.5115123</td>
<td>13</td>
</tr>
<tr>
<td>AGCTA</td>
<td>1664435</td>
<td>0.93765086</td>
<td>5.1961794</td>
<td>23</td>
</tr>
<tr>
<td>TACTC</td>
<td>1493185</td>
<td>0.8534932</td>
<td>5.146389</td>
<td>26</td>
</tr>
<tr>
<td>TGTAG</td>
<td>1419010</td>
<td>0.8294846</td>
<td>5.1727557</td>
<td>14</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
