<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Wed 27 Aug 2014<br />
SRR315304_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315304_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>27489268</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>55</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGT</td>
<td>78766</td>
<td>0.2865336392369561</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTCACCCGGCCCGGACACGGACAGGATTGACAGATTGATAGCTCTTTCTCGATTCCGTGGGTG</td>
<td>58841</td>
<td>0.2140508070276735</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGAAGGCCCGCGGCGGGTGTTGACGCGATGTGATTTCTGCCCAGTGCTCTGAATGTCAAAGTGA</td>
<td>57212</td>
<td>0.20812485803550682</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCCGTTCTTAGTTGGTGGAGCGATTTGTCTGGTTAATTCCGATAACGAACGAGACTCTGGCAT</td>
<td>53411</td>
<td>0.1942976437204512</td>
<td>No Hit</td>
</tr>
<tr>
<td>GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATTAAAAAAA</td>
<td>50245</td>
<td>0.18278042179951826</td>
<td>Illumina Single End PCR Primer 1 (100% over 57bp)</td>
</tr>
<tr>
<td>GCCCCTCTCCGGCCCCGGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTCGG</td>
<td>45789</td>
<td>0.16657045942438337</td>
<td>No Hit</td>
</tr>
<tr>
<td>AGTTGGTGGAGCGATTTGTCTGGTTAATTCCGATAACGAACGAGACTCTGGCATGCTAACTAGT</td>
<td>41789</td>
<td>0.1520193262330594</td>
<td>No Hit</td>
</tr>
<tr>
<td>AGCGATTTGTCTGGTTAATTCCGATAACGAACGAGACTCTGGCATGCTAACTAGTTACGCGACC</td>
<td>40547</td>
<td>0.1475011993771533</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCGAAGGCCCGCGGCGGGTGTTGACGCGATGTGATTTCTGCCCAGTGCTCTGAATGTCAAAGT</td>
<td>35565</td>
<td>0.12937776298735928</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTA</td>
<td>34922</td>
<td>0.12703866832685395</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTTTGACTGGGGCGGTACACCTGTCAAACGGTAACGCAGGTGTCCTAAGGCGAGCTCAGGGAGG</td>
<td>30518</td>
<td>0.11101787068320626</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCTTGAGTCCA</td>
<td>27995</td>
<td>0.10183974342277867</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTGAT</td>
<td>2330865</td>
<td>1.9672527</td>
<td>5.142246</td>
<td>36</td>
</tr>
<tr>
<td>ATTGA</td>
<td>2209310</td>
<td>1.9152652</td>
<td>5.2375884</td>
<td>27</td>
</tr>
<tr>
<td>TTCTT</td>
<td>2217060</td>
<td>1.8498588</td>
<td>5.6337814</td>
<td>6</td>
</tr>
<tr>
<td>TCAAA</td>
<td>1969695</td>
<td>1.7809339</td>
<td>9.077659</td>
<td>24</td>
</tr>
<tr>
<td>AACGA</td>
<td>2405750</td>
<td>1.7746315</td>
<td>5.6926885</td>
<td>45</td>
</tr>
<tr>
<td>TTTCT</td>
<td>2118000</td>
<td>1.7672056</td>
<td>5.595233</td>
<td>34</td>
</tr>
<tr>
<td>GATTT</td>
<td>2007445</td>
<td>1.6942859</td>
<td>5.292393</td>
<td>23</td>
</tr>
<tr>
<td>CCATG</td>
<td>2813045</td>
<td>1.6294161</td>
<td>5.5281672</td>
<td>36</td>
</tr>
<tr>
<td>TGATT</td>
<td>1922850</td>
<td>1.6228876</td>
<td>5.0399528</td>
<td>31</td>
</tr>
<tr>
<td>AGCCA</td>
<td>2726620</td>
<td>1.6222179</td>
<td>5.547188</td>
<td>34</td>
</tr>
<tr>
<td>AGATT</td>
<td>1835985</td>
<td>1.5916274</td>
<td>7.05133</td>
<td>28</td>
</tr>
<tr>
<td>AAAGA</td>
<td>1734315</td>
<td>1.5862029</td>
<td>7.201095</td>
<td>26</td>
</tr>
<tr>
<td>TAATT</td>
<td>1478970</td>
<td>1.5715303</td>
<td>6.504291</td>
<td>35</td>
</tr>
<tr>
<td>ATGTC</td>
<td>2186855</td>
<td>1.5290412</td>
<td>6.6460013</td>
<td>42</td>
</tr>
<tr>
<td>CAAAG</td>
<td>2029200</td>
<td>1.4968647</td>
<td>5.9779425</td>
<td>25</td>
</tr>
<tr>
<td>AAGAT</td>
<td>1652255</td>
<td>1.4712231</td>
<td>7.278224</td>
<td>27</td>
</tr>
<tr>
<td>CTCAA</td>
<td>1986065</td>
<td>1.4483355</td>
<td>6.3002725</td>
<td>23</td>
</tr>
<tr>
<td>TGTCT</td>
<td>2123885</td>
<td>1.4457757</td>
<td>5.574911</td>
<td>43</td>
</tr>
<tr>
<td>GCCAT</td>
<td>2451525</td>
<td>1.4200108</td>
<td>5.282282</td>
<td>35</td>
</tr>
<tr>
<td>CATGC</td>
<td>2373120</td>
<td>1.3745958</td>
<td>5.4708233</td>
<td>37</td>
</tr>
<tr>
<td>AAGCC</td>
<td>2183290</td>
<td>1.2989606</td>
<td>5.367986</td>
<td>33</td>
</tr>
<tr>
<td>CATGT</td>
<td>1855345</td>
<td>1.2972506</td>
<td>5.9829535</td>
<td>41</td>
</tr>
<tr>
<td>TGCAT</td>
<td>1842810</td>
<td>1.2884861</td>
<td>5.5619674</td>
<td>39</td>
</tr>
<tr>
<td>TTGTC</td>
<td>1848225</td>
<td>1.2581278</td>
<td>5.496288</td>
<td>19</td>
</tr>
<tr>
<td>ATGCT</td>
<td>1781090</td>
<td>1.2453318</td>
<td>5.6106315</td>
<td>15</td>
</tr>
<tr>
<td>GCTTG</td>
<td>2239930</td>
<td>1.2439796</td>
<td>5.0146155</td>
<td>17</td>
</tr>
<tr>
<td>AAAAA</td>
<td>1079360</td>
<td>1.2428465</td>
<td>5.856587</td>
<td>59</td>
</tr>
<tr>
<td>AGCAT</td>
<td>1686485</td>
<td>1.211186</td>
<td>6.1515727</td>
<td>10</td>
</tr>
<tr>
<td>TGCTT</td>
<td>1713025</td>
<td>1.1660942</td>
<td>5.6312747</td>
<td>16</td>
</tr>
<tr>
<td>TTAAT</td>
<td>1091885</td>
<td>1.16022</td>
<td>6.4194975</td>
<td>34</td>
</tr>
<tr>
<td>TAGAT</td>
<td>1317385</td>
<td>1.1420496</td>
<td>5.756432</td>
<td>52</td>
</tr>
<tr>
<td>AAAGT</td>
<td>1249370</td>
<td>1.1124809</td>
<td>5.24029</td>
<td>58</td>
</tr>
<tr>
<td>ATTAA</td>
<td>1016845</td>
<td>1.1098069</td>
<td>8.15423</td>
<td>30</td>
</tr>
<tr>
<td>ATGCA</td>
<td>1530235</td>
<td>1.0989718</td>
<td>6.0315666</td>
<td>38</td>
</tr>
<tr>
<td>AGTAC</td>
<td>1516500</td>
<td>1.0891076</td>
<td>5.581919</td>
<td>49</td>
</tr>
<tr>
<td>TCTCA</td>
<td>1500960</td>
<td>1.0656525</td>
<td>5.7836432</td>
<td>22</td>
</tr>
<tr>
<td>TTAAG</td>
<td>1220910</td>
<td>1.0584148</td>
<td>6.7371635</td>
<td>31</td>
</tr>
<tr>
<td>GATTA</td>
<td>1220325</td>
<td>1.0579077</td>
<td>6.7421045</td>
<td>29</td>
</tr>
<tr>
<td>CTTGT</td>
<td>1537285</td>
<td>1.0464641</td>
<td>5.4256287</td>
<td>18</td>
</tr>
<tr>
<td>CAGTA</td>
<td>1428760</td>
<td>1.0260954</td>
<td>5.5343046</td>
<td>6</td>
</tr>
<tr>
<td>TCTTA</td>
<td>1125365</td>
<td>0.9644589</td>
<td>5.0942907</td>
<td>7</td>
</tr>
<tr>
<td>TTAAA</td>
<td>848465</td>
<td>0.9260333</td>
<td>5.31708</td>
<td>56</td>
</tr>
<tr>
<td>TCTAA</td>
<td>1028390</td>
<td>0.90526855</td>
<td>6.393191</td>
<td>45</td>
</tr>
<tr>
<td>TATGC</td>
<td>1234390</td>
<td>0.863081</td>
<td>5.2318063</td>
<td>14</td>
</tr>
<tr>
<td>CATAT</td>
<td>964420</td>
<td>0.8489571</td>
<td>6.7808647</td>
<td>12</td>
</tr>
<tr>
<td>AAGTA</td>
<td>915265</td>
<td>0.8149826</td>
<td>6.4613414</td>
<td>48</td>
</tr>
<tr>
<td>CTAAG</td>
<td>1062955</td>
<td>0.76338434</td>
<td>5.294291</td>
<td>46</td>
</tr>
<tr>
<td>TAAGC</td>
<td>1055790</td>
<td>0.7582387</td>
<td>5.345784</td>
<td>32</td>
</tr>
<tr>
<td>GCATA</td>
<td>1039015</td>
<td>0.7461913</td>
<td>5.4571724</td>
<td>11</td>
</tr>
<tr>
<td>ATATG</td>
<td>851605</td>
<td>0.7382619</td>
<td>6.5850215</td>
<td>13</td>
</tr>
<tr>
<td>TAAAA</td>
<td>618355</td>
<td>0.69320196</td>
<td>5.1291037</td>
<td>57</td>
</tr>
<tr>
<td>TAAGT</td>
<td>788485</td>
<td>0.6835428</td>
<td>6.175408</td>
<td>47</td>
</tr>
<tr>
<td>AGTAG</td>
<td>886855</td>
<td>0.6272399</td>
<td>5.1807737</td>
<td>7</td>
</tr>
<tr>
<td>GTCTA</td>
<td>852985</td>
<td>0.5964041</td>
<td>5.121474</td>
<td>44</td>
</tr>
<tr>
<td>TAGCA</td>
<td>797095</td>
<td>0.57245123</td>
<td>5.391029</td>
<td>9</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
