<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR315336_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315336_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>24612295</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>45</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/tick.png" alt="[OK]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTA</td>
<td>71304</td>
<td>0.2897088629890061</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGT</td>
<td>67711</td>
<td>0.27511046816235546</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGCC</td>
<td>2335255</td>
<td>2.1858397</td>
<td>6.581233</td>
<td>1</td>
</tr>
<tr>
<td>CCTGC</td>
<td>2158095</td>
<td>2.0200148</td>
<td>6.285264</td>
<td>1</td>
</tr>
<tr>
<td>TGCCA</td>
<td>2479430</td>
<td>1.9155303</td>
<td>5.3043036</td>
<td>2</td>
</tr>
<tr>
<td>AGCCA</td>
<td>2464745</td>
<td>1.8417588</td>
<td>5.7233443</td>
<td>34</td>
</tr>
<tr>
<td>GCCAG</td>
<td>1997155</td>
<td>1.7966027</td>
<td>5.9928293</td>
<td>3</td>
</tr>
<tr>
<td>CCATG</td>
<td>2310585</td>
<td>1.785086</td>
<td>5.561337</td>
<td>35</td>
</tr>
<tr>
<td>GCCAT</td>
<td>2221785</td>
<td>1.7164818</td>
<td>5.5448813</td>
<td>34</td>
</tr>
<tr>
<td>CCAGT</td>
<td>2053450</td>
<td>1.5864315</td>
<td>5.1001134</td>
<td>4</td>
</tr>
<tr>
<td>GCTTG</td>
<td>1474865</td>
<td>1.1705736</td>
<td>5.105438</td>
<td>16</td>
</tr>
<tr>
<td>GGCCG</td>
<td>933765</td>
<td>1.0112474</td>
<td>6.095978</td>
<td>57</td>
</tr>
<tr>
<td>GCCGG</td>
<td>795160</td>
<td>0.8611412</td>
<td>5.9562736</td>
<td>58</td>
</tr>
<tr>
<td>CGGCC</td>
<td>749135</td>
<td>0.8164831</td>
<td>6.042412</td>
<td>56</td>
</tr>
<tr>
<td>CGCAC</td>
<td>694410</td>
<td>0.62867105</td>
<td>5.003947</td>
<td>52</td>
</tr>
<tr>
<td>CCGGT</td>
<td>588390</td>
<td>0.54724526</td>
<td>5.012067</td>
<td>59</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
