<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR534291_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR534291_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Illumina 1.5</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>106005107</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>85</td>
</tr>
<tr>
<td>%GC</td>
<td>55</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GGGAACTTGAATTCGTATCCATCTGGCAGCTTGACGGTCAGGTTGGCCTG</td>
<td>224192</td>
<td>0.21149169728209416</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTTCAGGGGGGCCTCGGTCAGCAGCACTGGGTGCTCCTCCGGGGCCACGC</td>
<td>143966</td>
<td>0.13581043788767647</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>33523335</td>
<td>4.0847836</td>
<td>6.0489817</td>
<td>9</td>
</tr>
<tr>
<td>CCAGG</td>
<td>41508260</td>
<td>3.837806</td>
<td>5.5189166</td>
<td>2</td>
</tr>
<tr>
<td>CACCA</td>
<td>26293680</td>
<td>3.414421</td>
<td>4.6742835</td>
<td>9</td>
</tr>
<tr>
<td>TCCAG</td>
<td>28966975</td>
<td>3.307541</td>
<td>4.580463</td>
<td>80-81</td>
</tr>
<tr>
<td>ACCAG</td>
<td>27020120</td>
<td>3.2923737</td>
<td>4.9538827</td>
<td>7</td>
</tr>
<tr>
<td>CTTCT</td>
<td>24003405</td>
<td>3.1718225</td>
<td>4.8386784</td>
<td>6</td>
</tr>
<tr>
<td>TTCTT</td>
<td>20561705</td>
<td>3.1485093</td>
<td>5.9943647</td>
<td>6</td>
</tr>
<tr>
<td>CCAGC</td>
<td>31651175</td>
<td>3.1187618</td>
<td>3.8890817</td>
<td>8</td>
</tr>
<tr>
<td>CAGGG</td>
<td>35921710</td>
<td>3.11646</td>
<td>5.229976</td>
<td>4</td>
</tr>
<tr>
<td>CAGGA</td>
<td>27153405</td>
<td>3.1045752</td>
<td>5.657333</td>
<td>3</td>
</tr>
<tr>
<td>TCTTC</td>
<td>22126785</td>
<td>2.923845</td>
<td>5.308878</td>
<td>7</td>
</tr>
<tr>
<td>GCCAG</td>
<td>29739780</td>
<td>2.7497058</td>
<td>5.5423956</td>
<td>1</td>
</tr>
<tr>
<td>TTTTT</td>
<td>14517015</td>
<td>2.575923</td>
<td>11.225639</td>
<td>1</td>
</tr>
<tr>
<td>GCAGG</td>
<td>29007950</td>
<td>2.516643</td>
<td>6.0346394</td>
<td>2</td>
</tr>
<tr>
<td>GGCAG</td>
<td>26615535</td>
<td>2.3090842</td>
<td>9.4146385</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>25582260</td>
<td>2.0798106</td>
<td>6.600558</td>
<td>1</td>
</tr>
<tr>
<td>GGGCA</td>
<td>21546970</td>
<td>1.8693504</td>
<td>5.2340164</td>
<td>3</td>
</tr>
<tr>
<td>CTTGA</td>
<td>13406045</td>
<td>1.7738321</td>
<td>5.4585724</td>
<td>6</td>
</tr>
<tr>
<td>TGGGG</td>
<td>23180205</td>
<td>1.7683098</td>
<td>6.876271</td>
<td>2</td>
</tr>
<tr>
<td>GGGGG</td>
<td>28376920</td>
<td>1.7528815</td>
<td>5.262788</td>
<td>3</td>
</tr>
<tr>
<td>GGGAA</td>
<td>14660035</td>
<td>1.5727838</td>
<td>6.811254</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>19190450</td>
<td>1.5622331</td>
<td>6.419277</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>18668385</td>
<td>1.5197334</td>
<td>5.4257336</td>
<td>1</td>
</tr>
<tr>
<td>AACTT</td>
<td>8473895</td>
<td>1.4776388</td>
<td>5.1192503</td>
<td>4</td>
</tr>
<tr>
<td>GGGGT</td>
<td>19142140</td>
<td>1.4602643</td>
<td>5.8929257</td>
<td>1</td>
</tr>
<tr>
<td>TTGAA</td>
<td>8464265</td>
<td>1.3849388</td>
<td>7.297107</td>
<td>7</td>
</tr>
<tr>
<td>GTGGG</td>
<td>17175210</td>
<td>1.3102167</td>
<td>7.3180466</td>
<td>1</td>
</tr>
<tr>
<td>GTGGA</td>
<td>11463420</td>
<td>1.1524673</td>
<td>5.2134914</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>15320360</td>
<td>1.0085564</td>
<td>5.513326</td>
<td>1</td>
</tr>
<tr>
<td>GGGTT</td>
<td>10215520</td>
<td>0.9623989</td>
<td>5.1304913</td>
<td>3</td>
</tr>
<tr>
<td>TGAAT</td>
<td>5871020</td>
<td>0.96062726</td>
<td>5.156423</td>
<td>8</td>
</tr>
<tr>
<td>GCGGG</td>
<td>10049865</td>
<td>0.66159385</td>
<td>5.620799</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
