<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR317061_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317061_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>82822719</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>48</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GGTTGGTGCGCGGACACGCACTGCCTGCGTAACTAGAGGGAGCTGACGGATGACGCCCCCGCGC</td>
<td>180231</td>
<td>0.2176105809808055</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>143317</td>
<td>0.1730406846459605</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>135850</td>
<td>0.16402504245242178</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>132404</td>
<td>0.1598643483317663</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCGGGGGTGCGTCGGGTCTGCGAGAGCGCCAGCTATCCTGAGGGAAACTTCGGAGGGAACCAG</td>
<td>112424</td>
<td>0.13574053273957354</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGAT</td>
<td>86437</td>
<td>0.1043638762934117</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>GGGAG</td>
<td>13881150</td>
<td>2.4681242</td>
<td>7.018583</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>10361925</td>
<td>2.368181</td>
<td>7.462704</td>
<td>1</td>
</tr>
<tr>
<td>GGAGG</td>
<td>13259090</td>
<td>2.3575196</td>
<td>6.3541117</td>
<td>2</td>
</tr>
<tr>
<td>CCAGC</td>
<td>9892020</td>
<td>2.2954288</td>
<td>5.412323</td>
<td>30</td>
</tr>
<tr>
<td>GCCTG</td>
<td>9034450</td>
<td>2.2564304</td>
<td>5.1561384</td>
<td>23</td>
</tr>
<tr>
<td>CCCCC</td>
<td>7294590</td>
<td>2.2382658</td>
<td>5.231301</td>
<td>56</td>
</tr>
<tr>
<td>GGCTG</td>
<td>9528295</td>
<td>2.1776578</td>
<td>6.2656465</td>
<td>1</td>
</tr>
<tr>
<td>CTGCC</td>
<td>7229560</td>
<td>1.9732302</td>
<td>5.096648</td>
<td>21</td>
</tr>
<tr>
<td>GCTGG</td>
<td>8602635</td>
<td>1.9661015</td>
<td>7.0982184</td>
<td>1</td>
</tr>
<tr>
<td>CCTGC</td>
<td>6896260</td>
<td>1.8822596</td>
<td>5.05538</td>
<td>24</td>
</tr>
<tr>
<td>GGCAG</td>
<td>9455400</td>
<td>1.8372457</td>
<td>5.2576423</td>
<td>1</td>
</tr>
<tr>
<td>GCGCC</td>
<td>7139435</td>
<td>1.8343539</td>
<td>5.035777</td>
<td>27</td>
</tr>
<tr>
<td>GGGGG</td>
<td>9120355</td>
<td>1.7955347</td>
<td>9.505332</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>9988675</td>
<td>1.7760265</td>
<td>7.463928</td>
<td>1</td>
</tr>
<tr>
<td>GTGTG</td>
<td>7386350</td>
<td>1.6409874</td>
<td>5.2527103</td>
<td>1</td>
</tr>
<tr>
<td>CCCGC</td>
<td>5718505</td>
<td>1.6056377</td>
<td>5.0588284</td>
<td>58</td>
</tr>
<tr>
<td>GTGGG</td>
<td>7274740</td>
<td>1.5214097</td>
<td>8.487637</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>7100440</td>
<td>1.4849573</td>
<td>6.7787657</td>
<td>1</td>
</tr>
<tr>
<td>GTGGT</td>
<td>6527050</td>
<td>1.4500811</td>
<td>5.025036</td>
<td>1</td>
</tr>
<tr>
<td>GGGCA</td>
<td>7119095</td>
<td>1.3832865</td>
<td>5.2085605</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>6605215</td>
<td>1.381388</td>
<td>5.5599003</td>
<td>2</td>
</tr>
<tr>
<td>GGGGT</td>
<td>6520645</td>
<td>1.3637012</td>
<td>6.978595</td>
<td>3</td>
</tr>
<tr>
<td>GGGCT</td>
<td>5905790</td>
<td>1.3497473</td>
<td>5.863313</td>
<td>3</td>
</tr>
<tr>
<td>CGGGG</td>
<td>6147715</td>
<td>1.3226397</td>
<td>7.172792</td>
<td>1</td>
</tr>
<tr>
<td>GGGGC</td>
<td>5870350</td>
<td>1.2629663</td>
<td>5.5585065</td>
<td>2</td>
</tr>
<tr>
<td>GGCGG</td>
<td>5827620</td>
<td>1.2537735</td>
<td>6.6731644</td>
<td>1</td>
</tr>
<tr>
<td>GGTGC</td>
<td>5419870</td>
<td>1.2386919</td>
<td>6.3034253</td>
<td>5</td>
</tr>
<tr>
<td>GCGGG</td>
<td>5680955</td>
<td>1.2222195</td>
<td>5.689142</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
