<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317057_2.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317057_2.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>116300335</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>72</td>
</tr>
<tr>
<td>%GC</td>
<td>54</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/error.png" alt="[FAIL]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GCCCCTCTCCGGCCCCGGCCGGGGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTCGGGCCGATCG</td>
<td>375233</td>
<td>0.32264137502269447</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTCACCCGGCCCGGACACGGACAGGATTGACAGATTGATAGCTCTTTCTCGATTCCGTGGGTGGTGGTGCA</td>
<td>171454</td>
<td>0.14742347904672845</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTACAGTGAA</td>
<td>159955</td>
<td>0.13753614725185442</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>9514700</td>
<td>2.4058578</td>
<td>7.8462167</td>
<td>59</td>
</tr>
<tr>
<td>TGGTG</td>
<td>20280675</td>
<td>2.3441398</td>
<td>6.8486786</td>
<td>43</td>
</tr>
<tr>
<td>AACCT</td>
<td>11525120</td>
<td>1.8057753</td>
<td>9.059626</td>
<td>57</td>
</tr>
<tr>
<td>ATTGA</td>
<td>9278570</td>
<td>1.6799655</td>
<td>5.026025</td>
<td>27</td>
</tr>
<tr>
<td>ACCTC</td>
<td>12815340</td>
<td>1.6244979</td>
<td>7.4305983</td>
<td>58</td>
</tr>
<tr>
<td>TCAAA</td>
<td>8190505</td>
<td>1.5862008</td>
<td>7.122058</td>
<td>24</td>
</tr>
<tr>
<td>ATAAC</td>
<td>8017495</td>
<td>1.5526948</td>
<td>10.011119</td>
<td>55</td>
</tr>
<tr>
<td>AAAGA</td>
<td>7670485</td>
<td>1.5496678</td>
<td>5.4752274</td>
<td>26</td>
</tr>
<tr>
<td>CTTTG</td>
<td>11062815</td>
<td>1.5341138</td>
<td>8.21462</td>
<td>40</td>
</tr>
<tr>
<td>TTGGT</td>
<td>10971180</td>
<td>1.5025032</td>
<td>7.7190742</td>
<td>42</td>
</tr>
<tr>
<td>GGTGA</td>
<td>12273440</td>
<td>1.4985309</td>
<td>6.5666246</td>
<td>44</td>
</tr>
<tr>
<td>GGCGG</td>
<td>17589230</td>
<td>1.4664125</td>
<td>5.10367</td>
<td>25</td>
</tr>
<tr>
<td>TGACT</td>
<td>9773340</td>
<td>1.4316378</td>
<td>7.397423</td>
<td>46</td>
</tr>
<tr>
<td>GATCG</td>
<td>11412260</td>
<td>1.4109154</td>
<td>6.8036313</td>
<td>68</td>
</tr>
<tr>
<td>CCGGC</td>
<td>16295610</td>
<td>1.3929634</td>
<td>5.3894506</td>
<td>15</td>
</tr>
<tr>
<td>AGATT</td>
<td>7619340</td>
<td>1.3795475</td>
<td>5.0121565</td>
<td>28</td>
</tr>
<tr>
<td>AAGAT</td>
<td>7191205</td>
<td>1.3753691</td>
<td>5.2643175</td>
<td>27</td>
</tr>
<tr>
<td>GATAA</td>
<td>7095420</td>
<td>1.3570495</td>
<td>9.562125</td>
<td>54</td>
</tr>
<tr>
<td>CGATC</td>
<td>10729100</td>
<td>1.3431436</td>
<td>6.2143764</td>
<td>67</td>
</tr>
<tr>
<td>GCCGA</td>
<td>12573515</td>
<td>1.3284799</td>
<td>5.7360954</td>
<td>65</td>
</tr>
<tr>
<td>TTTGG</td>
<td>9671970</td>
<td>1.3245764</td>
<td>7.802797</td>
<td>41</td>
</tr>
<tr>
<td>ACTCT</td>
<td>8881655</td>
<td>1.3173887</td>
<td>7.5221252</td>
<td>48</td>
</tr>
<tr>
<td>CCTCT</td>
<td>10730885</td>
<td>1.2877345</td>
<td>6.463444</td>
<td>4</td>
</tr>
<tr>
<td>GCTTT</td>
<td>9206245</td>
<td>1.2766576</td>
<td>7.3043976</td>
<td>39</td>
</tr>
<tr>
<td>TCTCC</td>
<td>10623180</td>
<td>1.2748096</td>
<td>6.5805087</td>
<td>6</td>
</tr>
<tr>
<td>TAGAT</td>
<td>6995180</td>
<td>1.2665379</td>
<td>9.949265</td>
<td>52</td>
</tr>
<tr>
<td>GGGCG</td>
<td>15147425</td>
<td>1.2628394</td>
<td>5.1643505</td>
<td>24</td>
</tr>
<tr>
<td>GGCTT</td>
<td>10700590</td>
<td>1.2523876</td>
<td>6.3876743</td>
<td>38</td>
</tr>
<tr>
<td>GCCGG</td>
<td>14731160</td>
<td>1.243587</td>
<td>5.4114957</td>
<td>18</td>
</tr>
<tr>
<td>AGATA</td>
<td>6374835</td>
<td>1.2192324</td>
<td>9.371557</td>
<td>53</td>
</tr>
<tr>
<td>ATTAA</td>
<td>5375800</td>
<td>1.2182082</td>
<td>5.625463</td>
<td>30</td>
</tr>
<tr>
<td>CGGCC</td>
<td>14159015</td>
<td>1.2103254</td>
<td>5.149012</td>
<td>16</td>
</tr>
<tr>
<td>TTAAA</td>
<td>5322505</td>
<td>1.206131</td>
<td>5.5569215</td>
<td>56</td>
</tr>
<tr>
<td>CTCTC</td>
<td>9906135</td>
<td>1.1887623</td>
<td>6.4933133</td>
<td>5</td>
</tr>
<tr>
<td>GTGAC</td>
<td>9315175</td>
<td>1.1516492</td>
<td>6.1128383</td>
<td>45</td>
</tr>
<tr>
<td>CCCTC</td>
<td>10783230</td>
<td>1.1058818</td>
<td>5.4988832</td>
<td>3</td>
</tr>
<tr>
<td>GCCCC</td>
<td>12160705</td>
<td>1.0525863</td>
<td>5.133325</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>12471675</td>
<td>1.0268433</td>
<td>5.6283026</td>
<td>21</td>
</tr>
<tr>
<td>GACTC</td>
<td>8087680</td>
<td>1.0124723</td>
<td>6.29522</td>
<td>47</td>
</tr>
<tr>
<td>CCGAT</td>
<td>8079895</td>
<td>1.0114976</td>
<td>6.0532756</td>
<td>66</td>
</tr>
<tr>
<td>GGCCC</td>
<td>11724090</td>
<td>1.0021857</td>
<td>5.084048</td>
<td>11</td>
</tr>
<tr>
<td>CTAGA</td>
<td>6287615</td>
<td>0.97291374</td>
<td>7.957579</td>
<td>51</td>
</tr>
<tr>
<td>CGGCT</td>
<td>9688425</td>
<td>0.9690657</td>
<td>5.2843614</td>
<td>37</td>
</tr>
<tr>
<td>TAAAA</td>
<td>4045330</td>
<td>0.96834594</td>
<td>5.6477594</td>
<td>57</td>
</tr>
<tr>
<td>CTCGG</td>
<td>9527140</td>
<td>0.9529336</td>
<td>5.280537</td>
<td>60</td>
</tr>
<tr>
<td>CTCTA</td>
<td>6381340</td>
<td>0.9465246</td>
<td>7.303622</td>
<td>49</td>
</tr>
<tr>
<td>CCCCT</td>
<td>9144600</td>
<td>0.9378309</td>
<td>5.4465947</td>
<td>2</td>
</tr>
<tr>
<td>TCTAG</td>
<td>6263555</td>
<td>0.91751045</td>
<td>7.5301967</td>
<td>50</td>
</tr>
<tr>
<td>CCTCG</td>
<td>9000815</td>
<td>0.91161567</td>
<td>5.1656814</td>
<td>59</td>
</tr>
<tr>
<td>CATAT</td>
<td>4861750</td>
<td>0.89133686</td>
<td>5.002731</td>
<td>12</td>
</tr>
<tr>
<td>TAACC</td>
<td>5475075</td>
<td>0.857844</td>
<td>7.9404564</td>
<td>56</td>
</tr>
<tr>
<td>TCCGG</td>
<td>8305110</td>
<td>0.8307024</td>
<td>5.36642</td>
<td>8</td>
</tr>
<tr>
<td>CTCCG</td>
<td>6732215</td>
<td>0.6818485</td>
<td>5.3027387</td>
<td>7</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
