<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR534335_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR534335_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>142891462</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>85</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/tick.png" alt="[OK]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td>541805</td>
<td>0.3791724098952812</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td>502588</td>
<td>0.35172710319109196</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCG</td>
<td>406936</td>
<td>0.28478678453160483</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCAAACCCACTCCACCTTACTACCAGACAACCTTAGCCAAACCATTTAC</td>
<td>291392</td>
<td>0.2039254101830101</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCGT</td>
<td>274251</td>
<td>0.19192959198639875</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGA</td>
<td>217919</td>
<td>0.15250666271438948</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGGA</td>
<td>207281</td>
<td>0.14506185121123613</td>
<td>No Hit</td>
</tr>
<tr>
<td>CAAACCCACTCCACCTTACTACCAGACAACCTTAGCCAAACCATTTACCC</td>
<td>169997</td>
<td>0.11896931952449337</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCGTTC</td>
<td>163487</td>
<td>0.11441341400789923</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCTGG</td>
<td>27203095</td>
<td>3.0164275</td>
<td>6.9973893</td>
<td>1</td>
</tr>
<tr>
<td>CCCAG</td>
<td>26035235</td>
<td>3.0043404</td>
<td>5.8482647</td>
<td>20-24</td>
</tr>
<tr>
<td>GGCTG</td>
<td>25356695</td>
<td>2.7173657</td>
<td>5.2617464</td>
<td>35-39</td>
</tr>
<tr>
<td>GGAGG</td>
<td>24205220</td>
<td>2.5213861</td>
<td>5.04933</td>
<td>30-34</td>
</tr>
<tr>
<td>CCAGG</td>
<td>22549795</td>
<td>2.5148442</td>
<td>5.29145</td>
<td>60-64</td>
</tr>
<tr>
<td>GCCTG</td>
<td>21710985</td>
<td>2.4074323</td>
<td>5.2609267</td>
<td>10-14</td>
</tr>
<tr>
<td>CTGGG</td>
<td>22450295</td>
<td>2.4058995</td>
<td>5.2427707</td>
<td>70-74</td>
</tr>
<tr>
<td>CCAGC</td>
<td>20162295</td>
<td>2.3266313</td>
<td>5.4555497</td>
<td>20-24</td>
</tr>
<tr>
<td>CTGGC</td>
<td>20370360</td>
<td>2.2587767</td>
<td>6.3339996</td>
<td>2</td>
</tr>
<tr>
<td>GGTGG</td>
<td>18401190</td>
<td>1.9058214</td>
<td>11.707821</td>
<td>2</td>
</tr>
<tr>
<td>TGGCC</td>
<td>16654720</td>
<td>1.8467661</td>
<td>5.7178984</td>
<td>3</td>
</tr>
<tr>
<td>ACCCA</td>
<td>16134765</td>
<td>1.6452396</td>
<td>5.3243766</td>
<td>6</td>
</tr>
<tr>
<td>GTGGC</td>
<td>15062570</td>
<td>1.6141894</td>
<td>11.51956</td>
<td>3</td>
</tr>
<tr>
<td>GGCCT</td>
<td>14337780</td>
<td>1.5898513</td>
<td>5.471665</td>
<td>4</td>
</tr>
<tr>
<td>GTGCC</td>
<td>12740240</td>
<td>1.4127072</td>
<td>5.512756</td>
<td>9</td>
</tr>
<tr>
<td>AGTGC</td>
<td>13039960</td>
<td>1.2348396</td>
<td>5.66827</td>
<td>80-81</td>
</tr>
<tr>
<td>GGCGC</td>
<td>9640625</td>
<td>1.2167319</td>
<td>12.885574</td>
<td>5</td>
</tr>
<tr>
<td>TGGCG</td>
<td>8491500</td>
<td>0.90999675</td>
<td>10.807246</td>
<td>4</td>
</tr>
<tr>
<td>GCGCG</td>
<td>6681485</td>
<td>0.8432623</td>
<td>12.449055</td>
<td>6</td>
</tr>
<tr>
<td>CGTGC</td>
<td>7115785</td>
<td>0.78903705</td>
<td>11.325995</td>
<td>9</td>
</tr>
<tr>
<td>GCGTG</td>
<td>7075160</td>
<td>0.7582138</td>
<td>10.738888</td>
<td>8</td>
</tr>
<tr>
<td>CGGTG</td>
<td>6718900</td>
<td>0.720035</td>
<td>11.108977</td>
<td>1</td>
</tr>
<tr>
<td>GTGCG</td>
<td>6392235</td>
<td>0.6850278</td>
<td>7.508537</td>
<td>80-81</td>
</tr>
<tr>
<td>CGCGT</td>
<td>3805070</td>
<td>0.42192692</td>
<td>10.6432495</td>
<td>7</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
