<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sat 30 Aug 2014<br />
SRR317049_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317049_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>108082749</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>53</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATATCGTATGCCGT</td>
<td>372049</td>
<td>0.3442260707118025</td>
<td>Illumina Paired End PCR Primer 2 (98% over 50bp)</td>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGA</td>
<td>266583</td>
<td>0.24664713144925654</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAG</td>
<td>230732</td>
<td>0.21347717571469246</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGG</td>
<td>219696</td>
<td>0.2032664805740646</td>
<td>No Hit</td>
</tr>
<tr>
<td>GATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGT</td>
<td>202219</td>
<td>0.1870964625446379</td>
<td>Illumina Paired End PCR Primer 2 (100% over 50bp)</td>
</tr>
<tr>
<td>GGCAACAACACATCATCAGTAGGGTAAAACTAACCTGTCTCACGACGGTC</td>
<td>113931</td>
<td>0.10541090141961507</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CAGCA</td>
<td>22830235</td>
<td>2.754749</td>
<td>8.202742</td>
<td>17</td>
</tr>
<tr>
<td>CAGGA</td>
<td>22575005</td>
<td>2.4042006</td>
<td>7.3184404</td>
<td>20</td>
</tr>
<tr>
<td>AAAAA</td>
<td>17026435</td>
<td>2.383998</td>
<td>7.8003016</td>
<td>62</td>
</tr>
<tr>
<td>GTTCA</td>
<td>15621655</td>
<td>2.3564</td>
<td>9.230951</td>
<td>14</td>
</tr>
<tr>
<td>AGCAG</td>
<td>21661720</td>
<td>2.3069367</td>
<td>7.25601</td>
<td>18</td>
</tr>
<tr>
<td>TCAGC</td>
<td>17126585</td>
<td>2.3013744</td>
<td>8.841148</td>
<td>16</td>
</tr>
<tr>
<td>CTGCT</td>
<td>15250955</td>
<td>2.282225</td>
<td>8.629987</td>
<td>54</td>
</tr>
<tr>
<td>TTCAG</td>
<td>14965930</td>
<td>2.257489</td>
<td>9.142335</td>
<td>15</td>
</tr>
<tr>
<td>GCAGG</td>
<td>23757535</td>
<td>2.215414</td>
<td>6.6074085</td>
<td>19</td>
</tr>
<tr>
<td>AGGAA</td>
<td>19917515</td>
<td>2.1381516</td>
<td>6.9718804</td>
<td>21</td>
</tr>
<tr>
<td>CTTCT</td>
<td>11198640</td>
<td>2.13138</td>
<td>10.858683</td>
<td>51</td>
</tr>
<tr>
<td>TCTTC</td>
<td>11152680</td>
<td>2.1226327</td>
<td>10.543814</td>
<td>50</td>
</tr>
<tr>
<td>GGAAG</td>
<td>22484300</td>
<td>2.113457</td>
<td>7.4271383</td>
<td>5</td>
</tr>
<tr>
<td>GAAGA</td>
<td>19326110</td>
<td>2.074664</td>
<td>7.574153</td>
<td>6</td>
</tr>
<tr>
<td>TTCTG</td>
<td>11201285</td>
<td>1.8816321</td>
<td>9.120521</td>
<td>52</td>
</tr>
<tr>
<td>AAGAG</td>
<td>17502040</td>
<td>1.8788496</td>
<td>7.1390524</td>
<td>7</td>
</tr>
<tr>
<td>GGCTG</td>
<td>17939890</td>
<td>1.8630226</td>
<td>5.0679736</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>17381355</td>
<td>1.8050199</td>
<td>5.3266554</td>
<td>1</td>
</tr>
<tr>
<td>AGAGC</td>
<td>16620000</td>
<td>1.7700024</td>
<td>6.838169</td>
<td>8</td>
</tr>
<tr>
<td>ATGCC</td>
<td>13167145</td>
<td>1.769327</td>
<td>7.9951334</td>
<td>44</td>
</tr>
<tr>
<td>GGTTC</td>
<td>13260400</td>
<td>1.7514155</td>
<td>8.011198</td>
<td>13</td>
</tr>
<tr>
<td>TCTGC</td>
<td>10731035</td>
<td>1.6058427</td>
<td>8.050781</td>
<td>53</td>
</tr>
<tr>
<td>CTTGA</td>
<td>10473325</td>
<td>1.5798161</td>
<td>7.814142</td>
<td>57</td>
</tr>
<tr>
<td>TGCTT</td>
<td>9154550</td>
<td>1.537814</td>
<td>8.8092785</td>
<td>55</td>
</tr>
<tr>
<td>GAATG</td>
<td>12849670</td>
<td>1.5361733</td>
<td>6.9587545</td>
<td>23</td>
</tr>
<tr>
<td>ATCCA</td>
<td>9982210</td>
<td>1.5319077</td>
<td>5.487601</td>
<td>37</td>
</tr>
<tr>
<td>GGCAT</td>
<td>12899710</td>
<td>1.5299162</td>
<td>5.2137456</td>
<td>4</td>
</tr>
<tr>
<td>GGAAT</td>
<td>12651075</td>
<td>1.5124315</td>
<td>7.1866508</td>
<td>22</td>
</tr>
<tr>
<td>GCTTG</td>
<td>11363370</td>
<td>1.5008584</td>
<td>7.1773295</td>
<td>56</td>
</tr>
<tr>
<td>AATGC</td>
<td>10914495</td>
<td>1.4783617</td>
<td>7.7931194</td>
<td>24</td>
</tr>
<tr>
<td>GTCTT</td>
<td>8569935</td>
<td>1.4396083</td>
<td>9.244364</td>
<td>49</td>
</tr>
<tr>
<td>ATCGG</td>
<td>12054375</td>
<td>1.4296588</td>
<td>6.6417365</td>
<td>2</td>
</tr>
<tr>
<td>GATCG</td>
<td>11722890</td>
<td>1.3903444</td>
<td>6.510146</td>
<td>1</td>
</tr>
<tr>
<td>GAGCG</td>
<td>14487395</td>
<td>1.350964</td>
<td>5.6853433</td>
<td>9</td>
</tr>
<tr>
<td>TCGGA</td>
<td>11025295</td>
<td>1.307609</td>
<td>6.6757545</td>
<td>3</td>
</tr>
<tr>
<td>GTGGG</td>
<td>13969735</td>
<td>1.2804354</td>
<td>6.598219</td>
<td>1</td>
</tr>
<tr>
<td>GAAAA</td>
<td>10402835</td>
<td>1.2753943</td>
<td>5.9835033</td>
<td>60</td>
</tr>
<tr>
<td>TGAAA</td>
<td>9115345</td>
<td>1.2445453</td>
<td>6.6791677</td>
<td>59</td>
</tr>
<tr>
<td>CGGTT</td>
<td>9385455</td>
<td>1.2396182</td>
<td>7.174175</td>
<td>12</td>
</tr>
<tr>
<td>AGACC</td>
<td>10245300</td>
<td>1.2362216</td>
<td>6.9921274</td>
<td>31</td>
</tr>
<tr>
<td>TTGAA</td>
<td>8117425</td>
<td>1.2342434</td>
<td>7.1627707</td>
<td>58</td>
</tr>
<tr>
<td>GCGGT</td>
<td>11682680</td>
<td>1.2132236</td>
<td>5.832894</td>
<td>11</td>
</tr>
<tr>
<td>CGGAA</td>
<td>11246480</td>
<td>1.1977315</td>
<td>6.1181717</td>
<td>4</td>
</tr>
<tr>
<td>TGCCG</td>
<td>10145940</td>
<td>1.1937666</td>
<td>6.5940623</td>
<td>45</td>
</tr>
<tr>
<td>GGGGA</td>
<td>14452085</td>
<td>1.1894747</td>
<td>5.1537666</td>
<td>1</td>
</tr>
<tr>
<td>AATTA</td>
<td>6699610</td>
<td>1.1633794</td>
<td>5.3624754</td>
<td>24</td>
</tr>
<tr>
<td>GAGAC</td>
<td>10904015</td>
<td>1.1612594</td>
<td>6.1552863</td>
<td>30</td>
</tr>
<tr>
<td>AGCGG</td>
<td>12322130</td>
<td>1.1490508</td>
<td>5.272801</td>
<td>10</td>
</tr>
<tr>
<td>TTACC</td>
<td>6503185</td>
<td>1.1114165</td>
<td>5.1700554</td>
<td>26</td>
</tr>
<tr>
<td>TTATC</td>
<td>5422070</td>
<td>1.0402106</td>
<td>5.216994</td>
<td>35</td>
</tr>
<tr>
<td>GCCGA</td>
<td>9525535</td>
<td>1.0064025</td>
<td>5.8203506</td>
<td>27</td>
</tr>
<tr>
<td>CCGAG</td>
<td>8849980</td>
<td>0.93502784</td>
<td>5.8551717</td>
<td>28</td>
</tr>
<tr>
<td>TATTA</td>
<td>4758930</td>
<td>0.9202929</td>
<td>5.1815467</td>
<td>12</td>
</tr>
<tr>
<td>GATAT</td>
<td>5807235</td>
<td>0.88298213</td>
<td>5.177141</td>
<td>36</td>
</tr>
<tr>
<td>CGAGA</td>
<td>8074200</td>
<td>0.8598889</td>
<td>5.8291836</td>
<td>29</td>
</tr>
<tr>
<td>CCGTC</td>
<td>6366980</td>
<td>0.84876883</td>
<td>6.6347694</td>
<td>47</td>
</tr>
<tr>
<td>TCGTA</td>
<td>5493300</td>
<td>0.82861966</td>
<td>7.99524</td>
<td>40</td>
</tr>
<tr>
<td>ATATC</td>
<td>4715340</td>
<td>0.8123147</td>
<td>5.721623</td>
<td>37</td>
</tr>
<tr>
<td>ACCGA</td>
<td>6514060</td>
<td>0.78600156</td>
<td>6.4665484</td>
<td>33</td>
</tr>
<tr>
<td>GACCG</td>
<td>7137425</td>
<td>0.7540911</td>
<td>5.7256446</td>
<td>32</td>
</tr>
<tr>
<td>ATCGT</td>
<td>4780120</td>
<td>0.7210423</td>
<td>5.058116</td>
<td>39</td>
</tr>
<tr>
<td>TATGC</td>
<td>4737580</td>
<td>0.7146255</td>
<td>7.843442</td>
<td>43</td>
</tr>
<tr>
<td>GTATG</td>
<td>5246465</td>
<td>0.6984896</td>
<td>6.890716</td>
<td>42</td>
</tr>
<tr>
<td>GCCGT</td>
<td>5762355</td>
<td>0.6779961</td>
<td>6.0595875</td>
<td>46</td>
</tr>
<tr>
<td>CCGAT</td>
<td>5022020</td>
<td>0.67483085</td>
<td>6.8156424</td>
<td>34</td>
</tr>
<tr>
<td>CGTCT</td>
<td>4227180</td>
<td>0.6325752</td>
<td>7.444973</td>
<td>48</td>
</tr>
<tr>
<td>TATCG</td>
<td>3606165</td>
<td>0.5439607</td>
<td>5.045103</td>
<td>38</td>
</tr>
<tr>
<td>CGTAT</td>
<td>3458695</td>
<td>0.5217161</td>
<td>7.548035</td>
<td>41</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
