<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR534335_2.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR534335_2.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>157580617</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>101</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/tick.png" alt="[OK]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/tick.png" alt="[OK]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td>578432</td>
<td>0.3670705261929518</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td>524730</td>
<td>0.3329914617608078</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCG</td>
<td>426184</td>
<td>0.2704545826216685</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCAAACCCACTCCACCTTACTACCAGACAACCTTAGCCAAACCATTTAC</td>
<td>301038</td>
<td>0.19103745481590545</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCGT</td>
<td>286986</td>
<td>0.18212011443006343</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGA</td>
<td>231790</td>
<td>0.1470929638510046</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGGA</td>
<td>216503</td>
<td>0.13739189763421222</td>
<td>No Hit</td>
</tr>
<tr>
<td>CAAACCCACTCCACCTTACTACCAGACAACCTTAGCCAAACCATTTACCC</td>
<td>175439</td>
<td>0.11133285510615813</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGCCTTTCTATTAGCTCTTAGTAAGATTACACATGCAAGCATCCCCGTTC</td>
<td>171291</td>
<td>0.1087005516674681</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTTTT</td>
<td>79828785</td>
<td>3.960293</td>
<td>4.3470607</td>
<td>5</td>
</tr>
<tr>
<td>AAAAA</td>
<td>63549960</td>
<td>3.266135</td>
<td>4.3680277</td>
<td>2</td>
</tr>
<tr>
<td>CCTGG</td>
<td>36341155</td>
<td>2.9161127</td>
<td>6.5965705</td>
<td>1</td>
</tr>
<tr>
<td>CCCAG</td>
<td>33378855</td>
<td>2.8053074</td>
<td>5.62217</td>
<td>20-24</td>
</tr>
<tr>
<td>GGCTG</td>
<td>33530130</td>
<td>2.5870667</td>
<td>5.0654955</td>
<td>35-39</td>
</tr>
<tr>
<td>CCAGG</td>
<td>29625225</td>
<td>2.3940737</td>
<td>5.0426106</td>
<td>60-64</td>
</tr>
<tr>
<td>CTGGG</td>
<td>29979305</td>
<td>2.3130975</td>
<td>5.0133433</td>
<td>70-74</td>
</tr>
<tr>
<td>GCCTG</td>
<td>28680795</td>
<td>2.3014247</td>
<td>5.06239</td>
<td>10-14</td>
</tr>
<tr>
<td>CCAGC</td>
<td>26508755</td>
<td>2.2279136</td>
<td>5.2714195</td>
<td>20-24</td>
</tr>
<tr>
<td>CTGGC</td>
<td>27718000</td>
<td>2.224167</td>
<td>6.070202</td>
<td>2</td>
</tr>
<tr>
<td>GGTGG</td>
<td>24969515</td>
<td>1.8524619</td>
<td>11.076427</td>
<td>2</td>
</tr>
<tr>
<td>TGGCC</td>
<td>22123320</td>
<td>1.7752349</td>
<td>5.3780866</td>
<td>3</td>
</tr>
<tr>
<td>GTGGC</td>
<td>20306980</td>
<td>1.566815</td>
<td>10.870305</td>
<td>3</td>
</tr>
<tr>
<td>ACCCA</td>
<td>20338130</td>
<td>1.556673</td>
<td>5.062357</td>
<td>6</td>
</tr>
<tr>
<td>GGCCT</td>
<td>19025345</td>
<td>1.5266451</td>
<td>5.1389966</td>
<td>4</td>
</tr>
<tr>
<td>GTGCC</td>
<td>16654995</td>
<td>1.3364418</td>
<td>5.2893643</td>
<td>9</td>
</tr>
<tr>
<td>GGGTG</td>
<td>17537480</td>
<td>1.3010871</td>
<td>5.15688</td>
<td>95-97</td>
</tr>
<tr>
<td>GGCGC</td>
<td>14264215</td>
<td>1.2657453</td>
<td>11.926301</td>
<td>5</td>
</tr>
<tr>
<td>TGGCG</td>
<td>12012690</td>
<td>0.92685676</td>
<td>10.161211</td>
<td>4</td>
</tr>
<tr>
<td>GCGCG</td>
<td>9756430</td>
<td>0.86574376</td>
<td>11.4915495</td>
<td>6</td>
</tr>
<tr>
<td>CGTGC</td>
<td>9482175</td>
<td>0.76087534</td>
<td>10.856637</td>
<td>9</td>
</tr>
<tr>
<td>GCGTG</td>
<td>9373665</td>
<td>0.72323895</td>
<td>10.095415</td>
<td>8</td>
</tr>
<tr>
<td>CGGTG</td>
<td>9310740</td>
<td>0.7183839</td>
<td>10.553244</td>
<td>1</td>
</tr>
<tr>
<td>CGCGT</td>
<td>4928300</td>
<td>0.3954601</td>
<td>10.009738</td>
<td>7</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
