<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR315306_2.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR315306_2.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>40441408</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/warning.png" alt="[WARN]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td>
<td>261767</td>
<td>0.647274694293532</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATTCGGCTGA</td>
<td>70968</td>
<td>0.17548350443189317</td>
<td>No Hit</td>
</tr>
<tr>
<td>GGGCGATCTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATT</td>
<td>65387</td>
<td>0.16168329253026997</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAG</td>
<td>62977</td>
<td>0.15572405392018002</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td>58970</td>
<td>0.14581589246348692</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGATCTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATTCGG</td>
<td>57693</td>
<td>0.14265823781407413</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCGATCTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATTCG</td>
<td>48713</td>
<td>0.12045327403041953</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td>40987</td>
<td>0.10134909249450465</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCCCC</td>
<td>6481585</td>
<td>3.596161</td>
<td>6.6026087</td>
<td>72</td>
</tr>
<tr>
<td>AAAAA</td>
<td>9632465</td>
<td>3.4533424</td>
<td>5.644232</td>
<td>2</td>
</tr>
<tr>
<td>TTTTT</td>
<td>16877465</td>
<td>3.227814</td>
<td>4.100345</td>
<td>3</td>
</tr>
<tr>
<td>CCCAG</td>
<td>5926655</td>
<td>2.8965416</td>
<td>8.210781</td>
<td>21</td>
</tr>
<tr>
<td>CCAGG</td>
<td>5867995</td>
<td>2.7567868</td>
<td>6.3732715</td>
<td>62</td>
</tr>
<tr>
<td>GGCTG</td>
<td>6850960</td>
<td>2.7285256</td>
<td>6.382857</td>
<td>36</td>
</tr>
<tr>
<td>GGGGG</td>
<td>5810390</td>
<td>2.6459394</td>
<td>7.601408</td>
<td>71</td>
</tr>
<tr>
<td>CTGGG</td>
<td>6299680</td>
<td>2.5089676</td>
<td>5.770247</td>
<td>71</td>
</tr>
<tr>
<td>GGAGG</td>
<td>5471435</td>
<td>2.3752065</td>
<td>6.610391</td>
<td>33</td>
</tr>
<tr>
<td>CTGGC</td>
<td>5216375</td>
<td>2.1612353</td>
<td>5.676573</td>
<td>56</td>
</tr>
<tr>
<td>CCAGC</td>
<td>4297010</td>
<td>2.1000829</td>
<td>7.3575063</td>
<td>22</td>
</tr>
<tr>
<td>TCCCA</td>
<td>5024075</td>
<td>2.064294</td>
<td>6.2359962</td>
<td>20</td>
</tr>
<tr>
<td>GCCTG</td>
<td>4914475</td>
<td>2.0361528</td>
<td>6.6891913</td>
<td>12</td>
</tr>
<tr>
<td>CAGGA</td>
<td>4707440</td>
<td>2.0265956</td>
<td>5.86735</td>
<td>63</td>
</tr>
<tr>
<td>GGGAG</td>
<td>4603935</td>
<td>1.9986159</td>
<td>6.4009113</td>
<td>32</td>
</tr>
<tr>
<td>TGCCT</td>
<td>5631485</td>
<td>1.9615597</td>
<td>5.657884</td>
<td>11</td>
</tr>
<tr>
<td>CCCCA</td>
<td>3788850</td>
<td>1.9263456</td>
<td>5.0597</td>
<td>20</td>
</tr>
<tr>
<td>GAGGC</td>
<td>4237365</td>
<td>1.9136065</td>
<td>6.621074</td>
<td>34</td>
</tr>
<tr>
<td>AGGAG</td>
<td>4562005</td>
<td>1.8879105</td>
<td>5.3452682</td>
<td>64</td>
</tr>
<tr>
<td>CTGAG</td>
<td>4907825</td>
<td>1.863336</td>
<td>5.515844</td>
<td>38</td>
</tr>
<tr>
<td>AGGCT</td>
<td>4844255</td>
<td>1.8392006</td>
<td>5.6201067</td>
<td>35</td>
</tr>
<tr>
<td>GAGGA</td>
<td>4248970</td>
<td>1.7583661</td>
<td>5.4542627</td>
<td>47</td>
</tr>
<tr>
<td>GGTGG</td>
<td>4503800</td>
<td>1.7242453</td>
<td>5.748865</td>
<td>2</td>
</tr>
<tr>
<td>CAGCT</td>
<td>4256485</td>
<td>1.6811632</td>
<td>5.6535044</td>
<td>23</td>
</tr>
<tr>
<td>GCTGA</td>
<td>4339655</td>
<td>1.647621</td>
<td>5.422414</td>
<td>37</td>
</tr>
<tr>
<td>TGAGG</td>
<td>4440370</td>
<td>1.6205583</td>
<td>5.178741</td>
<td>39</td>
</tr>
<tr>
<td>CCTGT</td>
<td>4421595</td>
<td>1.5401307</td>
<td>5.158217</td>
<td>13</td>
</tr>
<tr>
<td>GTGGC</td>
<td>3742710</td>
<td>1.4906057</td>
<td>5.5077395</td>
<td>3</td>
</tr>
<tr>
<td>GTGCC</td>
<td>3028765</td>
<td>1.2548703</td>
<td>5.2247596</td>
<td>10</td>
</tr>
<tr>
<td>GTCCC</td>
<td>2781250</td>
<td>1.1987536</td>
<td>5.2809043</td>
<td>19</td>
</tr>
<tr>
<td>GGCGC</td>
<td>2248975</td>
<td>1.108339</td>
<td>5.909744</td>
<td>5</td>
</tr>
<tr>
<td>GCGCG</td>
<td>1804650</td>
<td>0.8893668</td>
<td>5.58214</td>
<td>6</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
