<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR317069_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317069_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>125001968</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>76</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGA</td>
<td>299172</td>
<td>0.2393338319281501</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAG</td>
<td>289446</td>
<td>0.23155315442713673</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGC</td>
<td>270780</td>
<td>0.2166205895254385</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGG</td>
<td>246008</td>
<td>0.19680330152882075</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCG</td>
<td>200947</td>
<td>0.16075506907219253</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CCCCC</td>
<td>11352045</td>
<td>2.6520646</td>
<td>5.209139</td>
<td>32</td>
</tr>
<tr>
<td>CTGGG</td>
<td>16520920</td>
<td>2.3862498</td>
<td>8.477853</td>
<td>1</td>
</tr>
<tr>
<td>GGAGG</td>
<td>21613465</td>
<td>2.3485537</td>
<td>6.650235</td>
<td>2</td>
</tr>
<tr>
<td>GGGGG</td>
<td>17927410</td>
<td>2.338737</td>
<td>10.322982</td>
<td>1</td>
</tr>
<tr>
<td>GGCTG</td>
<td>15251015</td>
<td>2.2028272</td>
<td>6.531913</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>20265190</td>
<td>2.2020476</td>
<td>7.2457404</td>
<td>2</td>
</tr>
<tr>
<td>GCTGG</td>
<td>14425330</td>
<td>2.083567</td>
<td>10.312471</td>
<td>1</td>
</tr>
<tr>
<td>CACAG</td>
<td>16924270</td>
<td>1.9338231</td>
<td>5.0208583</td>
<td>30</td>
</tr>
<tr>
<td>GGCAG</td>
<td>15605255</td>
<td>1.9052684</td>
<td>5.791824</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>17192635</td>
<td>1.8681792</td>
<td>8.544759</td>
<td>1</td>
</tr>
<tr>
<td>TTCAT</td>
<td>15479455</td>
<td>1.8082845</td>
<td>5.213792</td>
<td>42</td>
</tr>
<tr>
<td>GTGGG</td>
<td>13675555</td>
<td>1.7579942</td>
<td>10.166329</td>
<td>1</td>
</tr>
<tr>
<td>TCACT</td>
<td>12763860</td>
<td>1.7001758</td>
<td>5.158293</td>
<td>52</td>
</tr>
<tr>
<td>TGGGG</td>
<td>13221725</td>
<td>1.6996542</td>
<td>7.6153426</td>
<td>1</td>
</tr>
<tr>
<td>CTGGA</td>
<td>14073855</td>
<td>1.693195</td>
<td>5.5988564</td>
<td>3</td>
</tr>
<tr>
<td>GTGTG</td>
<td>12893215</td>
<td>1.6332109</td>
<td>6.013739</td>
<td>1</td>
</tr>
<tr>
<td>GGCAT</td>
<td>13016545</td>
<td>1.5659924</td>
<td>5.2252374</td>
<td>4</td>
</tr>
<tr>
<td>CATCC</td>
<td>10107525</td>
<td>1.5351727</td>
<td>5.2604437</td>
<td>36</td>
</tr>
<tr>
<td>GCTCT</td>
<td>9387605</td>
<td>1.5012544</td>
<td>5.5231724</td>
<td>17</td>
</tr>
<tr>
<td>GGGCA</td>
<td>12291430</td>
<td>1.5006787</td>
<td>5.977906</td>
<td>1</td>
</tr>
<tr>
<td>GGGTG</td>
<td>11380565</td>
<td>1.4629729</td>
<td>6.195215</td>
<td>2</td>
</tr>
<tr>
<td>ATCCA</td>
<td>12982280</td>
<td>1.4617273</td>
<td>7.094661</td>
<td>37</td>
</tr>
<tr>
<td>GGGGC</td>
<td>9812685</td>
<td>1.4383378</td>
<td>6.3534093</td>
<td>2</td>
</tr>
<tr>
<td>GGGGT</td>
<td>11149260</td>
<td>1.4332386</td>
<td>7.424351</td>
<td>3</td>
</tr>
<tr>
<td>GGGCT</td>
<td>9890365</td>
<td>1.4285452</td>
<td>6.0945787</td>
<td>3</td>
</tr>
<tr>
<td>ACCAC</td>
<td>10881690</td>
<td>1.3970513</td>
<td>5.186674</td>
<td>28</td>
</tr>
<tr>
<td>CGGGG</td>
<td>9213935</td>
<td>1.3505734</td>
<td>8.4424925</td>
<td>1</td>
</tr>
<tr>
<td>GTGGT</td>
<td>10462430</td>
<td>1.3252982</td>
<td>5.318909</td>
<td>1</td>
</tr>
<tr>
<td>GGCGG</td>
<td>8036275</td>
<td>1.1779526</td>
<td>6.241812</td>
<td>1</td>
</tr>
<tr>
<td>TTACC</td>
<td>8559025</td>
<td>1.1400819</td>
<td>5.0226426</td>
<td>26</td>
</tr>
<tr>
<td>GCGGG</td>
<td>7616430</td>
<td>1.116412</td>
<td>5.7115946</td>
<td>1</td>
</tr>
<tr>
<td>CCCGT</td>
<td>4771715</td>
<td>0.8701109</td>
<td>6.6654234</td>
<td>1</td>
</tr>
<tr>
<td>CCGTC</td>
<td>4726945</td>
<td>0.86194724</td>
<td>6.311829</td>
<td>2</td>
</tr>
<tr>
<td>GTCGG</td>
<td>5434355</td>
<td>0.7849277</td>
<td>5.0758195</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
