<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR534291_1.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR534291_1.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Illumina 1.5</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>114006059</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>101</td>
</tr>
<tr>
<td>%GC</td>
<td>55</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GGGAACTTGAATTCGTATCCATCTGGCAGCTTGACGGTCAGGTTGGCCTG</td>
<td>254885</td>
<td>0.22357145070684356</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTTCAGGGGGGCCTCGGTCAGCAGCACTGGGTGCTCCTCCGGGGCCACGC</td>
<td>147882</td>
<td>0.12971415843784231</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTTTT</td>
<td>29485345</td>
<td>4.146294</td>
<td>14.890982</td>
<td>1</td>
</tr>
<tr>
<td>CAGCA</td>
<td>42336250</td>
<td>3.9810708</td>
<td>5.9118695</td>
<td>9</td>
</tr>
<tr>
<td>CCAGG</td>
<td>54812775</td>
<td>3.8942485</td>
<td>5.416262</td>
<td>2</td>
</tr>
<tr>
<td>AAAAA</td>
<td>17773765</td>
<td>3.4841588</td>
<td>4.229874</td>
<td>10-14</td>
</tr>
<tr>
<td>CACCA</td>
<td>34623615</td>
<td>3.4337044</td>
<td>4.5120735</td>
<td>9</td>
</tr>
<tr>
<td>TCCAG</td>
<td>37835235</td>
<td>3.3291323</td>
<td>4.348823</td>
<td>2</td>
</tr>
<tr>
<td>ACCAG</td>
<td>35023135</td>
<td>3.2933853</td>
<td>4.80377</td>
<td>7</td>
</tr>
<tr>
<td>CTTCT</td>
<td>30917440</td>
<td>3.1526659</td>
<td>4.7753034</td>
<td>6</td>
</tr>
<tr>
<td>CAGGA</td>
<td>35151980</td>
<td>3.1342552</td>
<td>5.6233635</td>
<td>3</td>
</tr>
<tr>
<td>CAGGG</td>
<td>46457840</td>
<td>3.1296654</td>
<td>5.173946</td>
<td>4</td>
</tr>
<tr>
<td>TTCTT</td>
<td>26009645</td>
<td>3.1145735</td>
<td>5.9971013</td>
<td>6</td>
</tr>
<tr>
<td>CCAGC</td>
<td>40871535</td>
<td>3.0624273</td>
<td>3.7612045</td>
<td>8</td>
</tr>
<tr>
<td>TCTTC</td>
<td>28398920</td>
<td>2.8958511</td>
<td>5.222136</td>
<td>7</td>
</tr>
<tr>
<td>GCCAG</td>
<td>37451205</td>
<td>2.660772</td>
<td>5.443475</td>
<td>1</td>
</tr>
<tr>
<td>GCAGG</td>
<td>37755265</td>
<td>2.54341</td>
<td>5.996319</td>
<td>2</td>
</tr>
<tr>
<td>GGCAG</td>
<td>34185875</td>
<td>2.3029556</td>
<td>9.355848</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>32767110</td>
<td>2.065495</td>
<td>6.5516806</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>40330400</td>
<td>1.9463583</td>
<td>5.7194214</td>
<td>3</td>
</tr>
<tr>
<td>GGGCA</td>
<td>27573540</td>
<td>1.857511</td>
<td>5.1843123</td>
<td>3</td>
</tr>
<tr>
<td>TGGGG</td>
<td>29550795</td>
<td>1.7662499</td>
<td>6.983617</td>
<td>2</td>
</tr>
<tr>
<td>CTTGA</td>
<td>16866410</td>
<td>1.7427979</td>
<td>5.6429076</td>
<td>6</td>
</tr>
<tr>
<td>GGGAA</td>
<td>18853660</td>
<td>1.5939593</td>
<td>6.992156</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>23931260</td>
<td>1.5286268</td>
<td>6.5371633</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>23560980</td>
<td>1.504975</td>
<td>5.441934</td>
<td>1</td>
</tr>
<tr>
<td>AACTT</td>
<td>10680415</td>
<td>1.4606901</td>
<td>5.395345</td>
<td>4</td>
</tr>
<tr>
<td>GGGGT</td>
<td>24260975</td>
<td>1.4500775</td>
<td>5.9738326</td>
<td>1</td>
</tr>
<tr>
<td>TTGAA</td>
<td>10434835</td>
<td>1.3531706</td>
<td>7.565445</td>
<td>7</td>
</tr>
<tr>
<td>GTGGG</td>
<td>21532470</td>
<td>1.2869949</td>
<td>7.3607073</td>
<td>1</td>
</tr>
<tr>
<td>GTGGA</td>
<td>14216600</td>
<td>1.124668</td>
<td>5.17798</td>
<td>1</td>
</tr>
<tr>
<td>CGGGG</td>
<td>20798280</td>
<td>1.0585725</td>
<td>5.543157</td>
<td>1</td>
</tr>
<tr>
<td>TGAAT</td>
<td>7406980</td>
<td>0.9605238</td>
<td>5.4135933</td>
<td>8</td>
</tr>
<tr>
<td>GGGTT</td>
<td>12733270</td>
<td>0.94257444</td>
<td>5.1030025</td>
<td>3</td>
</tr>
<tr>
<td>GGCGG</td>
<td>16686235</td>
<td>0.84928143</td>
<td>5.080562</td>
<td>1</td>
</tr>
<tr>
<td>GCGGG</td>
<td>14097415</td>
<td>0.7175178</td>
<td>5.6427813</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
