<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR317068_pe_2.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317068_pe_2.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>91876666</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/warning.png" alt="[WARN]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/error.png" alt="[FAIL]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/error.png" alt="[FAIL]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGT</td>
<td>2679383</td>
<td>2.9162823561751794</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTA</td>
<td>906736</td>
<td>0.9869056415259996</td>
<td>No Hit</td>
</tr>
<tr>
<td>TCCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGG</td>
<td>905619</td>
<td>0.9856898812588607</td>
<td>No Hit</td>
</tr>
<tr>
<td>TGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTAC</td>
<td>148159</td>
<td>0.16125857244319247</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTACGCACGGCCGGTACA</td>
<td>118042</td>
<td>0.12847875868721662</td>
<td>No Hit</td>
</tr>
<tr>
<td>CAGACGTGGCGACCCGCTGAATTTAAGCATATTAGTCAGCGGAGGAAAAGAAACTAACCAGGAT</td>
<td>111488</td>
<td>0.12134528259873949</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGCC</td>
<td>13748935</td>
<td>2.9673543</td>
<td>37.712772</td>
<td>2</td>
</tr>
<tr>
<td>AGCCA</td>
<td>13061030</td>
<td>2.9486785</td>
<td>39.793995</td>
<td>34</td>
</tr>
<tr>
<td>GCCAG</td>
<td>11858050</td>
<td>2.8787258</td>
<td>41.806194</td>
<td>4</td>
</tr>
<tr>
<td>TGTCT</td>
<td>19185180</td>
<td>2.8489478</td>
<td>26.094915</td>
<td>20</td>
</tr>
<tr>
<td>AAAGA</td>
<td>14113040</td>
<td>2.7645814</td>
<td>34.29045</td>
<td>26</td>
</tr>
<tr>
<td>CAAAG</td>
<td>12745500</td>
<td>2.6803179</td>
<td>36.53604</td>
<td>25</td>
</tr>
<tr>
<td>CCATG</td>
<td>12988885</td>
<td>2.611269</td>
<td>35.150276</td>
<td>36</td>
</tr>
<tr>
<td>AAGCC</td>
<td>11521830</td>
<td>2.6011863</td>
<td>39.378345</td>
<td>33</td>
</tr>
<tr>
<td>GGCCG</td>
<td>9845970</td>
<td>2.5702927</td>
<td>44.65533</td>
<td>58</td>
</tr>
<tr>
<td>CCTGC</td>
<td>11823115</td>
<td>2.5517154</td>
<td>37.795387</td>
<td>1</td>
</tr>
<tr>
<td>GCCGG</td>
<td>9564725</td>
<td>2.4968736</td>
<td>44.594402</td>
<td>59</td>
</tr>
<tr>
<td>TGCCA</td>
<td>11911450</td>
<td>2.3946629</td>
<td>34.702885</td>
<td>3</td>
</tr>
<tr>
<td>AGCAT</td>
<td>12744395</td>
<td>2.386593</td>
<td>32.851604</td>
<td>10</td>
</tr>
<tr>
<td>CGGCC</td>
<td>9108500</td>
<td>2.3738565</td>
<td>44.68649</td>
<td>57</td>
</tr>
<tr>
<td>CCAGT</td>
<td>11729665</td>
<td>2.358117</td>
<td>34.546135</td>
<td>5</td>
</tr>
<tr>
<td>TCTCA</td>
<td>14034155</td>
<td>2.336462</td>
<td>29.334509</td>
<td>22</td>
</tr>
<tr>
<td>TCAAA</td>
<td>13259545</td>
<td>2.3091426</td>
<td>30.401064</td>
<td>24</td>
</tr>
<tr>
<td>CTCAA</td>
<td>12055605</td>
<td>2.2538846</td>
<td>32.548878</td>
<td>23</td>
</tr>
<tr>
<td>GGAGG</td>
<td>9250685</td>
<td>2.2531695</td>
<td>5.524046</td>
<td>46</td>
</tr>
<tr>
<td>GCATG</td>
<td>10977020</td>
<td>2.21045</td>
<td>34.81582</td>
<td>40</td>
</tr>
<tr>
<td>GCCAT</td>
<td>10969675</td>
<td>2.2053297</td>
<td>34.79335</td>
<td>35</td>
</tr>
<tr>
<td>CATGT</td>
<td>13103870</td>
<td>2.1851866</td>
<td>29.220507</td>
<td>41</td>
</tr>
<tr>
<td>TCCTG</td>
<td>12054410</td>
<td>2.1580198</td>
<td>11.251237</td>
<td>1</td>
</tr>
<tr>
<td>CATGC</td>
<td>10593730</td>
<td>2.1297503</td>
<td>34.619835</td>
<td>37</td>
</tr>
<tr>
<td>CAGTA</td>
<td>11023430</td>
<td>2.0643146</td>
<td>32.307</td>
<td>6</td>
</tr>
<tr>
<td>TGCTT</td>
<td>13741720</td>
<td>2.0406086</td>
<td>26.012186</td>
<td>16</td>
</tr>
<tr>
<td>AAGAT</td>
<td>11555755</td>
<td>2.0157511</td>
<td>30.290602</td>
<td>27</td>
</tr>
<tr>
<td>TTAAG</td>
<td>12935050</td>
<td>2.0092611</td>
<td>26.882648</td>
<td>31</td>
</tr>
<tr>
<td>ATGCA</td>
<td>10637420</td>
<td>1.9920281</td>
<td>32.210762</td>
<td>38</td>
</tr>
<tr>
<td>TAAGC</td>
<td>10614470</td>
<td>1.9877305</td>
<td>32.009975</td>
<td>32</td>
</tr>
<tr>
<td>GTCTC</td>
<td>11085075</td>
<td>1.9844863</td>
<td>31.014708</td>
<td>21</td>
</tr>
<tr>
<td>GCATA</td>
<td>10584600</td>
<td>1.9821367</td>
<td>32.166603</td>
<td>11</td>
</tr>
<tr>
<td>GTAGC</td>
<td>9810890</td>
<td>1.9756252</td>
<td>34.690342</td>
<td>8</td>
</tr>
<tr>
<td>TGCAT</td>
<td>11809495</td>
<td>1.9693378</td>
<td>28.897976</td>
<td>39</td>
</tr>
<tr>
<td>CATAT</td>
<td>12639070</td>
<td>1.9600493</td>
<td>27.039738</td>
<td>12</td>
</tr>
<tr>
<td>AGTAG</td>
<td>10234090</td>
<td>1.9196625</td>
<td>32.410873</td>
<td>7</td>
</tr>
<tr>
<td>GCACG</td>
<td>7897475</td>
<td>1.9172347</td>
<td>40.98999</td>
<td>54</td>
</tr>
<tr>
<td>GCTTG</td>
<td>10299370</td>
<td>1.8468711</td>
<td>30.78666</td>
<td>17</td>
</tr>
<tr>
<td>AGATT</td>
<td>11733790</td>
<td>1.8226641</td>
<td>26.896631</td>
<td>28</td>
</tr>
<tr>
<td>ATGCT</td>
<td>10894310</td>
<td>1.8167226</td>
<td>28.767803</td>
<td>15</td>
</tr>
<tr>
<td>ATGTC</td>
<td>10881780</td>
<td>1.8146331</td>
<td>29.025246</td>
<td>42</td>
</tr>
<tr>
<td>CGCAC</td>
<td>7484200</td>
<td>1.813911</td>
<td>40.85161</td>
<td>53</td>
</tr>
<tr>
<td>CACGG</td>
<td>7469925</td>
<td>1.8134403</td>
<td>41.01774</td>
<td>55</td>
</tr>
<tr>
<td>TAGCA</td>
<td>9580050</td>
<td>1.7940185</td>
<td>32.418514</td>
<td>9</td>
</tr>
<tr>
<td>AAGTA</td>
<td>10270670</td>
<td>1.7915848</td>
<td>29.986767</td>
<td>48</td>
</tr>
<tr>
<td>CTAAG</td>
<td>9438830</td>
<td>1.7675728</td>
<td>32.053684</td>
<td>46</td>
</tr>
<tr>
<td>AGTAC</td>
<td>9371230</td>
<td>1.7549138</td>
<td>31.889639</td>
<td>49</td>
</tr>
<tr>
<td>ATTAA</td>
<td>11924285</td>
<td>1.7225176</td>
<td>25.22138</td>
<td>30</td>
</tr>
<tr>
<td>ACGGC</td>
<td>6996425</td>
<td>1.6984909</td>
<td>40.932964</td>
<td>56</td>
</tr>
<tr>
<td>CTTGT</td>
<td>11406260</td>
<td>1.693799</td>
<td>25.70698</td>
<td>18</td>
</tr>
<tr>
<td>GATTA</td>
<td>10629585</td>
<td>1.6511428</td>
<td>26.92599</td>
<td>29</td>
</tr>
<tr>
<td>TTGTC</td>
<td>10914580</td>
<td>1.6207861</td>
<td>25.649292</td>
<td>19</td>
</tr>
<tr>
<td>TATGC</td>
<td>9636105</td>
<td>1.6069059</td>
<td>28.490602</td>
<td>14</td>
</tr>
<tr>
<td>TCTAA</td>
<td>10327965</td>
<td>1.6016463</td>
<td>26.62377</td>
<td>45</td>
</tr>
<tr>
<td>ATATG</td>
<td>10181510</td>
<td>1.5815413</td>
<td>26.770157</td>
<td>13</td>
</tr>
<tr>
<td>ACGCA</td>
<td>6997330</td>
<td>1.5797281</td>
<td>38.010635</td>
<td>52</td>
</tr>
<tr>
<td>TAAGT</td>
<td>9898835</td>
<td>1.5376321</td>
<td>26.669594</td>
<td>47</td>
</tr>
<tr>
<td>GTACG</td>
<td>6964260</td>
<td>1.4023975</td>
<td>33.8264</td>
<td>50</td>
</tr>
<tr>
<td>GTCTA</td>
<td>8300040</td>
<td>1.3841052</td>
<td>28.333914</td>
<td>44</td>
</tr>
<tr>
<td>CCGGT</td>
<td>6385335</td>
<td>1.3803858</td>
<td>36.224308</td>
<td>60</td>
</tr>
<tr>
<td>TACGC</td>
<td>6507985</td>
<td>1.3083571</td>
<td>33.78271</td>
<td>51</td>
</tr>
<tr>
<td>CGGTA</td>
<td>2836845</td>
<td>0.5712574</td>
<td>11.609342</td>
<td>60</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
