<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Fri 29 Aug 2014<br />
SRR317069_pe_1.f.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>SRR317069_pe_1.f.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>97891379</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>64</td>
</tr>
<tr>
<td>%GC</td>
<td>45</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/warning.png" alt="[WARN]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>CCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCA</td>
<td>258757</td>
<td>0.26433073335293394</td>
<td>No Hit</td>
</tr>
<tr>
<td>CCCGTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACC</td>
<td>249227</td>
<td>0.2545954531910313</td>
<td>No Hit</td>
</tr>
<tr>
<td>GCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGAT</td>
<td>232642</td>
<td>0.2376532053961565</td>
<td>No Hit</td>
</tr>
<tr>
<td>GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAGCGAGCGACCAAA</td>
<td>212642</td>
<td>0.21722239708156527</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGCTGGATAGTAGGTAGGGACAGTGGGAATCTCGTTCATCCATTCATGCGCGTCACTAATTAGA</td>
<td>173213</td>
<td>0.17694408002976442</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CTGGG</td>
<td>10533515</td>
<td>2.4160671</td>
<td>8.953529</td>
<td>1</td>
</tr>
<tr>
<td>CCAGC</td>
<td>9314765</td>
<td>2.379335</td>
<td>5.057728</td>
<td>30</td>
</tr>
<tr>
<td>GGAGG</td>
<td>13767110</td>
<td>2.334017</td>
<td>6.7171235</td>
<td>2</td>
</tr>
<tr>
<td>GGCTG</td>
<td>9703405</td>
<td>2.225665</td>
<td>6.9070263</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>13019675</td>
<td>2.2073</td>
<td>7.409999</td>
<td>2</td>
</tr>
<tr>
<td>GCTGG</td>
<td>9547410</td>
<td>2.1898847</td>
<td>11.057559</td>
<td>1</td>
</tr>
<tr>
<td>CACAG</td>
<td>11421370</td>
<td>2.0671124</td>
<td>5.4522867</td>
<td>30</td>
</tr>
<tr>
<td>GGCAG</td>
<td>9914640</td>
<td>1.9269879</td>
<td>6.059241</td>
<td>1</td>
</tr>
<tr>
<td>GGGGG</td>
<td>9170620</td>
<td>1.914066</td>
<td>10.484468</td>
<td>1</td>
</tr>
<tr>
<td>CCACA</td>
<td>9211005</td>
<td>1.9111456</td>
<td>5.4498425</td>
<td>29</td>
</tr>
<tr>
<td>TTCAT</td>
<td>10954795</td>
<td>1.8754967</td>
<td>5.2454963</td>
<td>42</td>
</tr>
<tr>
<td>GTGGG</td>
<td>9319740</td>
<td>1.8646556</td>
<td>10.506806</td>
<td>1</td>
</tr>
<tr>
<td>GGGGA</td>
<td>10718005</td>
<td>1.8170847</td>
<td>8.716661</td>
<td>1</td>
</tr>
<tr>
<td>CTGGA</td>
<td>9710390</td>
<td>1.8091534</td>
<td>5.887188</td>
<td>3</td>
</tr>
<tr>
<td>TCACT</td>
<td>8597890</td>
<td>1.7603867</td>
<td>5.4551797</td>
<td>52</td>
</tr>
<tr>
<td>CATCC</td>
<td>6991260</td>
<td>1.7118899</td>
<td>5.795355</td>
<td>36</td>
</tr>
<tr>
<td>GTGTG</td>
<td>8791720</td>
<td>1.6861857</td>
<td>6.094512</td>
<td>1</td>
</tr>
<tr>
<td>TGGGG</td>
<td>8387740</td>
<td>1.6781849</td>
<td>7.910941</td>
<td>1</td>
</tr>
<tr>
<td>TCATC</td>
<td>7881140</td>
<td>1.6136347</td>
<td>5.0868816</td>
<td>35</td>
</tr>
<tr>
<td>GCTCT</td>
<td>6326085</td>
<td>1.5945868</td>
<td>6.0131955</td>
<td>17</td>
</tr>
<tr>
<td>GGCCA</td>
<td>7127965</td>
<td>1.5882118</td>
<td>5.156566</td>
<td>1</td>
</tr>
<tr>
<td>ATCCA</td>
<td>9084315</td>
<td>1.5760673</td>
<td>7.571522</td>
<td>37</td>
</tr>
<tr>
<td>GGCAT</td>
<td>8383770</td>
<td>1.5619894</td>
<td>5.482129</td>
<td>4</td>
</tr>
<tr>
<td>CCATT</td>
<td>7533445</td>
<td>1.5424455</td>
<td>5.0820293</td>
<td>39</td>
</tr>
<tr>
<td>GGGCA</td>
<td>7910660</td>
<td>1.5374987</td>
<td>6.3311863</td>
<td>1</td>
</tr>
<tr>
<td>CATTC</td>
<td>7426165</td>
<td>1.5204804</td>
<td>5.1437607</td>
<td>40</td>
</tr>
<tr>
<td>GGGTG</td>
<td>7579340</td>
<td>1.5164435</td>
<td>6.383362</td>
<td>2</td>
</tr>
<tr>
<td>TCCAT</td>
<td>7398765</td>
<td>1.5148703</td>
<td>5.0916114</td>
<td>38</td>
</tr>
<tr>
<td>GGGGT</td>
<td>7427170</td>
<td>1.4859979</td>
<td>7.6692896</td>
<td>2</td>
</tr>
<tr>
<td>CATGC</td>
<td>6956550</td>
<td>1.485845</td>
<td>5.197397</td>
<td>44</td>
</tr>
<tr>
<td>ATCTC</td>
<td>7195995</td>
<td>1.4733539</td>
<td>5.034242</td>
<td>28</td>
</tr>
<tr>
<td>ACCAC</td>
<td>7011455</td>
<td>1.4547719</td>
<td>5.322799</td>
<td>28</td>
</tr>
<tr>
<td>AGCTC</td>
<td>6698465</td>
<td>1.4307207</td>
<td>5.1617064</td>
<td>16</td>
</tr>
<tr>
<td>GGGCT</td>
<td>6191905</td>
<td>1.4202342</td>
<td>6.247887</td>
<td>3</td>
</tr>
<tr>
<td>GTGGT</td>
<td>7172470</td>
<td>1.3756256</td>
<td>5.3764405</td>
<td>1</td>
</tr>
<tr>
<td>GGGGC</td>
<td>5257690</td>
<td>1.2580386</td>
<td>6.6832633</td>
<td>2</td>
</tr>
<tr>
<td>CGTTC</td>
<td>4638705</td>
<td>1.1692568</td>
<td>5.221307</td>
<td>32</td>
</tr>
<tr>
<td>TATCC</td>
<td>5668515</td>
<td>1.1606079</td>
<td>5.2052464</td>
<td>36</td>
</tr>
<tr>
<td>GAGCG</td>
<td>5832835</td>
<td>1.1336573</td>
<td>5.050758</td>
<td>54</td>
</tr>
<tr>
<td>CGGGG</td>
<td>4731380</td>
<td>1.1321055</td>
<td>8.748814</td>
<td>1</td>
</tr>
<tr>
<td>TCGGC</td>
<td>3715410</td>
<td>0.97697455</td>
<td>5.0454435</td>
<td>4</td>
</tr>
<tr>
<td>CCCGT</td>
<td>2992540</td>
<td>0.90210545</td>
<td>7.361353</td>
<td>1</td>
</tr>
<tr>
<td>GGCGG</td>
<td>3749915</td>
<td>0.8972644</td>
<td>6.1349864</td>
<td>1</td>
</tr>
<tr>
<td>TCTCG</td>
<td>3505300</td>
<td>0.8835647</td>
<td>5.252689</td>
<td>29</td>
</tr>
<tr>
<td>CCGTC</td>
<td>2924500</td>
<td>0.88159466</td>
<td>6.879332</td>
<td>2</td>
</tr>
<tr>
<td>CTCGT</td>
<td>3465455</td>
<td>0.87352115</td>
<td>5.0814104</td>
<td>30</td>
</tr>
<tr>
<td>GTCGG</td>
<td>3778435</td>
<td>0.86665773</td>
<td>5.3554792</td>
<td>1</td>
</tr>
<tr>
<td>GCGGG</td>
<td>3558950</td>
<td>0.85157114</td>
<td>5.63624</td>
<td>1</td>
</tr>
<tr>
<td>GCGCG</td>
<td>3084540</td>
<td>0.84611666</td>
<td>5.2665653</td>
<td>47</td>
</tr>
<tr>
<td>CGGGC</td>
<td>2985705</td>
<td>0.81900537</td>
<td>5.1313586</td>
<td>1</td>
</tr>
<tr>
<td>CGCTG</td>
<td>3088170</td>
<td>0.8120406</td>
<td>5.310421</td>
<td>1</td>
</tr>
<tr>
<td>TGCGC</td>
<td>3056810</td>
<td>0.8037943</td>
<td>5.1903563</td>
<td>46</td>
</tr>
<tr>
<td>GCGTC</td>
<td>3052810</td>
<td>0.8027426</td>
<td>5.074395</td>
<td>49</td>
</tr>
<tr>
<td>CGTCG</td>
<td>2916965</td>
<td>0.7670219</td>
<td>5.1484685</td>
<td>2</td>
</tr>
<tr>
<td>CGACC</td>
<td>2183650</td>
<td>0.5577849</td>
<td>5.0393367</td>
<td>59</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
